---
title: "R Notebook"
output:
  pdf_document: default
  html_document:
    df_print: paged
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Cmd+Shift+Enter*. 

```{r}

###first isolate monocytes from donors from the pbmc covid single cell paper.GSE150728
###start with covid positive samples
covid_555_1 <- readRDS(file = "~/Desktop/Covid analysis/VC start/GSM4557327_555_1_cell.counts.matrices.rds")
covid_555_1 <- CreateSeuratObject(counts = covid_555_1$exon, project = "covid", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
covid_555_1 ###10813 samples with 26141 features
Idents(object = covid_555_1) <- "Cov2 Sample 1"
covid_555_1$sample <- "Cov2 Sample 1"
covid_555_1$stim <- "Cov2"
levels(covid_555_1)
covid_555_1

covid_555_1[["percent.mt"]] <- PercentageFeatureSet(covid_555_1, pattern = "^MT-")

VlnPlot(covid_555_1, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid_555_1 <- FeatureScatter(covid_555_1, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid_555_1 <- FeatureScatter(covid_555_1, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid_555_1, plot2.covid_555_1))
covid_555_1<- subset(covid_555_1, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 20)

##you can do sct transform 
covid_555_1 <- SCTransform(covid_555_1, vars.to.regress = "percent.mt", verbose = FALSE)
covid_555_1

Idents(object = covid_555_1) <- "sample"
levels(covid_555_1)


###linear dimensional reduction
covid_555_1<- RunPCA(covid_555_1, features = VariableFeatures(object = covid_555_1))
print(covid_555_1[["pca"]], dims = 1:5, nfeatures = 5)
VizDimLoadings(covid_555_1, dims = 1:2, reduction = "pca")
DimPlot(covid_555_1, reduction = "pca")
DimHeatmap(covid_555_1, dims = 1, cells = 500, balanced = TRUE)
DimHeatmap(covid_555_1, dims = 1:15, cells = 500, balanced = TRUE)

####determine dimentionality of your data
ElbowPlot(covid_555_1, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid_555_1[["pca"]]@stdev / sum(covid_555_1[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
# Determine which PC exhibits cumulative percent greater than 90% and % variation associated with the PC as less than 5
co1 <- which(cumu > 90 & pct < 5)[1]
co1
#The first metric returns PC43 as the PC matching these requirements. Letâ€™s check the second metric, which identifies the PC where the percent change in variation between consequtive PCs is less than 0.1%:
# Determine the difference between variation of PC and subsequent PC
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
# last point where change of % of variation is more than 0.1%.
co2
#Now, to determine the selection of PCs, we will use the minimum of the two metrics:
# Minimum of the two calculation
pcs <- min(co1, co2)
pcs

###clustering
covid_555_1 <- FindNeighbors(covid_555_1, dims = 1:14)
covid_555_1 <- FindClusters(covid_555_1, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid_555_1 <- RunUMAP(covid_555_1, dims = 1:14)
DimPlot(covid_555_1, reduction = "umap", min.dist = 0.75, pt.size =0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid_555_1.markers <- FindAllMarkers(covid_555_1, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid_555_1.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


#check for mature monocyte markers and plot to see which cluster has highest level of expression

VlnPlot(covid_555_1, features = c("CD14", "MS4A7", "FCGR3A"))

# you can plot raw counts as well
VlnPlot(covid_555_1, features = c("FCGR3B", "CXCR2","FCGR3A"), slot = "counts", log = TRUE)
FeaturePlot(covid_555_1, features = c("FCGR3B", "CXCR2","FCGR3A"))


levels(covid_555_1)
new.cluster.ids <- c("0", "1", "2", "3", "CD14 monocytes", "5","CD14 monocytes", "B cells", "8","9","CD16 monocytes", "11", "12", "13", "14")
names(new.cluster.ids) <- levels(covid_555_1)
covid_555_1 <- RenameIdents(covid_555_1, new.cluster.ids)
levels(covid_555_1)
DimPlot(covid_555_1, reduction = "umap", label = TRUE, pt.size = 1, label.size = 10) + NoLegend()

##save before extracting CD16 monocytes

saveRDS(covid_555_1, file = "~/Desktop/Covid analysis/VC start/covid_555_1.rds")

table(Idents(covid_555_1)) ###114 cd16 monocytes. proceed. 


covid_551_1_monocytes <- subset(covid_555_1, idents = "CD16 monocytes")
table(Idents(covid_551_1_monocytes))

Idents(object = covid_551_1_monocytes) <- "Cov2 Sample 1 monocytes"
covid_551_1_monocytes$sample <- "Cov2 Sample 1"
covid_551_1_monocytes$stim <- "Cov2"
levels(covid_551_1_monocytes)
covid_551_1_monocytes


saveRDS(covid_551_1_monocytes, file = "~/Desktop/Covid analysis/VC start/covid_551_1_monocytes.rds")

################################################################### 555_2
###do the same for the other donors too
covid_555_2 <- readRDS(file = "~/Desktop/Covid analysis/VC start/GSM4557328_555_2_cell.counts.matrices.rds")
covid_555_2 <- CreateSeuratObject(counts = covid_555_2$exon, project = "covid", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
covid_555_2
Idents(object = covid_555_2) <- "Cov2 Sample 2"
covid_555_2$sample <- "Cov2 Sample 2"
covid_555_2$stim <- "Cov2"
levels(covid_555_2)
covid_555_2

covid_555_2[["percent.mt"]] <- PercentageFeatureSet(covid_555_2, pattern = "^MT-")

VlnPlot(covid_555_2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid_555_2 <- FeatureScatter(covid_555_2, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid_555_2 <- FeatureScatter(covid_555_2, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid_555_2, plot2.covid_555_2))
##### decided on the following parameters for subsetting QC percent.mt < 15%, nFeature_RNA > 200 < 8,000
covid_555_2 <- subset(covid_555_2, subset = nFeature_RNA > 200 & nFeature_RNA < 5000 & percent.mt < 20)


##you can do sct transform 
covid_555_2 <- SCTransform(covid_555_2, vars.to.regress = "percent.mt", verbose = FALSE)
covid_555_2

Idents(object = covid_555_2) <- "sample"
levels(covid_555_2)

#############
###linear dimensional reduction
covid_555_2<- RunPCA(covid_555_2, features = VariableFeatures(object = covid_555_2)) 
print(covid_555_2[["pca"]], dims = 1:5, nfeatures = 5)
VizDimLoadings(covid_555_2, dims = 1:2, reduction = "pca")
DimPlot(covid_555_2, reduction = "pca")
DimHeatmap(covid_555_2, dims = 1, cells = 500, balanced = TRUE)
DimHeatmap(covid_555_2, dims = 1:15, cells = 500, balanced = TRUE)

####determine dimentionality of your data
ElbowPlot(covid_555_2, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid_555_2[["pca"]]@stdev / sum(covid_555_2[["pca"]]@stdev) * 100

# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1

co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2

pcs <- min(co1, co2)
pcs

###clustering
covid_555_2 <- FindNeighbors(covid_555_2, dims = 1:14)
covid_555_2 <- FindClusters(covid_555_2, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid_555_2 <- RunUMAP(covid_555_2, dims = 1:14)
DimPlot(covid_555_2, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)

# find markers for every cluster compared to all remaining cells, report only the positive ones
covid_555_2.markers <- FindAllMarkers(covid_555_2, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid_555_2.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


VlnPlot(covid_555_2, features = c("MS4A7" ,"FCGR3A","CD4"))
# you can plot raw counts as well
VlnPlot(covid_555_2, features = c("FCGR3B", "CXCR2","FCGR3A"), slot = "counts", log = TRUE)


FeaturePlot(covid_555_2, features = c("FCGR3B", "CXCR2","FCGR3A","FCER1A"))

top10 <- covid_555_2.markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_logFC)
DoHeatmap(covid_555_2, features = top10$gene) + NoLegend()


levels(covid_555_2)
new.cluster.ids <- c("CD4 T cells", "CD14 monocytes", "CD8 T cells", "CD16 monocytes", "NK cells", "B cells","Plasmablasts", "Plasmablasts", "Dendritic cells","Plasmablasts", "Neutrophils", "Platelets")
names(new.cluster.ids) <- levels(covid_555_2)
covid_555_2 <- RenameIdents(covid_555_2, new.cluster.ids)
levels(covid_555_2)
DimPlot(covid_555_2, reduction = "umap", label = TRUE, pt.size = 1, label.size = 8,repel=TRUE) + NoLegend()

##save before extracting CD16 monocytes

saveRDS(covid_555_2, file = "~/Desktop/Covid analysis/VC start/covid_555_2.rds")
covid_555_2 <- readRDS(file = "~/Desktop/Covid analysis/VC start/covid_555_2.rds")
table(Idents(covid_555_2))


covid_555_2_monocytes <- subset(covid_555_2, idents = c("CD16 monocytes"))
table(Idents(covid_555_2_monocytes))

Idents(object = covid_555_2_monocytes) <- "Cov2 Sample 1 monocytes"
covid_555_2_monocytes$sample <- "Cov2 Sample 1"
covid_555_2_monocytes$stim <- "Cov2"
levels(covid_555_2_monocytes)
covid_555_2_monocytes


saveRDS(covid_555_2_monocytes, file = "~/Desktop/Covid analysis/VC start/covid_555_2_monocytes.rds")

####557
covid_557 <- readRDS(file = "~/Desktop/Covid analysis/VC start/GSM4557330_557_cell.counts.matrices.rds")
covid_557 <- CreateSeuratObject(counts = covid_557$exon, project = "covid", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
covid_557
Idents(object = covid_557) <- "Cov2 Sample 2"
covid_557$sample <- "Cov2 Sample 2"
covid_557$stim <- "Cov2"
levels(covid_557)
covid_557

covid_557[["percent.mt"]] <- PercentageFeatureSet(covid_557, pattern = "^MT-")

VlnPlot(covid_557, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid_557 <- FeatureScatter(covid_557, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid_557 <- FeatureScatter(covid_557, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid_557, plot2.covid_557))
##### decided on the following parameters for subsetting QC percent.mt < 15%, nFeature_RNA > 200 < 8,000
covid_557 <- subset(covid_557, subset = nFeature_RNA > 200 & nFeature_RNA < 5000 & percent.mt < 20)


##you can do sct transform instead 
covid_557 <- SCTransform(covid_557, vars.to.regress = "percent.mt", verbose = FALSE)
covid_557

Idents(object = covid_557) <- "sample"
levels(covid_557)

#############
###linear dimensional reduction
covid_557<- RunPCA(covid_557, features = VariableFeatures(object = covid_557)) 
print(covid_557[["pca"]], dims = 1:5, nfeatures = 5)
VizDimLoadings(covid_557, dims = 1:2, reduction = "pca")
DimPlot(covid_557, reduction = "pca")
DimHeatmap(covid_557, dims = 1, cells = 500, balanced = TRUE)
DimHeatmap(covid_557, dims = 1:15, cells = 500, balanced = TRUE)

####determine dimentionality of your data
ElbowPlot(covid_557, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid_557[["pca"]]@stdev / sum(covid_557[["pca"]]@stdev) * 100

# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1

co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2

pcs <- min(co1, co2)
pcs

###clustering
covid_557 <- FindNeighbors(covid_557, dims = 1:16)
covid_557 <- FindClusters(covid_557, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid_557 <- RunUMAP(covid_557, dims = 1:16)
DimPlot(covid_557, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid_557.markers <- FindAllMarkers(covid_557, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid_557.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


VlnPlot(covid_557, features = c("MS4A7", "CD14", "FCGR3A"))
# you can plot raw counts as well
VlnPlot(covid_557, features = c("FCGR3B", "CXCR2","FCGR3A"), slot = "counts", log = TRUE)


FeaturePlot(covid_557, features = c("FCGR3B", "CXCR2","FCGR3A"))

top10 <- covid_557.markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_logFC)
DoHeatmap(covid_557, features = top10$gene) + NoLegend()

##open the see file to look at the genes, and identify them according to clusters 


#cluster 11 - 
#cluster 12 - 
#cluster 13 - 
#cluster 14 - 
#cluster 15 - cd16 monocytes

levels(covid_557)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5","6", "7", "8","9","10", "11", "12", "13", "14", "15", "CD16 monocytes")
names(new.cluster.ids) <- levels(covid_557)
covid_557 <- RenameIdents(covid_557, new.cluster.ids)
levels(covid_557)
DimPlot(covid_557, reduction = "umap", label = TRUE, pt.size = 1, label.size = 10) + NoLegend()

##save before extracting CD16 monocytes

saveRDS(covid_557, file = "~/Desktop/Covid analysis/VC start/covid_557.rds")

table(Idents(covid_557))

WhichCells(covid_557, idents = "CD16 monocytes") ###only 51 cells 

covid_557_monocytes <- subset(covid_557, idents = "CD16 monocytes")
table(Idents(covid_555_7_monocytes))

Idents(object = covid_557_monocytes) <- "Cov2 Sample 1 monocytes"
covid_557_monocytes$sample <- "Cov2 Sample 1"
covid_557_monocytes$stim <- "Cov2"
levels(covid_557_monocytes)
covid_557_monocytes


saveRDS(covid_557_monocytes, file = "~/Desktop/Covid analysis/VC start/covid_557_monocytes.rds")

###################################################################
###do the same for the other donors too
covid_558 <- readRDS(file = "~/Desktop/Covid analysis/VC start/GSM4557331_558_cell.counts.matrices.rds")
covid_558 <- CreateSeuratObject(counts = covid_558$exon, project = "covid", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
covid_558
Idents(object = covid_558) <- "Cov2 Sample 2"
covid_558$sample <- "Cov2 Sample 2"
covid_558$stim <- "Cov2"
levels(covid_558)
covid_558

covid_558[["percent.mt"]] <- PercentageFeatureSet(covid_558, pattern = "^MT-")

VlnPlot(covid_558, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid_558 <- FeatureScatter(covid_558, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid_558 <- FeatureScatter(covid_558, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid_558, plot2.covid_558))
##### decided on the following parameters for subsetting QC percent.mt < 15%, nFeature_RNA > 200 < 8,000
covid_558 <- subset(covid_558, subset = nFeature_RNA > 200 & nFeature_RNA < 5000 & percent.mt < 20)


##you can do sct transform instead 
covid_558 <- SCTransform(covid_558, vars.to.regress = "percent.mt", verbose = FALSE)
covid_558

Idents(object = covid_558) <- "sample"
levels(covid_558)

#############
###linear dimensional reduction
covid_558<- RunPCA(covid_558, features = VariableFeatures(object = covid_558)) 
print(covid_558[["pca"]], dims = 1:5, nfeatures = 5)
VizDimLoadings(covid_558, dims = 1:2, reduction = "pca")
DimPlot(covid_558, reduction = "pca")
DimHeatmap(covid_558, dims = 1, cells = 500, balanced = TRUE)
DimHeatmap(covid_558, dims = 1:15, cells = 500, balanced = TRUE)

####determine dimentionality of your data
ElbowPlot(covid_558, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid_558[["pca"]]@stdev / sum(covid_558[["pca"]]@stdev) * 100

# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1

co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2

pcs <- min(co1, co2)
pcs

###clustering
covid_558 <- FindNeighbors(covid_558, dims = 1:10)
covid_558 <- FindClusters(covid_558, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid_558 <- RunUMAP(covid_558, dims = 1:10)
DimPlot(covid_558, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid_558.markers <- FindAllMarkers(covid_558, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid_558.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


VlnPlot(covid_558, features = c("CD14", "MS4A7", "FCGR3A"))

# you can plot raw counts as well
VlnPlot(covid_558, features = c("FCGR3B", "CXCR2","FCGR3A"), slot = "counts", log = TRUE)

###no monocytes

################################################################### 556

covid_556 <- readRDS(file = "~/Desktop/Covid analysis/VC start/GSM4557329_556_cell.counts.matrices.rds")
covid_556 <- CreateSeuratObject(counts = covid_556$exon, project = "covid", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
covid_556
Idents(object = covid_556) <- "Cov2 Sample 2"
covid_556$sample <- "Cov2 Sample 2"
covid_556$stim <- "Cov2"
levels(covid_556)
covid_556

covid_556[["percent.mt"]] <- PercentageFeatureSet(covid_556, pattern = "^MT-")

VlnPlot(covid_556, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid_556 <- FeatureScatter(covid_556, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid_556 <- FeatureScatter(covid_556, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid_556, plot2.covid_556))
##### decided on the following parameters for subsetting QC percent.mt < 15%, nFeature_RNA > 200 < 8,000
covid_556 <- subset(covid_556, subset = nFeature_RNA > 200 & nFeature_RNA < 5000 & percent.mt < 20)


##you can do sct transform instead 
covid_556 <- SCTransform(covid_556, vars.to.regress = "percent.mt", verbose = FALSE)
covid_556

Idents(object = covid_556) <- "sample"
levels(covid_556)

#############
###linear dimensional reduction
covid_556<- RunPCA(covid_556, features = VariableFeatures(object = covid_556)) 
print(covid_556[["pca"]], dims = 1:5, nfeatures = 5)
VizDimLoadings(covid_556, dims = 1:2, reduction = "pca")
DimPlot(covid_556, reduction = "pca")
DimHeatmap(covid_556, dims = 1, cells = 500, balanced = TRUE)
DimHeatmap(covid_556, dims = 1:15, cells = 500, balanced = TRUE)

####determine dimentionality of your data
ElbowPlot(covid_556, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid_556[["pca"]]@stdev / sum(covid_556[["pca"]]@stdev) * 100

# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1

co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2

pcs <- min(co1, co2)
pcs

###clustering
covid_556 <- FindNeighbors(covid_556, dims = 1:12)
covid_556 <- FindClusters(covid_556, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid_556 <- RunUMAP(covid_556, dims = 1:12)
DimPlot(covid_556, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)

# find markers for every cluster compared to all remaining cells, report only the positive ones
covid_556.markers <- FindAllMarkers(covid_556, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid_556.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


VlnPlot(covid_556, features = c("MS4A7", "CD14", "FCGR3A"))
# you can plot raw counts as well
VlnPlot(covid_556, features = c("FCGR3B", "CXCR2","FCGR3A"), slot = "counts", log = TRUE)


FeaturePlot(covid_556, features = c("FCGR3B", "CXCR2","FCGR3A"))

top10 <- covid_556.markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_logFC)
DoHeatmap(covid_556, features = top10$gene) + NoLegend()

##open the see file to look at the genes, and identify them according to clusters 


levels(covid_556)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5","6", "7", "8","CD16 monocytes","10")
names(new.cluster.ids) <- levels(covid_556)
covid_556 <- RenameIdents(covid_556, new.cluster.ids)
levels(covid_556)
DimPlot(covid_556, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes

saveRDS(covid_556, file = "~/Desktop/Covid analysis/VC start/covid_556.rds")

table(Idents(covid_556))

WhichCells(covid_556, idents = "CD16 monocytes") ###very few cells = 29

covid_556_monocytes <- subset(covid_556, idents = "CD16 monocytes")
table(Idents(covid_556_monocytes))

Idents(object = covid_556_monocytes) <- "Cov2 Sample 1 monocytes"
covid_556_monocytes$sample <- "Cov2 Sample 1"
covid_556_monocytes$stim <- "Cov2"
levels(covid_556_monocytes)
covid_556_monocytes


saveRDS(covid_556_monocytes, file = "~/Desktop/Covid analysis/VC start/covid_556_monocytes.rds")

################################################################### 559

covid_559 <- readRDS(file = "~/Desktop/Covid analysis/VC start/GSM4557332_559_cell.counts.matrices.rds")
covid_559 <- CreateSeuratObject(counts = covid_559$exon, project = "covid", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
covid_559
Idents(object = covid_559) <- "Cov2 Sample 2"
covid_559$sample <- "Cov2 Sample 2"
covid_559$stim <- "Cov2"
levels(covid_559)
covid_559

covid_559[["percent.mt"]] <- PercentageFeatureSet(covid_559, pattern = "^MT-")

VlnPlot(covid_559, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid_559 <- FeatureScatter(covid_559, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid_559 <- FeatureScatter(covid_559, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid_559, plot2.covid_559))
##### decided on the following parameters for subsetting QC percent.mt < 15%, nFeature_RNA > 200 < 8,000
covid_559 <- subset(covid_559, subset = nFeature_RNA > 200 & nFeature_RNA < 5000 & percent.mt < 22)


##you can do sct transform instead 
covid_559 <- SCTransform(covid_559, vars.to.regress = "percent.mt", verbose = FALSE)
covid_559

Idents(object = covid_559) <- "sample"
levels(covid_559)

#############  
###linear dimensional reduction
covid_559<- RunPCA(covid_559, features = VariableFeatures(object = covid_559)) 
print(covid_559[["pca"]], dims = 1:5, nfeatures = 5)
VizDimLoadings(covid_559, dims = 1:2, reduction = "pca")
DimPlot(covid_559, reduction = "pca")
DimHeatmap(covid_559, dims = 1, cells = 500, balanced = TRUE)
DimHeatmap(covid_559, dims = 1:15, cells = 500, balanced = TRUE)

####determine dimentionality of your data
ElbowPlot(covid_559, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid_559[["pca"]]@stdev / sum(covid_559[["pca"]]@stdev) * 100

# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1

co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2

pcs <- min(co1, co2)
pcs

###clustering
covid_559 <- FindNeighbors(covid_559, dims = 1:12)
covid_559 <- FindClusters(covid_559, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid_559 <- RunUMAP(covid_559, dims = 1:12)
DimPlot(covid_559, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid_559.markers <- FindAllMarkers(covid_559, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid_559.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


VlnPlot(covid_559, features = c("FCGR3B", "CXCR2", "FCGR3A"))
# you can plot raw counts as well
VlnPlot(covid_559, features = c("FCGR3B", "CXCR2","FCGR3A"), slot = "counts", log = TRUE)


FeaturePlot(covid_559, features = c("FCGR3B", "CXCR2","FCGR3A"))

top10 <- covid_559.markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_logFC)
DoHeatmap(covid_559, features = top10$gene) + NoLegend()

##open the see file to look at the genes, and identify them according to clusters 

levels(covid_559)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5","6", "7", "8","9", "10", "11", "12", "CD16 monocytes")
names(new.cluster.ids) <- levels(covid_559)
covid_559 <- RenameIdents(covid_559, new.cluster.ids)
levels(covid_559)
DimPlot(covid_559, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes

saveRDS(covid_559, file = "~/Desktop/Covid analysis/VC start/covid_559.rds")

table(Idents(covid_559))

WhichCells(covid_559, idents = "CD16 monocytes") ###dont use downstream. few cells 

covid_559_monocytes <- subset(covid_559, idents = "CD16 monocytes")
table(Idents(covid_559_monocytes))

Idents(object = covid_559_monocytes) <- "Cov2 Sample 1 monocytes"
covid_559_monocytes$sample <- "Cov2 Sample 1"
covid_559_monocytes$stim <- "Cov2"
levels(covid_559_monocytes)
covid_559_monocytes


saveRDS(covid_559_monocytes, file = "~/Desktop/Covid analysis/VC start/covid_559_monocytes.rds")

################################################################### 559
###do the same for the other donors too
covid_561 <- readRDS(file = "~/Desktop/Covid analysis/VC start/GSM4557333_561_cell.counts.matrices.rds")
covid_561 <- CreateSeuratObject(counts = covid_561$exon, project = "covid", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
covid_561
Idents(object = covid_561) <- "Cov2 Sample 2"
covid_561$sample <- "Cov2 Sample 2"
covid_561$stim <- "Cov2"
levels(covid_561)
covid_561

covid_561[["percent.mt"]] <- PercentageFeatureSet(covid_561, pattern = "^MT-")

VlnPlot(covid_561, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid_561 <- FeatureScatter(covid_561, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid_561 <- FeatureScatter(covid_561, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid_561, plot2.covid_561))
##### decided on the following parameters for subsetting QC percent.mt < 15%, nFeature_RNA > 200 < 8,000
covid_561 <- subset(covid_561, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 20)


##you can do sct transform instead 
covid_561 <- SCTransform(covid_561, vars.to.regress = "percent.mt", verbose = FALSE)
covid_561

Idents(object = covid_561) <- "sample"
levels(covid_561)

#############  
###linear dimensional reduction
covid_561<- RunPCA(covid_561, features = VariableFeatures(object = covid_561)) 
print(covid_561[["pca"]], dims = 1:5, nfeatures = 5)
VizDimLoadings(covid_561, dims = 1:2, reduction = "pca")
DimPlot(covid_561, reduction = "pca")
DimHeatmap(covid_561, dims = 1, cells = 500, balanced = TRUE)
DimHeatmap(covid_561, dims = 1:15, cells = 500, balanced = TRUE)

####determine dimentionality of your data
ElbowPlot(covid_561, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid_561[["pca"]]@stdev / sum(covid_561[["pca"]]@stdev) * 100

# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1

co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2

pcs <- min(co1, co2)
pcs

###clustering
covid_561 <- FindNeighbors(covid_561, dims = 1:14)
covid_561 <- FindClusters(covid_561, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid_561 <- RunUMAP(covid_561, dims = 1:14)
DimPlot(covid_561, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid_561.markers <- FindAllMarkers(covid_561, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid_561.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


VlnPlot(covid_561, features = c("CD14", "MS4A7", "FCGR3A"))
# you can plot raw counts as well
VlnPlot(covid_561, features = c("FCGR3B", "CXCR2","FCGR3A"), slot = "counts", log = TRUE)


FeaturePlot(covid_561, features = c("FCGR3B", "CXCR2","FCGR3A"))

top10 <- covid_561.markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_logFC)
DoHeatmap(covid_561, features = top10$gene) + NoLegend()

##open the see file to look at the genes, and identify them according to clusters 

 

levels(covid_561)
new.cluster.ids <- c("0", "1", "2", "3", "CD16 monocytes", "5","6", "CD16 monocytes", "8","9", "10", "11", "12", "13")
names(new.cluster.ids) <- levels(covid_561)
covid_561 <- RenameIdents(covid_561, new.cluster.ids)
levels(covid_561)
DimPlot(covid_561, reduction = "umap", label = TRUE, pt.size = 1, label.size = 10) + NoLegend()

##save before extracting CD16 monocytes

saveRDS(covid_561, file = "~/Desktop/Covid analysis/VC start/covid_561.rds")

table(Idents(covid_561))

WhichCells(covid_561, idents = "CD16 monocytes")

covid_561_monocytes <- subset(covid_561, idents = "CD16 monocytes")
table(Idents(covid_561_monocytes))

Idents(object = covid_561_monocytes) <- "Cov2 Sample 1 monocytes"
covid_561_monocytes$sample <- "Cov2 Sample 1"
covid_561_monocytes$stim <- "Cov2"
levels(covid_561_monocytes)
covid_561_monocytes


saveRDS(covid_561_monocytes, file = "~/Desktop/Covid analysis/VC start/covid_561_monocytes.rds")



##### now proceed to covid samples from the covid flu paper
##put the zipped file in the GSE149689 then rename them to barcodes.tsv.gz; features.tsv.gz; matrix.mtx.gz
data <- Read10X(data.dir = "/Users/bermanlab/Desktop/Covid analysis/VC start/Single cell covid 091420/GSE149689")
                
data <- CreateSeuratObject(counts = data, project = "covid", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
data
table(Idents(data))
levels(data)

covid1 <- subset(data, idents = c("1"))
covid2 <- subset(data, idents = c("2"))
covid9 <- subset(data, idents = c("9"))
covid10 <- subset(data, idents = c("10"))
covid11 <- subset(data, idents = c("11"))
covid12 <- subset(data, idents = c("12"))
covid15 <- subset(data, idents = c("15"))
covid16 <- subset(data, idents = c("16"))
covid17 <- subset(data, idents = c("17"))
covid18 <- subset(data, idents = c("18"))
covid20 <- subset(data, idents = c("20"))

covid1
Idents(object = covid1) <- "Cov2 Sample 1"
covid1$sample <- "Cov2 Sample 1"
covid1$stim <- "Cov2"
levels(covid1)
covid1


covid2
Idents(object = covid2) <- "Cov2 Sample 2"
covid2$sample <- "Cov2 Sample 2"
covid2$stim <- "Cov2"
levels(covid2)
covid2

covid9
Idents(object = covid9) <- "Cov2 Sample 3"
covid9$sample <- "Cov2 Sample 3"
covid9$stim <- "Cov2"
levels(covid9)
covid9


covid10
Idents(object = covid10) <- "Cov2 Sample 4"
covid10$sample <- "Cov2 Sample 4"
covid10$stim <- "Cov2"
levels(covid10)
covid10

covid11
Idents(object = covid11) <- "Cov2 Sample 5"
covid11$sample <- "Cov2 Sample 5"
covid11$stim <- "Cov2"
levels(covid11)
covid11

covid12
Idents(object = covid12) <- "Cov2 Sample 6"
covid12$sample <- "Cov2 Sample 6"
covid12$stim <- "Cov2"
levels(covid12)
covid12

covid15
Idents(object = covid11) <- "Cov2 Sample 7"
covid15$sample <- "Cov2 Sample 7"
covid15$stim <- "Cov2"
levels(covid15)
covid15

covid16
Idents(object = covid16) <- "Cov2 Sample 8"
covid16$sample <- "Cov2 Sample 8"
covid16$stim <- "Cov2"
levels(covid16)
covid16

covid17
Idents(object = covid17) <- "Cov2 Sample 9"
covid17$sample <- "Cov2 Sample 9"
covid17$stim <- "Cov2"
levels(covid17)
covid17

covid18
Idents(object = covid18) <- "Cov2 Sample 10"
covid18$sample <- "Cov2 Sample 10"
covid18$stim <- "Cov2"
levels(covid18)
covid18

covid20
Idents(object = covid20) <- "Cov2 Sample 11"
covid20$sample <- "Cov2 Sample 11"
covid20$stim <- "Cov2"
levels(covid20)
covid20

rm(data)

covid1[["percent.mt"]] <- PercentageFeatureSet(covid1, pattern = "^MT-")
VlnPlot(covid1, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid1 <- FeatureScatter(covid1, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid1 <- FeatureScatter(covid1, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid1, plot2.covid1))
#For covid1 decided on the following parameters for subsetting QC percent.mt < 25%, nFeature_RNA > 200 < 6,000

covid2[["percent.mt"]] <- PercentageFeatureSet(covid2, pattern = "^MT-")
VlnPlot(covid2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid2 <- FeatureScatter(covid2, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid2 <- FeatureScatter(covid2, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid2, plot2.covid2))
#For covid1 decided on the following parameters for subsetting QC percent.mt < 25%, nFeature_RNA > 200 < 4,000

covid9[["percent.mt"]] <- PercentageFeatureSet(covid9, pattern = "^MT-")
VlnPlot(covid9, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid9 <- FeatureScatter(covid9, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid9 <- FeatureScatter(covid9, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid9, plot2.covid9))
#For covid1 decided on the following parameters for subsetting QC percent.mt < 15%, nFeature_RNA > 200 < 5,000

covid10[["percent.mt"]] <- PercentageFeatureSet(covid10, pattern = "^MT-")
VlnPlot(covid10, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid10 <- FeatureScatter(covid10, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid10 <- FeatureScatter(covid10, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid10, plot2.covid10))
#For covid1 decided on the following parameters for subsetting QC percent.mt < 15%, nFeature_RNA > 200 < 4,000

covid11[["percent.mt"]] <- PercentageFeatureSet(covid11, pattern = "^MT-")
VlnPlot(covid11, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid11 <- FeatureScatter(covid11, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid11 <- FeatureScatter(covid11, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid11, plot2.covid11))
#For covid1 decided on the following parameters for subsetting QC percent.mt < 25%, nFeature_RNA > 200 < 5,000

covid12[["percent.mt"]] <- PercentageFeatureSet(covid12, pattern = "^MT-")
VlnPlot(covid12, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid12 <- FeatureScatter(covid12, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid12 <- FeatureScatter(covid12, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid12, plot2.covid12))
#For covid1 decided on the following parameters for subsetting QC percent.mt < 25%, nFeature_RNA > 200 < 6,000

covid15[["percent.mt"]] <- PercentageFeatureSet(covid15, pattern = "^MT-")
VlnPlot(covid15, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid15 <- FeatureScatter(covid15, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid15 <- FeatureScatter(covid15, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid15, plot2.covid15))
#For covid1 decided on the following parameters for subsetting QC percent.mt < 15%, nFeature_RNA > 200 < 6,000

covid16[["percent.mt"]] <- PercentageFeatureSet(covid16, pattern = "^MT-")
VlnPlot(covid16, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid16 <- FeatureScatter(covid16, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid16 <- FeatureScatter(covid16, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid16, plot2.covid16))
#For covid1 decided on the following parameters for subsetting QC percent.mt < 25%, nFeature_RNA > 200 < 6,000

covid17[["percent.mt"]] <- PercentageFeatureSet(covid17, pattern = "^MT-")
VlnPlot(covid17, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid1 <- FeatureScatter(covid17, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid1 <- FeatureScatter(covid17, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid17, plot2.covid17))
#For covid1 decided on the following parameters for subsetting QC percent.mt < 15%, nFeature_RNA > 200 < 5,000

covid18[["percent.mt"]] <- PercentageFeatureSet(covid18, pattern = "^MT-")
VlnPlot(covid18, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid1 <- FeatureScatter(covid18, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid1 <- FeatureScatter(covid18, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid18, plot2.covid18))
#For covid1 decided on the following parameters for subsetting QC percent.mt < 25%, nFeature_RNA > 200 < 6,000

covid20[["percent.mt"]] <- PercentageFeatureSet(covid20, pattern = "^MT-")
VlnPlot(covid20, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.covid20 <- FeatureScatter(covid20, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.covid20 <- FeatureScatter(covid20, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.covid20, plot2.covid20))
#For covid1 decided on the following parameters for subsetting QC percent.mt < 25%, nFeature_RNA > 200 < 5,000

rm(plot1.covid,plot1.covid1,plot1.covid10,plot1.covid11,plot1.covid12,plot1.covid15,plot1.covid16,plot1.covid2,plot1.covid20,plot1.covid9,plot2.covid1)
rm(plot2.covid9,plot2.covid,plot2.covid1,plot2.covid11,plot2.covid12,plot2.covid15,plot2.covid16,plot2.covid2,plot2.covid20)



#subset the data based on QC metrics
covid1<- subset(covid1, subset = nFeature_RNA > 200 & nFeature_RNA < 6000 & percent.mt < 25)
covid2<- subset(covid2, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 25)
covid9<- subset(covid9, subset = nFeature_RNA > 200 & nFeature_RNA < 5000 & percent.mt < 15)
covid10<- subset(covid10, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 15)
covid11 <- subset(covid11, subset = nFeature_RNA > 200 & nFeature_RNA < 5000 & percent.mt < 25)
covid12 <- subset(covid12, subset = nFeature_RNA > 200 & nFeature_RNA < 6000 & percent.mt < 25)
covid15 <- subset(covid15, subset = nFeature_RNA > 200 & nFeature_RNA < 6000 & percent.mt < 15)
covid16 <- subset(covid16, subset = nFeature_RNA > 200 & nFeature_RNA < 6000 & percent.mt < 25)
covid17 <- subset(covid17, subset = nFeature_RNA > 200 & nFeature_RNA < 5000 & percent.mt < 15)
covid18 <- subset(covid18, subset = nFeature_RNA > 200 & nFeature_RNA < 6000 & percent.mt < 25)
covid20 <- subset(covid20, subset = nFeature_RNA > 200 & nFeature_RNA < 5000 & percent.mt < 25)

###check no of cells for each group
covid1
covid2
covid9 #very few
covid10 #very few
covid11
covid12
covid15 #few
covid16
covid17 #few
covid18
covid20

# store mitochondrial percentage in object meta data
covid1 <- PercentageFeatureSet(covid1, pattern = "^MT-", col.name = "percent.mt")
covid2 <- PercentageFeatureSet(covid2, pattern = "^MT-", col.name = "percent.mt")
covid9 <- PercentageFeatureSet(covid9, pattern = "^MT-", col.name = "percent.mt")
covid10 <- PercentageFeatureSet(covid10, pattern = "^MT-", col.name = "percent.mt")
covid11 <- PercentageFeatureSet(covid11, pattern = "^MT-", col.name = "percent.mt")
covid12 <- PercentageFeatureSet(covid12, pattern = "^MT-", col.name = "percent.mt")
covid15 <- PercentageFeatureSet(covid15, pattern = "^MT-", col.name = "percent.mt")
covid16 <- PercentageFeatureSet(covid16, pattern = "^MT-", col.name = "percent.mt")
covid17 <- PercentageFeatureSet(covid17, pattern = "^MT-", col.name = "percent.mt")
covid18 <- PercentageFeatureSet(covid18, pattern = "^MT-", col.name = "percent.mt")
covid20 <- PercentageFeatureSet(covid20, pattern = "^MT-", col.name = "percent.mt")

# run sctransform
covid1 <- SCTransform(covid1, vars.to.regress = "percent.mt", verbose = FALSE)
covid2 <- SCTransform(covid2, vars.to.regress = "percent.mt", verbose = FALSE)
covid9 <- SCTransform(covid9, vars.to.regress = "percent.mt", verbose = FALSE)
covid10 <- SCTransform(covid10, vars.to.regress = "percent.mt", verbose = FALSE)
covid11 <- SCTransform(covid11, vars.to.regress = "percent.mt", verbose = FALSE)
covid12 <- SCTransform(covid12, vars.to.regress = "percent.mt", verbose = FALSE)
covid15 <- SCTransform(covid15, vars.to.regress = "percent.mt", verbose = FALSE)
covid16 <- SCTransform(covid16, vars.to.regress = "percent.mt", verbose = FALSE)
covid17 <- SCTransform(covid17, vars.to.regress = "percent.mt", verbose = FALSE)
covid18 <- SCTransform(covid18, vars.to.regress = "percent.mt", verbose = FALSE)
covid20 <- SCTransform(covid20, vars.to.regress = "percent.mt", verbose = FALSE)


##covid1 analysis
###linear dimensional reduction
covid1<- RunPCA(covid1, features = VariableFeatures(object = covid1)) 
DimPlot(covid1, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(covid1, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid1[["pca"]]@stdev / sum(covid1[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2
pcs <- min(co1, co2)
pcs

###clustering
covid1 <- FindNeighbors(covid1, dims = 1:15)
covid1 <- FindClusters(covid1, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid1 <- RunUMAP(covid1, dims = 1:15)
DimPlot(covid1, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid1.markers <- FindAllMarkers(covid1, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid1.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(covid1, features = c("FCGR3A", "MS4A7"))
new.cluster.ids <- c("0","1","2","CD16 monocytes","4","5","6","7","8","9","10","11") 
names(new.cluster.ids) <- levels(covid1)
covid1 <- RenameIdents(covid1, new.cluster.ids)
DimPlot(covid1, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
table(Idents(covid1))
covid1monocytes <- subset(covid1, idents = c("CD16 monocytes"))
saveRDS(object=covid1monocytes, file = "~/Desktop/Covid analysis/VC start/Single cell covid 091420/covid1monocytes.rds")
rm(covid1,covid1monocytes,covid1.markers)

##covid2 analysis
###linear dimensional reduction
covid2<- RunPCA(covid2, features = VariableFeatures(object = covid2)) 
DimPlot(covid2, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(covid2, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid2[["pca"]]@stdev / sum(covid2[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2
pcs <- min(co1, co2)
pcs

###clustering
covid2 <- FindNeighbors(covid2, dims = 1:15)
covid2<- FindClusters(covid2, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid2 <- RunUMAP(covid2, dims = 1:15)
DimPlot(covid2, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid2.markers <- FindAllMarkers(covid2, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid2.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(covid2, features = c("FCGR3A", "MS4A7"))
new.cluster.ids <- c("0","1","2","3","4","5","6","7","8","9","10","11", "12","CD16 monocytes") 
names(new.cluster.ids) <- levels(covid2)
covid2 <- RenameIdents(covid2, new.cluster.ids)
DimPlot(covid2, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
table(Idents(covid2)) ###very few
covid2monocytes <- subset(covid2, idents = c("CD16 monocytes"))
saveRDS(object=covid2monocytes, file = "~/Desktop/Covid analysis/VC start/Single cell covid 091420/covid2monocytes.rds")
rm(covid2,covid2monocytes,covid2.markers)


##covid9 analysis
###linear dimensional reduction
covid9<- RunPCA(covid9, features = VariableFeatures(object = covid9)) 
DimPlot(covid9, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(covid9, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid9[["pca"]]@stdev / sum(covid9[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2
pcs <- min(co1, co2)
pcs

###clustering
covid9 <- FindNeighbors(covid9, dims = 1:12)
covid9<- FindClusters(covid9, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid9 <- RunUMAP(covid9, dims = 1:12)
DimPlot(covid9, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid9.markers <- FindAllMarkers(covid9, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid9.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(covid9, features = c("FCGR3A", "MS4A7","FCGR3B"))
#no mature monocytes
rm(covid9,covid9monocytes,covid9.markers)


##covid10 analysis
###linear dimensional reduction
covid10<- RunPCA(covid10, features = VariableFeatures(object = covid10)) 
DimPlot(covid10, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(covid10, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid10[["pca"]]@stdev / sum(covid10[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2
pcs <- min(co1, co2)
pcs

###clustering
covid10 <- FindNeighbors(covid10, dims = 1:11)
covid10 <- FindClusters(covid10, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid10 <- RunUMAP(covid10, dims = 1:11)
DimPlot(covid10, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid10.markers <- FindAllMarkers(covid10, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid10.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(covid10, features = c("FCGR3A", "MS4A7","FCGR3B","CXCR2"))
new.cluster.ids <- c("0","1","2","3","4","CD16 monocytes","6","7") 
names(new.cluster.ids) <- levels(covid10)
covid10 <- RenameIdents(covid10, new.cluster.ids)
DimPlot(covid10, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
table(Idents(covid10)) #few cells
covid10monocytes <- subset(covid10, idents = c("CD16 monocytes"))
saveRDS(object=covid10monocytes, file = "~/Desktop/Covid analysis/VC start/Single cell covid 091420/covid10monocytes.rds")
rm(covid10,covid10monocytes,covid10.markers)

##covid11 analysis
###linear dimensional reduction
covid11<- RunPCA(covid11, features = VariableFeatures(object = covid11)) 
DimPlot(covid11, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(covid11, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid11[["pca"]]@stdev / sum(covid11[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2
pcs <- min(co1, co2)
pcs

###clustering
covid11 <- FindNeighbors(covid11, dims = 1:14)
covid11 <- FindClusters(covid11, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid11 <- RunUMAP(covid11, dims = 1:14)
DimPlot(covid11, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid11.markers <- FindAllMarkers(covid11, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid11.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(covid11, features = c("FCGR3A", "MS4A7"))
new.cluster.ids <- c("0","1","2","3","4","5","6","CD16 monocytes","8","9","10","11") 
names(new.cluster.ids) <- levels(covid11)
covid11 <- RenameIdents(covid11, new.cluster.ids)
DimPlot(covid11, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
table(Idents(covid11))
covid11monocytes <- subset(covid11, idents = c("CD16 monocytes"))
saveRDS(object=covid11monocytes, file = "~/Desktop/Covid analysis/VC start/Single cell covid 091420/covid11monocytes.rds")
rm(covid11,covid11monocytes,covid11.markers)

##covid12 analysis
###linear dimensional reduction
covid12<- RunPCA(covid12, features = VariableFeatures(object = covid12)) 
DimPlot(covid12, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(covid12, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid12[["pca"]]@stdev / sum(covid12[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2
pcs <- min(co1, co2)
pcs

###clustering
covid12 <- FindNeighbors(covid12, dims = 1:15)
covid12 <- FindClusters(covid12, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid12 <- RunUMAP(covid12, dims = 1:15)
DimPlot(covid12, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid12.markers <- FindAllMarkers(covid12, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid12.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(covid12, features = c("FCGR3A", "MS4A7"))
new.cluster.ids <- c("0","1","2","3","4","5","6","7","8","9" ,"CD16 monocytes","11","12") 
names(new.cluster.ids) <- levels(covid12)
covid12 <- RenameIdents(covid12, new.cluster.ids)
DimPlot(covid12, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
table(Idents(covid12)) 
covid12monocytes <- subset(covid12, idents = c("CD16 monocytes"))
saveRDS(object=covid12monocytes, file = "~/Desktop/Covid analysis/VC start/Single cell covid 091420/covid12monocytes.rds")
rm(covid12,covid12monocytes,covid12.markers)

##covid15 analysis
###linear dimensional reduction
covid15<- RunPCA(covid15, features = VariableFeatures(object = covid15)) 
DimPlot(covid15, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(covid15, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid15[["pca"]]@stdev / sum(covid15[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2
pcs <- min(co1, co2)
pcs

###clustering
covid15 <- FindNeighbors(covid15, dims = 1:13)
covid15 <- FindClusters(covid15, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid15 <- RunUMAP(covid15, dims = 1:13)
DimPlot(covid15, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid15.markers <- FindAllMarkers(covid15, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid15.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(covid15, features = c("FCGR3A", "MS4A7"))
##no cd16 monocytes
new.cluster.ids <- c("Platelets","1","CD14 monocytes","CD14 monocytes","CD4 T cells","CD8 T cells","B cells","CD4 T cells") 
names(new.cluster.ids) <- levels(covid15)
covid15 <- RenameIdents(covid15, new.cluster.ids)
DimPlot(covid15, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
table(Idents(covid15)) 
rm(covid15,covid15monocytes,covid15.markers)

##covid16 analysis
###linear dimensional reduction
covid16<- RunPCA(covid16, features = VariableFeatures(object = covid16)) 
DimPlot(covid16, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(covid16, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid16[["pca"]]@stdev / sum(covid16[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2
pcs <- min(co1, co2)
pcs

###clustering
covid16 <- FindNeighbors(covid16, dims = 1:15)
covid16 <- FindClusters(covid16, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid16 <- RunUMAP(covid16, dims = 1:15)
DimPlot(covid16, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid16.markers <- FindAllMarkers(covid16, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid16.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(covid16, features = c("FCGR3A", "MS4A7"))
###no cd16 monocytes
rm(covid16,covid16monocytes,covid16.markers)


##covid17 analysis
###linear dimensional reduction
covid17<- RunPCA(covid17, features = VariableFeatures(object = covid17)) 
DimPlot(covid17, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(covid17, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid17[["pca"]]@stdev / sum(covid17[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2
pcs <- min(co1, co2)
pcs

###clustering
covid17 <- FindNeighbors(covid17, dims = 1:13)
covid17 <- FindClusters(covid17, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid17 <- RunUMAP(covid17, dims = 1:13)
DimPlot(covid17, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid17.markers <- FindAllMarkers(covid17, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid17.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(covid17, features = c("FCGR3A", "MS4A7"))
new.cluster.ids <- c("0","1","2","3","4","5","6") 
names(new.cluster.ids) <- levels(covid17)
covid17 <- RenameIdents(covid17, new.cluster.ids)
DimPlot(covid17, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
table(Idents(covid17)) #few cells
#no monocytes
rm(covid17,covid17monocytes,covid17.markers)

##covid18 analysis
###linear dimensional reduction
covid18<- RunPCA(covid18, features = VariableFeatures(object = covid18)) 
DimPlot(covid18, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(covid18, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid18[["pca"]]@stdev / sum(covid18[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2
pcs <- min(co1, co2)
pcs

###clustering
covid18 <- FindNeighbors(covid18, dims = 1:19)
covid18 <- FindClusters(covid18, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid18 <- RunUMAP(covid18, dims = 1:19)
DimPlot(covid18, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid18.markers <- FindAllMarkers(covid18, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid18.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(covid18, features = c("FCGR3A", "MS4A7"))
new.cluster.ids <- c("0","CD16 monocytes","2","3","4","5","6","7","8","9","10","11","12","13","14") 
names(new.cluster.ids) <- levels(covid18)
covid18 <- RenameIdents(covid18, new.cluster.ids)
DimPlot(covid18, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
table(Idents(covid18)) 
covid18monocytes <- subset(covid18, idents = c("CD16 monocytes"))
saveRDS(object=covid18monocytes, file = "~/Desktop/Covid analysis/VC start/Single cell covid 091420/covid18monocytes.rds")
rm(covid18,covid18monocytes,covid18.markers)

##covid20 analysis
###linear dimensional reduction
covid20<- RunPCA(covid20, features = VariableFeatures(object = covid20)) 
DimPlot(covid20, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(covid20, ndims = 100)
# Determine percent of variation associated with each PC
pct <- covid20[["pca"]]@stdev / sum(covid20[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
co2
pcs <- min(co1, co2)
pcs

###clustering
covid20 <- FindNeighbors(covid20, dims = 1:14)
covid20 <- FindClusters(covid20, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
covid20 <- RunUMAP(covid20, dims = 1:14)
DimPlot(covid20, reduction = "umap", min.dist = 0.75, pt.size = 0.2,label=TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
covid20.markers <- FindAllMarkers(covid20, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
covid20.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(covid20, features = c("FCGR3A", "MS4A7"))
new.cluster.ids <- c("0","1","CD16 monocytes","3","4","5","6","7","8","9","10") 
names(new.cluster.ids) <- levels(covid20)
covid20 <- RenameIdents(covid20, new.cluster.ids)
DimPlot(covid20, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
table(Idents(covid20))
covid20monocytes <- subset(covid20, idents = c("CD16 monocytes"))
saveRDS(object=covid20monocytes, file = "~/Desktop/Covid analysis/VC start/Single cell covid 091420/covid20monocytes.rds")
rm(covid20,covid20monocytes,covid20.markers)

data <- readRDS(file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/seuratCOVID19PBMCcohort2.rds")

##this is already a seurat object, delete data that we are not interested in (NULL)
data
data[["HTO"]]<- NULL
data
colnames(data@meta.data)                      
data@meta.data$donor
data@meta.data$group_per_sample

Idents(data) <- "donor"
table(Idents(data))

###subset each donor to an independent seurat object
####healthy donors from Reyes et al., Nat Med, 2020
BN_26 <- subset(data, idents = c("BN-26"))
BN_27 <- subset(data, idents = c("BN-27"))
BN_28 <- subset(data, idents = c("BN-28"))
BN_29 <- subset(data, idents = c("BN-29"))
BN_30 <- subset(data, idents = c("BN-30"))
BN_31 <- subset(data, idents = c("BN-31"))
BN_32  <- subset(data, idents = c("BN-32"))
BN_33 <- subset(data, idents = c("BN-33"))
BN_34 <- subset(data, idents = c("BN-34"))
BN_35 <- subset(data, idents = c("BN-35"))
BN_36 <- subset(data, idents = c("BN-36"))
BN_37 <- subset(data, idents = c("BN-37"))

###covid patients from Bonn (cohort 2)
BN_01 <- subset(data, idents = c("BN-01"))
BN_02 <- subset(data, idents = c("BN-02"))
BN_03 <- subset(data, idents = c("BN-03"))
BN_04 <- subset(data, idents = c("BN-04"))
BN_05 <- subset(data, idents = c("BN-05"))
BN_06 <- subset(data, idents = c("BN-06"))
BN_07 <- subset(data, idents = c("BN-07"))
BN_08 <- subset(data, idents = c("BN-08"))
BN_10 <- subset(data, idents = c("BN-10"))
BN_11 <- subset(data, idents = c("BN-11"))
BN_12 <- subset(data, idents = c("BN-12"))
BN_13 <- subset(data, idents = c("BN-13"))
BN_14 <- subset(data, idents = c("BN-14"))
BN_15 <- subset(data, idents = c("BN-15"))
BN_16 <- subset(data, idents = c("BN-16"))
BN_17 <- subset(data, idents = c("BN-17"))
BN_19 <- subset(data, idents = c("BN-19"))

rm(data)

##extract monocytes from healthy donors first

####BN-26
BN_26
Idents(object = BN_26) <- "Control 20"
BN_26$sample <- "Control 20"
BN_26$stim <- "Healthy"
levels(BN_26)
BN_26
BN_26[["percent.mt"]] <- PercentageFeatureSet(BN_26, pattern = "^MT-")
VlnPlot(BN_26, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_26<- subset(BN_26, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_26
BN_26 <- PercentageFeatureSet(BN_26, pattern = "^MT-", col.name = "percent.mt")
BN_26 <- SCTransform(BN_26, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_26) <- "sample"
levels(BN_26)
BN_26<- RunPCA(BN_26, features = VariableFeatures(object = BN_26))
DimPlot(BN_26, reduction = "pca")
ElbowPlot(BN_26, ndims = 100)
pct <- BN_26[["pca"]]@stdev / sum(BN_26[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_26 <- FindNeighbors(BN_26, dims = 1:12)
BN_26 <- FindClusters(BN_26, resolution = 0.5)
BN_26 <- RunUMAP(BN_26, dims = 1:12)
DimPlot(BN_26, reduction = "umap", pt.size =0.2, label = TRUE)
BN_26.markers <- FindAllMarkers(BN_22, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_26.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_26, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_26, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_26, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_26)
new.cluster.ids <- c("0", "1", "2", "3","4","5","6","CD16 monocytes" ,"8","9", "10", "11")
names(new.cluster.ids) <- levels(BN_26)
BN_26 <- RenameIdents(BN_26, new.cluster.ids)
levels(BN_26)
DimPlot(BN_26, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_26, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_26.rds")
table(Idents(BN_26)) 

#subset  and save CD16 monocytes (we have more than 50 cells (our cut-off)) 
BN_26_monocytes <- subset(BN_26, idents = c("CD16 monocytes"))
table(Idents(BN_26_monocytes))

Idents(object = BN_26_monocytes) <- "Control 20 monocytes"
BN_26_monocytes$sample <- "Control 20"
BN_26_monocytes$stim <- "Healthy"
levels(BN_26_monocytes)
BN_26_monocytes
saveRDS(BN_26_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_26_monocytes.rds")
rm(BN_26,BN_26_monocytes,BN_26.markers)

####BN-27
BN_27
Idents(object = BN_27) <- "Control 21"
BN_27$sample <- "Control 21"
BN_27$stim <- "Healthy"
levels(BN_27)
BN_27
BN_27[["percent.mt"]] <- PercentageFeatureSet(BN_27, pattern = "^MT-")
VlnPlot(BN_27, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_27<- subset(BN_27, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_27
BN_27 <- PercentageFeatureSet(BN_27, pattern = "^MT-", col.name = "percent.mt")
BN_27 <- SCTransform(BN_27, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_27) <- "sample"
levels(BN_27)
BN_27<- RunPCA(BN_27, features = VariableFeatures(object = BN_27))
DimPlot(BN_27, reduction = "pca")
ElbowPlot(BN_27, ndims = 100)
pct <- BN_27[["pca"]]@stdev / sum(BN_27[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_27 <- FindNeighbors(BN_27, dims = 1:9)
BN_27 <- FindClusters(BN_27, resolution = 0.5)
BN_27 <- RunUMAP(BN_27, dims = 1:9)
DimPlot(BN_27, reduction = "umap", pt.size =0.2, label = TRUE)
BN_27.markers <- FindAllMarkers(BN_27, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_27.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_27, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_27, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_27, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_27)
new.cluster.ids <- c("0", "1", "2", "3","4","5","CD16 monocytes","7")
names(new.cluster.ids) <- levels(BN_27)
BN_27 <- RenameIdents(BN_27, new.cluster.ids)
levels(BN_27)
DimPlot(BN_27, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_27, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_27.rds")
table(Idents(BN_27)) 

#few monocytes (43)

#subset  and save CD16 monocytes (we have less than 50 cells (our cut-off)) 
BN_27_monocytes <- subset(BN_27, idents = c("CD16 monocytes"))
table(Idents(BN_27_monocytes))

Idents(object = BN_27_monocytes) <- "Control 21 monocytes"
BN_27_monocytes$sample <- "Control 21"
BN_27_monocytes$stim <- "Healthy"
levels(BN_27_monocytes)
BN_27_monocytes
saveRDS(BN_27_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_27_monocytes.rds")
rm(BN_27,BN_27_monocytes,BN_27.markers)

####BN-28
BN_28
Idents(object = BN_28) <- "Control 22"
BN_28$sample <- "Control 22"
BN_28$stim <- "Healthy"
levels(BN_28)
BN_28
BN_28[["percent.mt"]] <- PercentageFeatureSet(BN_28, pattern = "^MT-")
VlnPlot(BN_28, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_28<- subset(BN_28, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_28
BN_28 <- PercentageFeatureSet(BN_28, pattern = "^MT-", col.name = "percent.mt")
BN_28 <- SCTransform(BN_28, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_28) <- "sample"
levels(BN_28)
BN_28<- RunPCA(BN_28, features = VariableFeatures(object = BN_28))
DimPlot(BN_28, reduction = "pca")
ElbowPlot(BN_28, ndims = 100)
pct <- BN_28[["pca"]]@stdev / sum(BN_28[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_28 <- FindNeighbors(BN_28, dims = 1:9)
BN_28 <- FindClusters(BN_28, resolution = 0.5)
BN_28 <- RunUMAP(BN_28, dims = 1:9)
DimPlot(BN_28, reduction = "umap", pt.size =0.2, label = TRUE)
BN_28.markers <- FindAllMarkers(BN_28, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_28.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_28, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_28, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_28, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_28)
new.cluster.ids <- c("0", "1", "2", "3","4","5","6","7","CD16 monocytes","9")
names(new.cluster.ids) <- levels(BN_28)
BN_28 <- RenameIdents(BN_28, new.cluster.ids)
levels(BN_28)
DimPlot(BN_28, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_28, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_28.rds")
table(Idents(BN_28)) 

#few monocytes (49)

#subset  and save CD16 monocytes (we have less than 50 cells (our cut-off)) 
BN_28_monocytes <- subset(BN_28, idents = c("CD16 monocytes"))
table(Idents(BN_28_monocytes))

Idents(object = BN_28_monocytes) <- "Control 22 monocytes"
BN_28_monocytes$sample <- "Control 22"
BN_28_monocytes$stim <- "Healthy"
levels(BN_28_monocytes)
BN_28_monocytes
saveRDS(BN_28_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_28_monocytes.rds")
rm(BN_28,BN_28_monocytes,BN_28.markers)

####BN-29
BN_29
Idents(object = BN_29) <- "Control 23"
BN_29$sample <- "Control 23"
BN_29$stim <- "Healthy"
levels(BN_29)
BN_29
BN_29[["percent.mt"]] <- PercentageFeatureSet(BN_29, pattern = "^MT-")
VlnPlot(BN_29, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_29<- subset(BN_29, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_29
BN_29 <- PercentageFeatureSet(BN_29, pattern = "^MT-", col.name = "percent.mt")
BN_29 <- SCTransform(BN_29, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_29) <- "sample"
levels(BN_29)
BN_29<- RunPCA(BN_29, features = VariableFeatures(object = BN_29))
DimPlot(BN_29, reduction = "pca")
ElbowPlot(BN_29, ndims = 100)
pct <- BN_29[["pca"]]@stdev / sum(BN_29[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_29 <- FindNeighbors(BN_29, dims = 1:9)
BN_29 <- FindClusters(BN_29, resolution = 0.5)
BN_29 <- RunUMAP(BN_29, dims = 1:9)
DimPlot(BN_29, reduction = "umap", pt.size =0.2, label = TRUE)
BN_29.markers <- FindAllMarkers(BN_29, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_29.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_29, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_29, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_29, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_29)
new.cluster.ids <- c("0", "1", "2", "3","4","5","6","CD16 monocytes","8","9")
names(new.cluster.ids) <- levels(BN_29)
BN_29 <- RenameIdents(BN_29, new.cluster.ids)
levels(BN_29)
DimPlot(BN_29, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_29, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_29.rds")
table(Idents(BN_29)) 

#subset  and save CD16 monocytes (we have less than 50 cells (our cut-off)) 
BN_29_monocytes <- subset(BN_29, idents = c("CD16 monocytes"))
table(Idents(BN_29_monocytes))

Idents(object = BN_29_monocytes) <- "Control 23 monocytes"
BN_29_monocytes$sample <- "Control 23"
BN_29_monocytes$stim <- "Healthy"
levels(BN_29_monocytes)
BN_29_monocytes
saveRDS(BN_29_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_29_monocytes.rds")
rm(BN_29,BN_29_monocytes,BN_29.markers)

####BN-30
BN_30
Idents(object = BN_30) <- "Control 24"
BN_30$sample <- "Control 24"
BN_30$stim <- "Healthy"
levels(BN_30)
BN_30
BN_30[["percent.mt"]] <- PercentageFeatureSet(BN_30, pattern = "^MT-")
VlnPlot(BN_30, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_30<- subset(BN_30, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_30
BN_30 <- PercentageFeatureSet(BN_30, pattern = "^MT-", col.name = "percent.mt")
BN_30 <- SCTransform(BN_30, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_30) <- "sample"
levels(BN_30)
BN_30<- RunPCA(BN_30, features = VariableFeatures(object = BN_30))
DimPlot(BN_30, reduction = "pca")
ElbowPlot(BN_30, ndims = 100)
pct <- BN_30[["pca"]]@stdev / sum(BN_30[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_30 <- FindNeighbors(BN_30, dims = 1:12)
BN_30 <- FindClusters(BN_30, resolution = 0.5)
BN_30 <- RunUMAP(BN_30, dims = 1:12)
DimPlot(BN_30, reduction = "umap", pt.size =0.2, label = TRUE)
BN_30.markers <- FindAllMarkers(BN_30, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_30.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_30, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_30, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_30, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_30)
new.cluster.ids <- c("0", "1", "2", "3","4","CD16 monocytes","6","7")
names(new.cluster.ids) <- levels(BN_30)
BN_30 <- RenameIdents(BN_30, new.cluster.ids)
levels(BN_30)
DimPlot(BN_30, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_30, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_30.rds")
table(Idents(BN_30)) 

#subset  and save CD16 monocytes 
BN_30_monocytes <- subset(BN_30, idents = c("CD16 monocytes"))
table(Idents(BN_30_monocytes))

Idents(object = BN_30_monocytes) <- "Control 24 monocytes"
BN_30_monocytes$sample <- "Control 24"
BN_30_monocytes$stim <- "Healthy"
levels(BN_30_monocytes)
BN_30_monocytes
saveRDS(BN_30_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_30_monocytes.rds")
rm(BN_30,BN_30_monocytes,BN_30.markers)

####BN-31
BN_31
Idents(object = BN_31) <- "Control 25"
BN_31$sample <- "Control 25"
BN_31$stim <- "Healthy"
levels(BN_31)
BN_31
BN_31[["percent.mt"]] <- PercentageFeatureSet(BN_31, pattern = "^MT-")
VlnPlot(BN_31, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_31<- subset(BN_31, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_31
BN_31 <- PercentageFeatureSet(BN_31, pattern = "^MT-", col.name = "percent.mt")
BN_31 <- SCTransform(BN_31, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_31) <- "sample"
levels(BN_31)
BN_31<- RunPCA(BN_31, features = VariableFeatures(object = BN_31))
DimPlot(BN_31, reduction = "pca")
ElbowPlot(BN_31, ndims = 100)
pct <- BN_31[["pca"]]@stdev / sum(BN_31[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_31 <- FindNeighbors(BN_31, dims = 1:10)
BN_31 <- FindClusters(BN_31, resolution = 0.5)
BN_31 <- RunUMAP(BN_31, dims = 1:10)
DimPlot(BN_31, reduction = "umap", pt.size =0.2, label = TRUE)
BN_31.markers <- FindAllMarkers(BN_31, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_31.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_31, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_31, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_31, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_31)
new.cluster.ids <- c("0", "1", "2", "3","4","5","6","7", "CD16 monocytes", "9", "10")
names(new.cluster.ids) <- levels(BN_31)
BN_31 <- RenameIdents(BN_31, new.cluster.ids)
levels(BN_31)
DimPlot(BN_31, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_31, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_31.rds")
table(Idents(BN_31)) 

#cluster 8 monocytes seem to be intermediate 

#subset  and save CD16 monocytes 
BN_31_monocytes <- subset(BN_31, idents = c("CD16 monocytes"))
table(Idents(BN_31_monocytes))

Idents(object = BN_31_monocytes) <- "Control 25 monocytes"
BN_31_monocytes$sample <- "Control 25"
BN_31_monocytes$stim <- "Healthy"
levels(BN_31_monocytes)
BN_31_monocytes
saveRDS(BN_31_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_31_monocytes.rds")
rm(BN_31,BN_31_monocytes,BN_31.markers)

####BN-32
BN_32
Idents(object = BN_32) <- "Control 26"
BN_32$sample <- "Control 26"
BN_32$stim <- "Healthy"
levels(BN_32)
BN_32
BN_32[["percent.mt"]] <- PercentageFeatureSet(BN_32, pattern = "^MT-")
VlnPlot(BN_32, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_32<- subset(BN_32, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_32
BN_32 <- PercentageFeatureSet(BN_32, pattern = "^MT-", col.name = "percent.mt")
BN_32 <- SCTransform(BN_32, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_32) <- "sample"
levels(BN_32)
BN_32<- RunPCA(BN_32, features = VariableFeatures(object = BN_32))
DimPlot(BN_32, reduction = "pca")
ElbowPlot(BN_32, ndims = 100)
pct <- BN_32[["pca"]]@stdev / sum(BN_32[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_32 <- FindNeighbors(BN_32, dims = 1:5)
BN_32 <- FindClusters(BN_32, resolution = 0.5)
BN_32 <- RunUMAP(BN_32, dims = 1:5)
DimPlot(BN_32, reduction = "umap", pt.size =0.2, label = TRUE)
BN_32.markers <- FindAllMarkers(BN_32, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_32.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_32, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_32, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_32, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_32)
new.cluster.ids <- c("0", "1", "CD16 monocytes")
names(new.cluster.ids) <- levels(BN_32)
BN_32 <- RenameIdents(BN_32, new.cluster.ids)
levels(BN_32)
DimPlot(BN_32, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_32, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_32.rds")
table(Idents(BN_32)) 

#very few cd16 monocytes do not include this donor

#subset  and save CD16 monocytes 
BN_32_monocytes <- subset(BN_32, idents = c("CD16 monocytes"))
table(Idents(BN_32_monocytes))

Idents(object = BN_32_monocytes) <- "Control 26 monocytes"
BN_32_monocytes$sample <- "Control 26"
BN_32_monocytes$stim <- "Healthy"
levels(BN_32_monocytes)
BN_32_monocytes
saveRDS(BN_32_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_32_monocytes.rds")
rm(BN_32,BN_32_monocytes,BN_32.markers)

####BN-33
BN_33
Idents(object = BN_33) <- "Control 27"
BN_33$sample <- "Control 27"
BN_33$stim <- "Healthy"
levels(BN_33)
BN_33
BN_33[["percent.mt"]] <- PercentageFeatureSet(BN_33, pattern = "^MT-")
VlnPlot(BN_33, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_33<- subset(BN_33, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_33
BN_33 <- PercentageFeatureSet(BN_33, pattern = "^MT-", col.name = "percent.mt")
BN_33 <- SCTransform(BN_33, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_33) <- "sample"
levels(BN_33)
BN_33<- RunPCA(BN_33, features = VariableFeatures(object = BN_33))
DimPlot(BN_33, reduction = "pca")
ElbowPlot(BN_33, ndims = 100)
pct <- BN_33[["pca"]]@stdev / sum(BN_33[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_33 <- FindNeighbors(BN_33, dims = 1:7)
BN_33 <- FindClusters(BN_33, resolution = 0.5)
BN_33 <- RunUMAP(BN_33, dims = 1:7)
DimPlot(BN_33, reduction = "umap", pt.size =0.2, label = TRUE)
BN_33.markers <- FindAllMarkers(BN_33, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_33.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_33, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_33, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_33, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_33)
new.cluster.ids <- c("0", "1", "2", "3", "4", "CD16 monocytes", "6", "CD16monocytes", "8")
names(new.cluster.ids) <- levels(BN_33)
BN_33 <- RenameIdents(BN_33, new.cluster.ids)
levels(BN_33)
DimPlot(BN_33, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_33, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_33.rds")
table(Idents(BN_33)) 

#subset  and save CD16 monocytes 
BN_33_monocytes <- subset(BN_33, idents = c("CD16 monocytes", "CD16monocytes"))
table(Idents(BN_33_monocytes))

Idents(object = BN_33_monocytes) <- "Control 27 monocytes"
BN_33_monocytes$sample <- "Control 27"
BN_33_monocytes$stim <- "Healthy"
levels(BN_33_monocytes)
BN_33_monocytes
saveRDS(BN_33_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_33_monocytes.rds")
rm(BN_33,BN_33_monocytes,BN_33.markers)


####BN-34
BN_34
Idents(object = BN_34) <- "Control 28"
BN_34$sample <- "Control 28"
BN_34$stim <- "Healthy"
levels(BN_34)
BN_34
BN_34[["percent.mt"]] <- PercentageFeatureSet(BN_34, pattern = "^MT-")
VlnPlot(BN_34, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_34<- subset(BN_34, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_34
BN_34 <- PercentageFeatureSet(BN_34, pattern = "^MT-", col.name = "percent.mt")
BN_34 <- SCTransform(BN_34, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_34) <- "sample"
levels(BN_34)
BN_34<- RunPCA(BN_34, features = VariableFeatures(object = BN_34))
DimPlot(BN_34, reduction = "pca")
ElbowPlot(BN_34, ndims = 100)
pct <- BN_34[["pca"]]@stdev / sum(BN_34[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_34 <- FindNeighbors(BN_34, dims = 1:14)
BN_34 <- FindClusters(BN_34, resolution = 0.5)
BN_34 <- RunUMAP(BN_34, dims = 1:14)
DimPlot(BN_34, reduction = "umap", pt.size =0.2, label = TRUE)
BN_34.markers <- FindAllMarkers(BN_34, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_34.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_34, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_34, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_34, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_34)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5", "6", "7", "8", "CD16 monocytes", "10", "11")
names(new.cluster.ids) <- levels(BN_34)
BN_34 <- RenameIdents(BN_34, new.cluster.ids)
levels(BN_34)
DimPlot(BN_34, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_34, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_34.rds")
table(Idents(BN_34)) 

#subset  and save CD16 monocytes 
BN_34_monocytes <- subset(BN_34, idents = c("CD16 monocytes"))
table(Idents(BN_34_monocytes))

Idents(object = BN_34_monocytes) <- "Control 28 monocytes"
BN_34_monocytes$sample <- "Control 28"
BN_34_monocytes$stim <- "Healthy"
levels(BN_34_monocytes)
BN_34_monocytes
saveRDS(BN_34_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_34_monocytes.rds")
rm(BN_34,BN_34_monocytes,BN_34.markers)

####BN-35
BN_35
Idents(object = BN_35) <- "Control 29"
BN_35$sample <- "Control 29"
BN_35$stim <- "Healthy"
levels(BN_35)
BN_35
BN_35[["percent.mt"]] <- PercentageFeatureSet(BN_35, pattern = "^MT-")
VlnPlot(BN_35, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_35<- subset(BN_35, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_35
BN_35 <- PercentageFeatureSet(BN_35, pattern = "^MT-", col.name = "percent.mt")
BN_35 <- SCTransform(BN_35, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_35) <- "sample"
levels(BN_35)
BN_35<- RunPCA(BN_35, features = VariableFeatures(object = BN_35))
DimPlot(BN_35, reduction = "pca")
ElbowPlot(BN_35, ndims = 100)
pct <- BN_35[["pca"]]@stdev / sum(BN_35[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_35 <- FindNeighbors(BN_35, dims = 1:14)
BN_35 <- FindClusters(BN_35, resolution = 0.5)
BN_35 <- RunUMAP(BN_35, dims = 1:14)
DimPlot(BN_35, reduction = "umap", pt.size =0.2, label = TRUE)
BN_35.markers <- FindAllMarkers(BN_35, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_35.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_35, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_35, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_35, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_35)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5", "6", "CD16 monocytes", "8", "9", "10")
names(new.cluster.ids) <- levels(BN_35)
BN_35 <- RenameIdents(BN_35, new.cluster.ids)
levels(BN_35)
DimPlot(BN_35, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_35, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_35.rds")
table(Idents(BN_35)) 

#subset  and save CD16 monocytes 
BN_35_monocytes <- subset(BN_35, idents = c("CD16 monocytes"))
table(Idents(BN_35_monocytes))

Idents(object = BN_35_monocytes) <- "Control 29 monocytes"
BN_35_monocytes$sample <- "Control 29"
BN_35_monocytes$stim <- "Healthy"
levels(BN_35_monocytes)
BN_35_monocytes
saveRDS(BN_35_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_35_monocytes.rds")
rm(BN_35,BN_35_monocytes,BN_35.markers)

####BN-36
BN_36
Idents(object = BN_36) <- "Control 30"
BN_36$sample <- "Control 30"
BN_36$stim <- "Healthy"
levels(BN_36)
BN_36
BN_36[["percent.mt"]] <- PercentageFeatureSet(BN_36, pattern = "^MT-")
VlnPlot(BN_36, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_36<- subset(BN_36, subset = nFeature_RNA > 200 & nFeature_RNA < 2000 & percent.mt < 25)
BN_36
BN_36 <- PercentageFeatureSet(BN_36, pattern = "^MT-", col.name = "percent.mt")
BN_36 <- SCTransform(BN_36, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_36) <- "sample"
levels(BN_36)
BN_36<- RunPCA(BN_36, features = VariableFeatures(object = BN_36))
DimPlot(BN_36, reduction = "pca")
ElbowPlot(BN_36, ndims = 100)
pct <- BN_36[["pca"]]@stdev / sum(BN_36[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_36 <- FindNeighbors(BN_36, dims = 1:8)
BN_36 <- FindClusters(BN_36, resolution = 0.5)
BN_36 <- RunUMAP(BN_36, dims = 1:8)
DimPlot(BN_36, reduction = "umap", pt.size =0.2, label = TRUE)
BN_36.markers <- FindAllMarkers(BN_36, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_36.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_36, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_36, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_36, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_36)
new.cluster.ids <- c("0", "1", "2", "3", "CD16 monocytes", "5")
names(new.cluster.ids) <- levels(BN_36)
BN_36 <- RenameIdents(BN_36, new.cluster.ids)
levels(BN_36)
DimPlot(BN_36, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_36, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_36.rds")
table(Idents(BN_36)) 

#subset  and save CD16 monocytes 
BN_36_monocytes <- subset(BN_36, idents = c("CD16 monocytes"))
table(Idents(BN_36_monocytes))

#Do not use this donor very few cells

Idents(object = BN_36_monocytes) <- "Control 30 monocytes"
BN_36_monocytes$sample <- "Control 30"
BN_36_monocytes$stim <- "Healthy"
levels(BN_36_monocytes)
BN_36_monocytes
saveRDS(BN_36_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_36_monocytes.rds")
rm(BN_36,BN_36_monocytes,BN_36.markers)


####BN-37
BN_37
Idents(object = BN_37) <- "Control 31"
BN_37$sample <- "Control 31"
BN_37$stim <- "Healthy"
levels(BN_37)
BN_37
BN_37[["percent.mt"]] <- PercentageFeatureSet(BN_37, pattern = "^MT-")
VlnPlot(BN_37, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_37<- subset(BN_37, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_37
BN_37 <- PercentageFeatureSet(BN_37, pattern = "^MT-", col.name = "percent.mt")
BN_37 <- SCTransform(BN_37, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_37) <- "sample"
levels(BN_37)
BN_37<- RunPCA(BN_37, features = VariableFeatures(object = BN_37))
DimPlot(BN_37, reduction = "pca")
ElbowPlot(BN_37, ndims = 100)
pct <- BN_37[["pca"]]@stdev / sum(BN_37[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_37 <- FindNeighbors(BN_37, dims = 1:11)
BN_37 <- FindClusters(BN_37, resolution = 0.5)
BN_37 <- RunUMAP(BN_37, dims = 1:11)
DimPlot(BN_37, reduction = "umap", pt.size =0.2, label = TRUE)
BN_37.markers <- FindAllMarkers(BN_37, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_37.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_37, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_37, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_37, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_37)

#no CD16 monocytes

new.cluster.ids <- c("0", "1", "2", "3", "CD16 monocytes", "5", "6", "7")
names(new.cluster.ids) <- levels(BN_37)
BN_37 <- RenameIdents(BN_37, new.cluster.ids)
levels(BN_37)
DimPlot(BN_37, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_37, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_37.rds")
table(Idents(BN_37)) 

#subset  and save CD16 monocytes 
BN_37_monocytes <- subset(BN_37, idents = c("CD16 monocytes"))
table(Idents(BN_37_monocytes))

#Do not use this donor very few cells

Idents(object = BN_37_monocytes) <- "Control 31 monocytes"
BN_37_monocytes$sample <- "Control 31"
BN_37_monocytes$stim <- "Healthy"
levels(BN_37_monocytes)
BN_37_monocytes
saveRDS(BN_37_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_37_monocytes.rds")
rm(BN_37,BN_37_monocytes,BN_37.markers)


#####start extracting from covid samples
####BN-01
BN_01
Idents(object = BN_01) <- "Cov2 Sample 19c"
BN_01$sample <- "Cov2 Sample 19c"
BN_01$stim <- "Cov2"
levels(BN_01)
BN_01
BN_01[["percent.mt"]] <- PercentageFeatureSet(BN_01, pattern = "^MT-")
VlnPlot(BN_01, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_01<- subset(BN_01, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_01
BN_01 <- PercentageFeatureSet(BN_01, pattern = "^MT-", col.name = "percent.mt")
BN_01 <- SCTransform(BN_01, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_01) <- "sample"
levels(BN_01)
BN_01<- RunPCA(BN_01, features = VariableFeatures(object = BN_01))
DimPlot(BN_01, reduction = "pca")
ElbowPlot(BN_01, ndims = 100)
pct <- BN_01[["pca"]]@stdev / sum(BN_01[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_01 <- FindNeighbors(BN_01, dims = 1:15)
BN_01 <- FindClusters(BN_01, resolution = 0.5)
BN_01 <- RunUMAP(BN_01, dims = 1:15)
DimPlot(BN_01, reduction = "umap", pt.size =0.2, label = TRUE)
BN_01.markers <- FindAllMarkers(BN_01, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_01.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_01, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_01, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_01, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_01)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_01)
BN_01 <- RenameIdents(BN_01, new.cluster.ids)
levels(BN_01)
DimPlot(BN_01, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_01, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_01.rds")
table(Idents(BN_01)) 


#subset  and save CD16 monocytes 
BN_01_monocytes <- subset(BN_01, idents = c("CD16 monocytes"))
table(Idents(BN_01_monocytes))

Idents(object = BN_01_monocytes) <- "Cov2 Sample 19c"
BN_01_monocytes$sample <- "Cov2 Sample 19c"
BN_01_monocytes$stim <- "Cov2"
levels(BN_01_monocytes)
BN_01_monocytes
saveRDS(BN_01_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_01_monocytes.rds")
rm(BN_01,BN_01_monocytes,BN_01.markers)


####BN-02
BN_02
Idents(object = BN_02) <- "Cov2 Sample 20c"
BN_02$sample <- "Cov2 Sample 20c"
BN_02$stim <- "Cov2"
levels(BN_02)
BN_02
BN_02[["percent.mt"]] <- PercentageFeatureSet(BN_02, pattern = "^MT-")
VlnPlot(BN_02, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_02<- subset(BN_02, subset = nFeature_RNA > 200 & nFeature_RNA < 2200 & percent.mt < 25)
BN_02
BN_02 <- PercentageFeatureSet(BN_02, pattern = "^MT-", col.name = "percent.mt")
BN_02 <- SCTransform(BN_02, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_02) <- "sample"
levels(BN_02)
BN_02<- RunPCA(BN_02, features = VariableFeatures(object = BN_02))
DimPlot(BN_02, reduction = "pca")
ElbowPlot(BN_02, ndims = 100)
pct <- BN_02[["pca"]]@stdev / sum(BN_02[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_02 <- FindNeighbors(BN_02, dims = 1:11)
BN_02 <- FindClusters(BN_02, resolution = 0.5)
BN_02 <- RunUMAP(BN_02, dims = 1:11)
DimPlot(BN_02, reduction = "umap", pt.size =0.2, label = TRUE)
BN_02.markers <- FindAllMarkers(BN_02, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_02.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_02, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_02, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_02, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_02)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_02)
BN_02 <- RenameIdents(BN_02, new.cluster.ids)
levels(BN_02)
DimPlot(BN_02, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#no CD16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_02, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_02.rds")
table(Idents(BN_02)) 

#subset  and save CD16 monocytes 
BN_02_monocytes <- subset(BN_02, idents = c("CD16 monocytes"))
table(Idents(BN_02_monocytes))

Idents(object = BN_02_monocytes) <- "Cov2 Sample 20c"
BN_02_monocytes$sample <- "Cov2 Sample 20c"
BN_02_monocytes$stim <- "Cov2"
levels(BN_02_monocytes)
BN_02_monocytes
saveRDS(BN_02_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_02_monocytes.rds")
rm(BN_02,BN_02_monocytes,BN_02.markers)

####BN-03
BN_03
Idents(object = BN_03) <- "Cov2 Sample 21c"
BN_03$sample <- "Cov2 Sample 21c"
BN_03$stim <- "Cov2"
levels(BN_03)
BN_03
BN_03[["percent.mt"]] <- PercentageFeatureSet(BN_03, pattern = "^MT-")
VlnPlot(BN_03, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_03<- subset(BN_03, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_03
BN_03 <- PercentageFeatureSet(BN_03, pattern = "^MT-", col.name = "percent.mt")
BN_03 <- SCTransform(BN_03, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_03) <- "sample"
levels(BN_03)
BN_03<- RunPCA(BN_03, features = VariableFeatures(object = BN_03))
DimPlot(BN_03, reduction = "pca")
ElbowPlot(BN_03, ndims = 100)
pct <- BN_03[["pca"]]@stdev / sum(BN_03[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_03 <- FindNeighbors(BN_03, dims = 1:14)
BN_03 <- FindClusters(BN_03, resolution = 0.5)
BN_03 <- RunUMAP(BN_03, dims = 1:14)
DimPlot(BN_03, reduction = "umap", pt.size =0.2, label = TRUE)
BN_03.markers <- FindAllMarkers(BN_03, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_03.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_03, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_03, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_03, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_03)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_03)
BN_03 <- RenameIdents(BN_03, new.cluster.ids)
levels(BN_03)
DimPlot(BN_03, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#no cd16

##save before extracting CD16 monocytes 
saveRDS(BN_03, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_03.rds")
table(Idents(BN_03)) 

#subset  and save CD16 monocytes 
BN_03_monocytes <- subset(BN_03, idents = c("CD16 monocytes"))
table(Idents(BN_03_monocytes))

Idents(object = BN_03_monocytes) <- "Cov2 Sample 21c"
BN_03_monocytes$sample <- "Cov2 Sample 21c"
BN_03_monocytes$stim <- "Cov2"
levels(BN_03_monocytes)
BN_03_monocytes
saveRDS(BN_03_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_03_monocytes.rds")
rm(BN_03,BN_03_monocytes,BN_03.markers)

####BN-04
BN_04
Idents(object = BN_04) <- "Cov2 Sample 22c"
BN_04$sample <- "Cov2 Sample 22c"
BN_04$stim <- "Cov2"
levels(BN_04)
BN_04
BN_04[["percent.mt"]] <- PercentageFeatureSet(BN_04, pattern = "^MT-")
VlnPlot(BN_04, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_04<- subset(BN_04, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_04
BN_04 <- PercentageFeatureSet(BN_04, pattern = "^MT-", col.name = "percent.mt")
BN_04 <- SCTransform(BN_04, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_04) <- "sample"
levels(BN_04)
BN_04<- RunPCA(BN_04, features = VariableFeatures(object = BN_04))
DimPlot(BN_04, reduction = "pca")
ElbowPlot(BN_04, ndims = 100)
pct <- BN_04[["pca"]]@stdev / sum(BN_04[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_04 <- FindNeighbors(BN_04, dims = 1:14)
BN_04 <- FindClusters(BN_04, resolution = 0.5)
BN_04 <- RunUMAP(BN_04, dims = 1:14)
DimPlot(BN_04, reduction = "umap", pt.size =0.2, label = TRUE)
BN_04.markers <- FindAllMarkers(BN_04, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_04.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_04, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_04, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_04, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_04)
new.cluster.ids <- c("0", "1", "2", "3","4","5","6","7" ,"8","9","CD16 monocytes", "11", "12", "13", "14", "15", "16", "17")
names(new.cluster.ids) <- levels(BN_04)
BN_04 <- RenameIdents(BN_04, new.cluster.ids)
levels(BN_04)
DimPlot(BN_04, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()


##save before extracting CD16 monocytes 
saveRDS(BN_04, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_04.rds")
table(Idents(BN_04)) 

#subset  and save CD16 monocytes 
BN_04_monocytes <- subset(BN_04, idents = c("CD16 monocytes"))
table(Idents(BN_04_monocytes))

Idents(object = BN_04_monocytes) <- "Cov2 Sample 22c"
BN_04_monocytes$sample <- "Cov2 Sample 22c"
BN_04_monocytes$stim <- "Cov2"
levels(BN_04_monocytes)
BN_04_monocytes
saveRDS(BN_04_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_04_monocytes.rds")
rm(BN_04,BN_04_monocytes,BN_04.markers)


####BN-05
BN_05
Idents(object = BN_05) <- "Cov2 Sample 23c"
BN_05$sample <- "Cov2 Sample 23c"
BN_05$stim <- "Cov2"
levels(BN_05)
BN_05
BN_05[["percent.mt"]] <- PercentageFeatureSet(BN_05, pattern = "^MT-")
VlnPlot(BN_05, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_05<- subset(BN_05, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_05
BN_05 <- PercentageFeatureSet(BN_05, pattern = "^MT-", col.name = "percent.mt")
BN_05 <- SCTransform(BN_05, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_05) <- "sample"
levels(BN_05)
BN_05<- RunPCA(BN_05, features = VariableFeatures(object = BN_05))
DimPlot(BN_05, reduction = "pca")
ElbowPlot(BN_05, ndims = 100)
pct <- BN_05[["pca"]]@stdev / sum(BN_05[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_05 <- FindNeighbors(BN_05, dims = 1:12)
BN_05 <- FindClusters(BN_05, resolution = 0.5)
BN_05 <- RunUMAP(BN_05, dims = 1:12)
DimPlot(BN_05, reduction = "umap", pt.size =0.2, label = TRUE)
BN_05.markers <- FindAllMarkers(BN_05, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_05.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_05, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_05, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_05, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_05)
new.cluster.ids <- c("0", "1", "2", "3","4","CD16 monocytes","6","7" ,"8","9","10", "11", "12", "13", "14")
names(new.cluster.ids) <- levels(BN_05)
BN_05 <- RenameIdents(BN_05, new.cluster.ids)
levels(BN_05)
DimPlot(BN_05, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#cluster 5 are intermediate monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_05, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_05.rds")
table(Idents(BN_05)) 

#subset  and save CD16 monocytes 
BN_05_monocytes <- subset(BN_05, idents = c("CD16 monocytes"))
table(Idents(BN_05_monocytes))

Idents(object = BN_05_monocytes) <- "Cov2 Sample 23c"
BN_05_monocytes$sample <- "Cov2 Sample 23c"
BN_05_monocytes$stim <- "Cov2"
levels(BN_05_monocytes)
BN_05_monocytes
saveRDS(BN_05_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_05_monocytes.rds")
rm(BN_05,BN_05_monocytes,BN_05.markers)

####BN-06
BN_06
Idents(object = BN_06) <- "Cov2 Sample 24c"
BN_06$sample <- "Cov2 Sample 24c"
BN_06$stim <- "Cov2"
levels(BN_06)
BN_06
BN_06[["percent.mt"]] <- PercentageFeatureSet(BN_06, pattern = "^MT-")
VlnPlot(BN_06, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_06<- subset(BN_06, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_06
BN_06 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_06 <- SCTransform(BN_06, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_06) <- "sample"
levels(BN_06)
BN_06<- RunPCA(BN_06, features = VariableFeatures(object = BN_06))
DimPlot(BN_06, reduction = "pca")
ElbowPlot(BN_06, ndims = 100)
pct <- BN_06[["pca"]]@stdev / sum(BN_06[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_06 <- FindNeighbors(BN_06, dims = 1:13)
BN_06 <- FindClusters(BN_06, resolution = 0.5)
BN_06 <- RunUMAP(BN_06, dims = 1:13)
DimPlot(BN_06, reduction = "umap", pt.size =0.2, label = TRUE)
BN_06.markers <- FindAllMarkers(BN_06, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_06.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_06, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_06, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_06, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_06)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12")
names(new.cluster.ids) <- levels(BN_06)
BN_06 <- RenameIdents(BN_06, new.cluster.ids)
levels(BN_06)
DimPlot(BN_06, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16

##save before extracting CD16 monocytes 
saveRDS(BN_06, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_06.rds")
table(Idents(BN_06)) 

#subset  and save CD16 monocytes 
BN_06_monocytes <- subset(BN_06, idents = c("CD16 monocytes"))
table(Idents(BN_06_monocytes))

Idents(object = BN_06_monocytes) <- "Cov2 Sample 24c"
BN_06_monocytes$sample <- "Cov2 Sample 24c"
BN_06_monocytes$stim <- "Cov2"
levels(BN_06_monocytes)
BN_06_monocytes
saveRDS(BN_06_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_06_monocytes.rds")
rm(BN_06,BN_06_monocytes,BN_06.markers)

####BN-07
BN_07
Idents(object = BN_07) <- "Cov2 Sample 25c"
BN_07$sample <- "Cov2 Sample 25c"
BN_07$stim <- "Cov2"
levels(BN_07)
BN_07
BN_07[["percent.mt"]] <- PercentageFeatureSet(BN_07, pattern = "^MT-")
VlnPlot(BN_07, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_07<- subset(BN_07, subset = nFeature_RNA > 200 & nFeature_RNA < 4500 & percent.mt < 25)
BN_07
BN_07 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_07 <- SCTransform(BN_07, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_07) <- "sample"
levels(BN_07)
BN_07<- RunPCA(BN_07, features = VariableFeatures(object = BN_07))
DimPlot(BN_07, reduction = "pca")
ElbowPlot(BN_07, ndims = 100)
pct <- BN_07[["pca"]]@stdev / sum(BN_07[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_07 <- FindNeighbors(BN_07, dims = 1:13)
BN_07 <- FindClusters(BN_07, resolution = 0.5)
BN_07 <- RunUMAP(BN_07, dims = 1:13)
DimPlot(BN_07, reduction = "umap", pt.size =0.2, label = TRUE)
BN_07.markers <- FindAllMarkers(BN_07, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_07.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_07, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_07, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_07, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_07)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_07)
BN_07 <- RenameIdents(BN_07, new.cluster.ids)
levels(BN_07)
DimPlot(BN_07, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_07, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_07.rds")
table(Idents(BN_07)) 

#subset  and save CD16 monocytes 
BN_07_monocytes <- subset(BN_07, idents = c("CD16 monocytes"))
table(Idents(BN_07_monocytes))

Idents(object = BN_07_monocytes) <- "Cov2 Sample 25c"
BN_07_monocytes$sample <- "Cov2 Sample 25c"
BN_07_monocytes$stim <- "Cov2"
levels(BN_07_monocytes)
BN_07_monocytes
saveRDS(BN_07_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_07_monocytes.rds")
rm(BN_07,BN_07_monocytes,BN_07.markers)

####BN-08
BN_08
Idents(object = BN_08) <- "Cov2 Sample 26c"
BN_08$sample <- "Cov2 Sample 26c"
BN_08$stim <- "Cov2"
levels(BN_08)
BN_08
BN_08[["percent.mt"]] <- PercentageFeatureSet(BN_08, pattern = "^MT-")
VlnPlot(BN_08, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_08<- subset(BN_08, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_08
BN_08 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_08 <- SCTransform(BN_08, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_08) <- "sample"
levels(BN_08)
BN_08<- RunPCA(BN_08, features = VariableFeatures(object = BN_08))
DimPlot(BN_08, reduction = "pca")
ElbowPlot(BN_08, ndims = 100)
pct <- BN_08[["pca"]]@stdev / sum(BN_08[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_08 <- FindNeighbors(BN_08, dims = 1:13)
BN_08 <- FindClusters(BN_08, resolution = 0.5)
BN_08 <- RunUMAP(BN_08, dims = 1:13)
DimPlot(BN_08, reduction = "umap", pt.size =0.2, label = TRUE)
BN_08.markers <- FindAllMarkers(BN_08, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_08.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_08, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_08, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_08, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_08)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_08)
BN_08 <- RenameIdents(BN_08, new.cluster.ids)
levels(BN_08)
DimPlot(BN_08, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16

##save before extracting CD16 monocytes 
saveRDS(BN_08, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_08.rds")
table(Idents(BN_08)) 

#subset  and save CD16 monocytes 
BN_08_monocytes <- subset(BN_08, idents = c("CD16 monocytes"))
table(Idents(BN_08_monocytes))

Idents(object = BN_08_monocytes) <- "Cov2 Sample 26c"
BN_08_monocytes$sample <- "Cov2 Sample 26c"
BN_08_monocytes$stim <- "Cov2"
levels(BN_08_monocytes)
BN_08_monocytes
saveRDS(BN_08_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_08_monocytes.rds")
rm(BN_08,BN_08_monocytes,BN_08.markers)

####BN-10
BN_10
Idents(object = BN_10) <- "Cov2 Sample 27c"
BN_10$sample <- "Cov2 Sample 27c"
BN_10$stim <- "Cov2"
levels(BN_10)
BN_10
BN_10[["percent.mt"]] <- PercentageFeatureSet(BN_10, pattern = "^MT-")
VlnPlot(BN_10, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_10<- subset(BN_10, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 25)
BN_10
BN_10 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_10 <- SCTransform(BN_10, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_10) <- "sample"
levels(BN_10)
BN_10<- RunPCA(BN_10, features = VariableFeatures(object = BN_10))
DimPlot(BN_10, reduction = "pca")
ElbowPlot(BN_10, ndims = 100)
pct <- BN_10[["pca"]]@stdev / sum(BN_10[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_10 <- FindNeighbors(BN_10, dims = 1:13)
BN_10 <- FindClusters(BN_10, resolution = 0.5)
BN_10 <- RunUMAP(BN_10, dims = 1:13)
DimPlot(BN_10, reduction = "umap", pt.size =0.2, label = TRUE)
BN_10.markers <- FindAllMarkers(BN_10, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_10.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_10, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_10, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_10, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_10)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_10)
BN_10 <- RenameIdents(BN_10, new.cluster.ids)
levels(BN_10)
DimPlot(BN_10, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#very few cd16

##save before extracting CD16 monocytes 
saveRDS(BN_10, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_10.rds")
table(Idents(BN_10)) 

#subset  and save CD16 monocytes 
BN_10_monocytes <- subset(BN_10, idents = c("CD16 monocytes"))
table(Idents(BN_10_monocytes))

Idents(object = BN_10_monocytes) <- "Cov2 Sample 27c"
BN_10_monocytes$sample <- "Cov2 Sample 27c"
BN_10_monocytes$stim <- "Cov2"
levels(BN_10_monocytes)
BN_10_monocytes
saveRDS(BN_10_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_10_monocytes.rds")
rm(BN_10,BN_10_monocytes,BN_10.markers)

####BN-11
BN_11
Idents(object = BN_11) <- "Cov2 Sample 28c"
BN_11$sample <- "Cov2 Sample 28c"
BN_11$stim <- "Cov2"
levels(BN_11)
BN_11
BN_11[["percent.mt"]] <- PercentageFeatureSet(BN_11, pattern = "^MT-")
VlnPlot(BN_11, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_11<- subset(BN_11, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 25)
BN_11
BN_11 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_11 <- SCTransform(BN_11, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_11) <- "sample"
levels(BN_11)
BN_11<- RunPCA(BN_11, features = VariableFeatures(object = BN_11))
DimPlot(BN_11, reduction = "pca")
ElbowPlot(BN_11, ndims = 100)
pct <- BN_11[["pca"]]@stdev / sum(BN_11[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_11 <- FindNeighbors(BN_11, dims = 1:13)
BN_11 <- FindClusters(BN_11, resolution = 0.5)
BN_11 <- RunUMAP(BN_11, dims = 1:13)
DimPlot(BN_11, reduction = "umap", pt.size =0.2, label = TRUE)
BN_11.markers <- FindAllMarkers(BN_11, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_11.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_11, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_11, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_11, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_11)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_11)
BN_11 <- RenameIdents(BN_11, new.cluster.ids)
levels(BN_11)
DimPlot(BN_11, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#very few cd16

##save before extracting CD16 monocytes 
saveRDS(BN_11, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_11.rds")
table(Idents(BN_11)) 

#subset  and save CD16 monocytes 
BN_11_monocytes <- subset(BN_11, idents = c("CD16 monocytes"))
table(Idents(BN_11_monocytes))

Idents(object = BN_11_monocytes) <- "Cov2 Sample 28c"
BN_11_monocytes$sample <- "Cov2 Sample 28c"
BN_11_monocytes$stim <- "Cov2"
levels(BN_11_monocytes)
BN_11_monocytes
saveRDS(BN_11_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_11_monocytes.rds")
rm(BN_11,BN_11_monocytes,BN_11.markers)


####BN-12
BN_12
Idents(object = BN_12) <- "Cov2 Sample 29c"
BN_12$sample <- "Cov2 Sample 29c"
BN_12$stim <- "Cov2"
levels(BN_12)
BN_12
BN_12[["percent.mt"]] <- PercentageFeatureSet(BN_12, pattern = "^MT-")
VlnPlot(BN_12, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_12<- subset(BN_12, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 25)
BN_12
BN_12 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_12 <- SCTransform(BN_12, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_12) <- "sample"
levels(BN_12)
BN_12<- RunPCA(BN_12, features = VariableFeatures(object = BN_12))
DimPlot(BN_12, reduction = "pca")
ElbowPlot(BN_12, ndims = 100)
pct <- BN_12[["pca"]]@stdev / sum(BN_12[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_12 <- FindNeighbors(BN_12, dims = 1:13)
BN_12 <- FindClusters(BN_12, resolution = 0.5)
BN_12 <- RunUMAP(BN_12, dims = 1:13)
DimPlot(BN_12, reduction = "umap", pt.size =0.2, label = TRUE)
BN_12.markers <- FindAllMarkers(BN_12, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_12.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_12, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_12, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_12, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_12)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_12)
BN_12 <- RenameIdents(BN_12, new.cluster.ids)
levels(BN_12)
DimPlot(BN_12, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16
##save before extracting CD16 monocytes 
saveRDS(BN_12, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_12.rds")
table(Idents(BN_12)) 

#subset  and save CD16 monocytes 
BN_12_monocytes <- subset(BN_12, idents = c("CD16 monocytes"))
table(Idents(BN_12_monocytes))

Idents(object = BN_12_monocytes) <- "Cov2 Sample 29c"
BN_12_monocytes$sample <- "Cov2 Sample 29c"
BN_12_monocytes$stim <- "Cov2"
levels(BN_12_monocytes)
BN_12_monocytes
saveRDS(BN_12_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_12_monocytes.rds")
rm(BN_12,BN_12_monocytes,BN_12.markers)


####BN-13
BN_13
Idents(object = BN_13) <- "Cov2 Sample 30c"
BN_13$sample <- "Cov2 Sample 30c"
BN_13$stim <- "Cov2"
levels(BN_13)
BN_13
BN_13[["percent.mt"]] <- PercentageFeatureSet(BN_13, pattern = "^MT-")
VlnPlot(BN_13, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_13<- subset(BN_13, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 25)
BN_13
BN_13 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_13 <- SCTransform(BN_13, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_13) <- "sample"
levels(BN_13)
BN_13<- RunPCA(BN_13, features = VariableFeatures(object = BN_13))
DimPlot(BN_13, reduction = "pca")
ElbowPlot(BN_13, ndims = 100)
pct <- BN_13[["pca"]]@stdev / sum(BN_13[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_13 <- FindNeighbors(BN_13, dims = 1:13)
BN_13 <- FindClusters(BN_13, resolution = 0.5)
BN_13 <- RunUMAP(BN_13, dims = 1:13)
DimPlot(BN_13, reduction = "umap", pt.size =0.2, label = TRUE)
BN_13.markers <- FindAllMarkers(BN_13, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_13.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_13, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_13, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_13, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_13)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_13)
BN_13 <- RenameIdents(BN_13, new.cluster.ids)
levels(BN_13)
DimPlot(BN_13, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_13, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_13.rds")
table(Idents(BN_13)) 

#subset  and save CD16 monocytes 
BN_13_monocytes <- subset(BN_13, idents = c("CD16 monocytes"))
table(Idents(BN_13_monocytes))

Idents(object = BN_13_monocytes) <- "Cov2 Sample 30c"
BN_13_monocytes$sample <- "Cov2 Sample 30c"
BN_13_monocytes$stim <- "Cov2"
levels(BN_13_monocytes)
BN_13_monocytes
saveRDS(BN_13_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_13_monocytes.rds")
rm(BN_13,BN_13_monocytes,BN_13.markers)


####BN-14
BN_14
Idents(object = BN_14) <- "Cov2 Sample 31c"
BN_14$sample <- "Cov2 Sample 31c"
BN_14$stim <- "Cov2"
levels(BN_14)
BN_14
BN_14[["percent.mt"]] <- PercentageFeatureSet(BN_14, pattern = "^MT-")
VlnPlot(BN_14, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_14<- subset(BN_14, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 25)
BN_14
BN_14 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_14 <- SCTransform(BN_14, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_14) <- "sample"
levels(BN_14)
BN_14<- RunPCA(BN_14, features = VariableFeatures(object = BN_14))
DimPlot(BN_14, reduction = "pca")
ElbowPlot(BN_14, ndims = 100)
pct <- BN_14[["pca"]]@stdev / sum(BN_14[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_14 <- FindNeighbors(BN_14, dims = 1:13)
BN_14 <- FindClusters(BN_14, resolution = 0.5)
BN_14 <- RunUMAP(BN_14, dims = 1:13)
DimPlot(BN_14, reduction = "umap", pt.size =0.2, label = TRUE)
BN_14.markers <- FindAllMarkers(BN_14, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_14.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_14, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_14, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_14, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_14)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_14)
BN_14 <- RenameIdents(BN_14, new.cluster.ids)
levels(BN_14)
DimPlot(BN_14, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_14, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_14.rds")
table(Idents(BN_14)) 

#subset  and save CD16 monocytes 
BN_14_monocytes <- subset(BN_14, idents = c("CD16 monocytes"))
table(Idents(BN_14_monocytes))

Idents(object = BN_14_monocytes) <- "Cov2 Sample 31c"
BN_14_monocytes$sample <- "Cov2 Sample 31c"
BN_14_monocytes$stim <- "Cov2"
levels(BN_14_monocytes)
BN_14_monocytes
saveRDS(BN_14_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_14_monocytes.rds")
rm(BN_14,BN_14_monocytes,BN_14.markers)


####BN-15
BN_15
Idents(object = BN_15) <- "Cov2 Sample 32c"
BN_15$sample <- "Cov2 Sample 32c"
BN_15$stim <- "Cov2"
levels(BN_15)
BN_15
BN_15[["percent.mt"]] <- PercentageFeatureSet(BN_15, pattern = "^MT-")
VlnPlot(BN_15, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_15<- subset(BN_15, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 25)
BN_15
BN_15 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_15 <- SCTransform(BN_15, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_15) <- "sample"
levels(BN_15)
BN_15<- RunPCA(BN_15, features = VariableFeatures(object = BN_15))
DimPlot(BN_15, reduction = "pca")
ElbowPlot(BN_15, ndims = 100)
pct <- BN_15[["pca"]]@stdev / sum(BN_15[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_15 <- FindNeighbors(BN_15, dims = 1:13)
BN_15 <- FindClusters(BN_15, resolution = 0.5)
BN_15 <- RunUMAP(BN_15, dims = 1:13)
DimPlot(BN_15, reduction = "umap", pt.size =0.2, label = TRUE)
BN_15.markers <- FindAllMarkers(BN_15, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_15.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_15, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_15, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_15, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_15)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_15)
BN_15 <- RenameIdents(BN_15, new.cluster.ids)
levels(BN_15)
DimPlot(BN_15, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_15, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_15.rds")
table(Idents(BN_15)) 

#subset  and save CD16 monocytes 
BN_15_monocytes <- subset(BN_15, idents = c("CD16 monocytes"))
table(Idents(BN_15_monocytes))

Idents(object = BN_15_monocytes) <- "Cov2 Sample 32c"
BN_15_monocytes$sample <- "Cov2 Sample 32c"
BN_15_monocytes$stim <- "Cov2"
levels(BN_15_monocytes)
BN_15_monocytes
saveRDS(BN_15_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_15_monocytes.rds")
rm(BN_15,BN_15_monocytes,BN_15.markers)

####BN-16
BN_16
Idents(object = BN_16) <- "Cov2 Sample 33c"
BN_16$sample <- "Cov2 Sample 33c"
BN_16$stim <- "Cov2"
levels(BN_16)
BN_16
BN_16[["percent.mt"]] <- PercentageFeatureSet(BN_16, pattern = "^MT-")
VlnPlot(BN_16, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_16<- subset(BN_16, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_16
BN_16 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_16 <- SCTransform(BN_16, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_16) <- "sample"
levels(BN_16)
BN_16<- RunPCA(BN_16, features = VariableFeatures(object = BN_16))
DimPlot(BN_16, reduction = "pca")
ElbowPlot(BN_16, ndims = 100)
pct <- BN_16[["pca"]]@stdev / sum(BN_16[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_16 <- FindNeighbors(BN_16, dims = 1:15)
BN_16 <- FindClusters(BN_16, resolution = 0.5)
BN_16 <- RunUMAP(BN_16, dims = 1:15)
DimPlot(BN_16, reduction = "umap", pt.size =0.2, label = TRUE)
BN_16.markers <- FindAllMarkers(BN_16, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_16.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_16, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_16, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_16, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_16)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_16)
BN_16 <- RenameIdents(BN_16, new.cluster.ids)
levels(BN_16)
DimPlot(BN_16, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#no cd16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_16, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_16.rds")
table(Idents(BN_16)) 

#subset  and save CD16 monocytes 
BN_16_monocytes <- subset(BN_16, idents = c("CD16 monocytes"))
table(Idents(BN_16_monocytes))

Idents(object = BN_16_monocytes) <- "Cov2 Sample 33c"
BN_16_monocytes$sample <- "Cov2 Sample 33c"
BN_16_monocytes$stim <- "Cov2"
levels(BN_16_monocytes)
BN_16_monocytes
saveRDS(BN_16_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_16_monocytes.rds")
rm(BN_16,BN_16_monocytes,BN_16.markers)


####BN-17
BN_17
Idents(object = BN_17) <- "Cov2 Sample 34c"
BN_17$sample <- "Cov2 Sample 34c"
BN_17$stim <- "Cov2"
levels(BN_17)
BN_17
BN_17[["percent.mt"]] <- PercentageFeatureSet(BN_17, pattern = "^MT-")
VlnPlot(BN_17, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_17<- subset(BN_17, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_17
BN_17 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_17 <- SCTransform(BN_17, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_17) <- "sample"
levels(BN_17)
BN_17<- RunPCA(BN_17, features = VariableFeatures(object = BN_17))
DimPlot(BN_17, reduction = "pca")
ElbowPlot(BN_17, ndims = 100)
pct <- BN_17[["pca"]]@stdev / sum(BN_17[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_17 <- FindNeighbors(BN_17, dims = 1:13)
BN_17 <- FindClusters(BN_17, resolution = 0.5)
BN_17 <- RunUMAP(BN_17, dims = 1:13)
DimPlot(BN_17, reduction = "umap", pt.size =0.2, label = TRUE)
BN_17.markers <- FindAllMarkers(BN_17, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_17.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_17, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_17, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_17, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_17)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_17)
BN_17 <- RenameIdents(BN_17, new.cluster.ids)
levels(BN_17)
DimPlot(BN_17, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()


##save before extracting CD16 monocytes 
saveRDS(BN_17, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_17.rds")
table(Idents(BN_17)) 

#subset  and save CD16 monocytes 
BN_17_monocytes <- subset(BN_17, idents = c("CD16 monocytes"))
table(Idents(BN_17_monocytes))

#few cd16 monocytes

Idents(object = BN_17_monocytes) <- "Cov2 Sample 34c"
BN_17_monocytes$sample <- "Cov2 Sample 34c"
BN_17_monocytes$stim <- "Cov2"
levels(BN_17_monocytes)
BN_17_monocytes
saveRDS(BN_17_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_17_monocytes.rds")
rm(BN_17,BN_17_monocytes,BN_17.markers)


####BN-19
BN_19
Idents(object = BN_19) <- "Cov2 Sample 36c"
BN_19$sample <- "Cov2 Sample 36c"
BN_19$stim <- "Cov2"
levels(BN_19)
BN_19
BN_19[["percent.mt"]] <- PercentageFeatureSet(BN_19, pattern = "^MT-")
VlnPlot(BN_19, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_19<- subset(BN_19, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 25)
BN_19
BN_19 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_19 <- SCTransform(BN_19, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_19) <- "sample"
levels(BN_19)
BN_19<- RunPCA(BN_19, features = VariableFeatures(object = BN_19))
DimPlot(BN_19, reduction = "pca")
ElbowPlot(BN_19, ndims = 100)
pct <- BN_19[["pca"]]@stdev / sum(BN_19[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_19 <- FindNeighbors(BN_19, dims = 1:13)
BN_19 <- FindClusters(BN_19, resolution = 0.5)
BN_19 <- RunUMAP(BN_19, dims = 1:13)
DimPlot(BN_19, reduction = "umap", pt.size =0.2, label = TRUE)
BN_19.markers <- FindAllMarkers(BN_19, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_19.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_19, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_19, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_19, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_19)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_19)
BN_19 <- RenameIdents(BN_19, new.cluster.ids)
levels(BN_19)
DimPlot(BN_19, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_19, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_19.rds")
table(Idents(BN_19)) 

#subset  and save CD16 monocytes 
BN_19_monocytes <- subset(BN_19, idents = c("CD16 monocytes"))
table(Idents(BN_19_monocytes))

Idents(object = BN_19_monocytes) <- "Cov2 Sample 36c"
BN_19_monocytes$sample <- "Cov2 Sample 36c"
BN_19_monocytes$stim <- "Cov2"
levels(BN_19_monocytes)
BN_19_monocytes
saveRDS(BN_19_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_19_monocytes.rds")
rm(BN_19,BN_19_monocytes,BN_19.markers)

normal5 <- subset(data, idents = c("5"))
normal13 <- subset(data, idents = c("13"))
normal14 <- subset(data, idents = c("14"))
normal19 <- subset(data, idents = c("19"))


normal5
Idents(object = normal5) <- "Uninfected Normal 1"
normal5$sample <- "Uninfected Normal 1"
normal5$stim <- "Uninfected"
levels(normal5)
normal5


normal13
Idents(object = normal13) <- "Uninfected Normal 2"
normal13$sample <- "Uninfected Normal 2"
normal13$stim <- "Uninfected"
levels(normal13)
normal13

normal14
Idents(object = normal14) <- "Uninfected Normal 3"
normal14$sample <- "Uninfected Normal 3"
normal14$stim <- "Uninfected"
levels(normal14)
normal14


normal19
Idents(object = normal19) <- "Uninfected Normal 4"
normal19$sample <- "Uninfected Normal 4"
normal19$stim <- "Uninfected"
levels(normal19)
normal19

rm(data)

normal5[["percent.mt"]] <- PercentageFeatureSet(normal5, pattern = "^MT-")
VlnPlot(normal5, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.normal5 <- FeatureScatter(normal5, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.normal5 <- FeatureScatter(normal5, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.normal5, plot2.normal5))
#For normal5 decided on the following parameters for subsetting QC percent.mt < 18%, nFeature_RNA > 200 < 4000

normal13[["percent.mt"]] <- PercentageFeatureSet(normal13, pattern = "^MT-")
VlnPlot(normal13, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.normal13 <- FeatureScatter(normal13, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.normal13 <- FeatureScatter(normal13, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.normal13, plot2.normal13))
#For normal13 decided on the following parameters for subsetting QC percent.mt < 20%, nFeature_RNA > 200 < 4000

normal14[["percent.mt"]] <- PercentageFeatureSet(normal14, pattern = "^MT-")
VlnPlot(normal14, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.normal14 <- FeatureScatter(normal14, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.normal14 <- FeatureScatter(normal14, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.normal14, plot2.normal14))
#For normal14 decided on the following parameters for subsetting QC percent.mt < 20%, nFeature_RNA > 200 < 4000

normal19[["percent.mt"]] <- PercentageFeatureSet(normal19, pattern = "^MT-")
VlnPlot(normal19, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.normal19 <- FeatureScatter(normal19, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.normal19 <- FeatureScatter(normal19, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.normal19, plot2.normal19))
#For normal19 decided on the following parameters for subsetting QC percent.mt < 20%, nFeature_RNA > 200 < 4000

rm(plot1.normal5,plot1.normal13,plot1.normal4,plot1.normal9)
rm(plot2.normal5,plot2.normal13,plot2.normal14,plot2.normal19)

#subset the data based on QC metrics
normal5<- subset(normal5, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 18)
normal13<- subset(normal13, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 20)
normal14<- subset(normal14, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 20)
normal19<- subset(normal19, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 20)

###check no of cells for each group
normal5
normal13
normal14
normal19

# store mitochondrial percentage in object meta data
normal5 <- PercentageFeatureSet(normal5, pattern = "^MT-", col.name = "percent.mt")
normal13 <- PercentageFeatureSet(normal13, pattern = "^MT-", col.name = "percent.mt")
normal14 <- PercentageFeatureSet(normal14, pattern = "^MT-", col.name = "percent.mt")
normal19 <- PercentageFeatureSet(normal19, pattern = "^MT-", col.name = "percent.mt")

# run sctransform
normal5 <- SCTransform(normal5, vars.to.regress = "percent.mt", verbose = FALSE)
normal13 <- SCTransform(normal13, vars.to.regress = "percent.mt", verbose = FALSE)
normal14 <- SCTransform(normal14, vars.to.regress = "percent.mt", verbose = FALSE)
normal19 <- SCTransform(normal19, vars.to.regress = "percent.mt", verbose = FALSE)

#dimensionality reduction and clustering
normal5 <- RunPCA(normal5, verbose = FALSE)
normal5 <- RunUMAP(normal5, dims = 1:30, verbose = FALSE)
normal5 <- FindNeighbors(normal5, dims = 1:30, verbose = FALSE)
normal5 <- FindClusters(normal5, verbose = FALSE)
DimPlot(normal5, label = TRUE) + NoLegend()
normal5.markers <- FindAllMarkers(normal5, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
normal5.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(normal5, features = c("FCGR3A", "MS4A7"))
new.cluster.ids <- c("0","1","2","3","4","5","6","7","8","9","10","CD16 monocytes","12","13","14","15","16") 
names(new.cluster.ids) <- levels(normal5)
normal5 <- RenameIdents(normal5, new.cluster.ids)
DimPlot(normal5, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
table(Idents(normal5))
normal5monocytes <- subset(normal5, idents = c("CD16 monocytes"))
saveRDS(object=normal5monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/PM folder/Covid_flu paper extract healthy monocytes 091620/normal5monocytes.rds")

normal13 <- RunPCA(normal13, verbose = FALSE)
normal13 <- RunUMAP(normal13, dims = 1:30, verbose = FALSE)
normal13 <- FindNeighbors(normal13, dims = 1:30, verbose = FALSE)
normal13 <- FindClusters(normal13, verbose = FALSE)
DimPlot(normal13, label = TRUE) + NoLegend()
normal13.markers <- FindAllMarkers(normal13, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
normal13.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(normal13, features = c("FCGR3A", "MS4A7"))
VlnPlot(normal13, features = c("FCGR3B", "CXCR2", "FCGR3A","MS4A7"))
new.cluster.ids <- c("0","1","2","3","4","CD16 monocytes","6","7","8","9","10","11","12","13","14","15","16","17") 
names(new.cluster.ids) <- levels(normal13)
normal13 <- RenameIdents(normal13, new.cluster.ids)
DimPlot(normal13, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
table(Idents(normal13))
normal13monocytes <- subset(normal13, idents = c("CD16 monocytes"))
saveRDS(object=normal13monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/PM folder/Covid_flu paper extract healthy monocytes 091620/normal13monocytes.rds")

normal14 <- RunPCA(normal14, verbose = FALSE)
normal14 <- RunUMAP(normal14, dims = 1:30, verbose = FALSE)
normal14 <- FindNeighbors(normal14, dims = 1:30, verbose = FALSE)
normal14 <- FindClusters(normal14, verbose = FALSE)
DimPlot(normal14, label = TRUE) + NoLegend()
normal14.markers <- FindAllMarkers(normal14, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
normal14.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(normal14, features = c("FCGR3A", "MS4A7"))
VlnPlot(normal14, features = c("FCGR3B", "CXCR2", "FCGR3A","MS4A7"))
new.cluster.ids <- c("0","1","2","3","4","CD16 monocytes","6","7","8","9","10","11","12","13","14","15","16","17","18","19")
names(new.cluster.ids) <- levels(normal14) 
normal14 <- RenameIdents(normal14, new.cluster.ids)
DimPlot(normal14, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
table(Idents(normal14))
normal14monocytes <- subset(normal14, idents = c("CD16 monocytes"))
saveRDS(object=normal14monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/PM folder/Covid_flu paper extract healthy monocytes 091620/normal14monocytes.rds")

normal19 <- RunPCA(normal19, verbose = FALSE)
normal19 <- RunUMAP(normal19, dims = 1:30, verbose = FALSE)
normal19 <- FindNeighbors(normal19, dims = 1:30, verbose = FALSE)
normal19 <- FindClusters(normal19, verbose = FALSE)
DimPlot(normal19, label = TRUE) + NoLegend()
normal19.markers <- FindAllMarkers(normal19, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
normal19.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(normal19, features = c("FCGR3A", "MS4A7"))
VlnPlot(normal19, features = c("FCGR3B", "CXCR2", "FCGR3A","MS4A7"))
new.cluster.ids <- c("0","1","2","3","4","5","CD16 monocytes","7","8","9","10","11","12","13","14","15")
names(new.cluster.ids) <- levels(normal19) 
normal19 <- RenameIdents(normal19, new.cluster.ids)
DimPlot(normal19, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
table(Idents(normal19))
normal19monocytes <- subset(normal19, idents = c("CD16 monocytes"))
saveRDS(object=normal19monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/PM folder/Covid_flu paper extract healthy monocytes 091620/normal19monocytes.rds")

rm(normal5.markers,normal13.markers,normal14.markers,normal19.markers)

###isolate monocytes from healthy donors from the pbmc covid single cell paper.
###HIP002
HIP002 <- readRDS(file = "~/Desktop/Covid analysis/GSE150728_RAW/GSM4557334_HIP002_cell.counts.matrices.rds")
HIP002 <- CreateSeuratObject(counts = HIP002$exon, project = "uninf pbmc", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
HIP002
Idents(object = HIP002) <- "Healthy PBMC Sample 1"
HIP002$sample <- "Healthy PBMC Sample 1"
HIP002$stim <- "Uninfected PBMC"
levels(HIP002)
HIP002

HIP002[["percent.mt"]] <- PercentageFeatureSet(HIP002, pattern = "^MT-")

VlnPlot(HIP002, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.HIP002 <- FeatureScatter(HIP002, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.HIP002 <- FeatureScatter(HIP002, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.HIP002, plot2.HIP002))
HIP002<- subset(HIP002, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 20)

##you can do sct transform 
HIP002 <- SCTransform(HIP002, vars.to.regress = "percent.mt", verbose = FALSE)
HIP002

Idents(object = HIP002) <- "sample"
levels(HIP002)


###linear dimensional reduction
HIP002<- RunPCA(HIP002, features = VariableFeatures(object = HIP002))
DimPlot(HIP002, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(HIP002, ndims = 100)
# Determine percent of variation associated with each PC
pct <- HIP002[["pca"]]@stdev / sum(HIP002[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
# Determine which PC exhibits cumulative percent greater than 90% and % variation associated with the PC as less than 5
co1 <- which(cumu > 90 & pct < 5)[1]
co1

co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
# last point where change of % of variation is more than 0.1%.
co2
#Now, to determine the selection of PCs, we will use the minimum of the two metrics:
# Minimum of the two calculation
pcs <- min(co1, co2)
pcs

###clustering
HIP002 <- FindNeighbors(HIP002, dims = 1:13)
HIP002 <- FindClusters(HIP002, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
HIP002 <- RunUMAP(HIP002, dims = 1:13)
DimPlot(HIP002, reduction = "umap", min.dist = 0.75, pt.size =0.2, label = TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
HIP002.markers <- FindAllMarkers(HIP002, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
HIP002.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


#check for mature monocyte markers and plot to see which cluster has highest level of expression
VlnPlot(HIP002, features = c( "FCGR3A","MS4A7"))
# you can plot raw counts as well
VlnPlot(HIP002, features = c("FCGR3A","MS4A7"), slot = "counts", log = TRUE)
FeaturePlot(HIP002, features = c("CD14","FCGR3A"))

levels(HIP002)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5","6", "7", "CD16monocytes","9","CD16 monocytes", "11", "12")
names(new.cluster.ids) <- levels(HIP002)
HIP002 <- RenameIdents(HIP002, new.cluster.ids)
levels(HIP002)
DimPlot(HIP002, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes

saveRDS(HIP002, file = "~/Desktop/Covid analysis/VC start/HIP002.rds")

table(Idents(HIP002))


HIP002_monocytes <- subset(HIP002, idents = c("CD16 monocytes","CD16monocytes"))
table(Idents(HIP002_monocytes))

Idents(object = HIP002_monocytes) <- "Healthy PBMC Sample 1 monocytes"
HIP002_monocytes$sample <- "Healthy PBMC Sample 1"
HIP002_monocytes$stim <- "Uninfected PBMC"
levels(HIP002_monocytes)
HIP002_monocytes
saveRDS(HIP002_monocytes, file = "~/Desktop/Covid analysis/VC start/HIP002_monocytes.rds")


###HIP0015
HIP015 <- readRDS(file = "~/Desktop/Covid analysis/GSE150728_RAW/GSM4557335_HIP015_cell.counts.matrices.rds")
HIP015 <- CreateSeuratObject(counts = HIP015$exon, project = "uninf pbmc", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
HIP015
Idents(object = HIP015) <- "Healthy PBMC Sample 2"
HIP015$sample <- "Healthy PBMC Sample 2"
HIP015$stim <- "Uninfected PBMC"
levels(HIP015)
HIP015

HIP015[["percent.mt"]] <- PercentageFeatureSet(HIP015, pattern = "^MT-")

VlnPlot(HIP015, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.HIP015 <- FeatureScatter(HIP015, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.HIP015 <- FeatureScatter(HIP015, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.HIP015, plot2.HIP002))
HIP015<- subset(HIP015, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 20)

##you can do sct transform 
HIP015 <- SCTransform(HIP015, vars.to.regress = "percent.mt", verbose = FALSE)
HIP015

Idents(object = HIP015) <- "sample"
levels(HIP015)


###linear dimensional reduction
HIP015<- RunPCA(HIP015, features = VariableFeatures(object = HIP015))
DimPlot(HIP015, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(HIP015, ndims = 100)
# Determine percent of variation associated with each PC
pct <- HIP015[["pca"]]@stdev / sum(HIP015[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
# Determine which PC exhibits cumulative percent greater than 90% and % variation associated with the PC as less than 5
co1 <- which(cumu > 90 & pct < 5)[1]
co1

co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
# last point where change of % of variation is more than 0.1%.
co2
#Now, to determine the selection of PCs, we will use the minimum of the two metrics:
# Minimum of the two calculation
pcs <- min(co1, co2)
pcs

###clustering
HIP015 <- FindNeighbors(HIP015, dims = 1:12)
HIP015 <- FindClusters(HIP015, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
HIP015 <- RunUMAP(HIP015, dims = 1:12)
DimPlot(HIP015, reduction = "umap", min.dist = 0.75, pt.size =0.2, label = TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
HIP015.markers <- FindAllMarkers(HIP015, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
HIP015.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


#check for mature monocyte markers and plot to see which cluster has highest level of expression
VlnPlot(HIP015, features = c("FCGR3B", "CXCR2", "FCGR3A","MSCXCR24A7"))
# you can plot raw counts as well
VlnPlot(HIP015, features = c("FCGR3B", "CXCR2","FCGR3A","MS4A7"), slot = "counts", log = TRUE)
FeaturePlot(HIP015, features = c("FCGR3B", "CXCR2","FCGR3A"))

levels(HIP015)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5","6", "7","8", "CD16 monocytes","10", "11", "12")
names(new.cluster.ids) <- levels(HIP015)
HIP015 <- RenameIdents(HIP015, new.cluster.ids)
levels(HIP015)
DimPlot(HIP015, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes

saveRDS(HIP015, file = "~/Desktop/Covid analysis/VC start/HIP015.rds")

table(Idents(HIP015)) ##few cells


HIP015_monocytes <- subset(HIP015, idents = c("CD16 monocytes"))
table(Idents(HIP015_monocytes))

Idents(object = HIP015_monocytes) <- "Healthy PBMC Sample 2 monocytes"
HIP015_monocytes$sample <- "Healthy PBMC Sample 2"
HIP015_monocytes$stim <- "Uninfected PBMC"
levels(HIP015_monocytes)
HIP015_monocytes
saveRDS(HIP015_monocytes, file = "~/Desktop/Covid analysis/VC start/HIP015_monocytes.rds")

###HIP0023
HIP023 <- readRDS(file = "~/Desktop/Covid analysis/GSE150728_RAW/GSM4557336_HIP023_cell.counts.matrices.rds")
HIP023 <- CreateSeuratObject(counts = HIP023$exon, project = "uninf pbmc", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
HIP023
Idents(object = HIP023) <- "Healthy PBMC Sample 3"
HIP023$sample <- "Healthy PBMC Sample 3"
HIP023$stim <- "Uninfected PBMC"
levels(HIP023)
HIP023

HIP023[["percent.mt"]] <- PercentageFeatureSet(HIP023, pattern = "^MT-")

VlnPlot(HIP023, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.HIP023 <- FeatureScatter(HIP023, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.HIP023 <- FeatureScatter(HIP023, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.HIP023, plot2.HIP023))
HIP023<- subset(HIP023, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 20)

##you can do sct transform 
HIP023 <- SCTransform(HIP023, vars.to.regress = "percent.mt", verbose = FALSE)
HIP023

Idents(object = HIP023) <- "sample"
levels(HIP023)


###linear dimensional reduction
HIP023<- RunPCA(HIP023, features = VariableFeatures(object = HIP023))
DimPlot(HIP023, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(HIP023, ndims = 100)
# Determine percent of variation associated with each PC
pct <- HIP023[["pca"]]@stdev / sum(HIP023[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
# Determine which PC exhibits cumulative percent greater than 90% and % variation associated with the PC as less than 5
co1 <- which(cumu > 90 & pct < 5)[1]
co1

co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
# last point where change of % of variation is more than 0.1%.
co2
#Now, to determine the selection of PCs, we will use the minimum of the two metrics:
# Minimum of the two calculation
pcs <- min(co1, co2)
pcs

###clustering
HIP023 <- FindNeighbors(HIP023, dims = 1:14)
HIP023 <- FindClusters(HIP023, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
HIP023 <- RunUMAP(HIP023, dims = 1:14)
DimPlot(HIP023, reduction = "umap", min.dist = 0.75, pt.size =0.2, label = TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
HIP023.markers <- FindAllMarkers(HIP023, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
HIP023.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


#check for mature monocyte markers and plot to see which cluster has highest level of expression
VlnPlot(HIP023, features = c("FCGR3B", "CXCR2", "FCGR3A","MS4A7"))
# you can plot raw counts as well
VlnPlot(HIP023, features = c("FCGR3B", "CXCR2","FCGR3A","MS4A7"), slot = "counts", log = TRUE)
FeaturePlot(HIP023, features = c("FCGR3B", "CXCR2","FCGR3A"))

levels(HIP023)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5","6", "7","8", "CD16 monocytes","10", "11", "12")
names(new.cluster.ids) <- levels(HIP023)
HIP023 <- RenameIdents(HIP023, new.cluster.ids)
levels(HIP023)
DimPlot(HIP023, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes

saveRDS(HIP023, file = "~/Desktop/Covid analysis/VC start/HIP023.rds")

table(Idents(HIP023)) ##few cells didnt proceed


###HIP043
HIP043 <- readRDS(file = "~/Desktop/Covid analysis/GSE150728_RAW/GSM4557337_HIP043_cell.counts.matrices.rds")
HIP043 <- CreateSeuratObject(counts = HIP043$exon, project = "uninf pbmc", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
HIP043
Idents(object = HIP043) <- "Healthy PBMC Sample 4"
HIP043$sample <- "Healthy PBMC Sample 4"
HIP043$stim <- "Uninfected PBMC"
levels(HIP043)
HIP043

HIP043[["percent.mt"]] <- PercentageFeatureSet(HIP043, pattern = "^MT-")

VlnPlot(HIP043, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.HIP043 <- FeatureScatter(HIP043, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.HIP043 <- FeatureScatter(HIP043, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.HIP043, plot2.HIP043))
HIP043<- subset(HIP043, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 20)

##you can do sct transform 
HIP043 <- SCTransform(HIP043, vars.to.regress = "percent.mt", verbose = FALSE)
HIP043

Idents(object = HIP043) <- "sample"
levels(HIP043)


###linear dimensional reduction
HIP043<- RunPCA(HIP043, features = VariableFeatures(object = HIP043))
DimPlot(HIP043, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(HIP043, ndims = 100)
# Determine percent of variation associated with each PC
pct <- HIP043[["pca"]]@stdev / sum(HIP043[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
# Determine which PC exhibits cumulative percent greater than 90% and % variation associated with the PC as less than 5
co1 <- which(cumu > 90 & pct < 5)[1]
co1

co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
# last point where change of % of variation is more than 0.1%.
co2
#Now, to determine the selection of PCs, we will use the minimum of the two metrics:
# Minimum of the two calculation
pcs <- min(co1, co2)
pcs

###clustering
HIP043 <- FindNeighbors(HIP043, dims = 1:11)
HIP043 <- FindClusters(HIP043, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
HIP043 <- RunUMAP(HIP043, dims = 1:11)
DimPlot(HIP043, reduction = "umap", min.dist = 0.75, pt.size =0.2, label = TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
HIP043.markers <- FindAllMarkers(HIP043, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
HIP043.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


#check for mature monocyte markers and plot to see which cluster has highest level of expression
VlnPlot(HIP043, features = c( "FCGR3A","MS4A7"))
# you can plot raw counts as well
VlnPlot(HIP043, features = c("FCGR3B", "CXCR2","FCGR3A","MS4A7"), slot = "counts", log = TRUE)
FeaturePlot(HIP043, features = c("FCGR3A","MS4A7","CD14"))

levels(HIP043)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5","6", "CD16 monocytes","8","9","10", "11")
names(new.cluster.ids) <- levels(HIP043)
HIP043 <- RenameIdents(HIP043, new.cluster.ids)
levels(HIP043)
DimPlot(HIP043, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes

saveRDS(HIP043, file = "~/Desktop/Covid analysis/VC start/HIP043.rds")

table(Idents(HIP043))


HIP043_monocytes <- subset(HIP043, idents = c("CD16 monocytes"))
table(Idents(HIP043_monocytes))

Idents(object = HIP043_monocytes) <- "Healthy PBMC Sample 4 monocytes"
HIP043_monocytes$sample <- "Healthy PBMC Sample 4"
HIP043_monocytes$stim <- "Uninfected PBMC"
levels(HIP043_monocytes)
HIP043_monocytes
saveRDS(HIP043_monocytes, file = "~/Desktop/Covid analysis/VC start/HIP043_monocytes.rds")


###HIP044
HIP044 <- readRDS(file = "~/Desktop/Covid analysis/GSE150728_RAW/GSM4557338_HIP044_cell.counts.matrices.rds")
HIP044 <- CreateSeuratObject(counts = HIP044$exon, project = "uninf pbmc", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
HIP044
Idents(object = HIP044) <- "Healthy PBMC Sample 5"
HIP044$sample <- "Healthy PBMC Sample 5"
HIP044$stim <- "Uninfected PBMC"
levels(HIP044)
HIP044

HIP044[["percent.mt"]] <- PercentageFeatureSet(HIP044, pattern = "^MT-")

VlnPlot(HIP044, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.HIP044 <- FeatureScatter(HIP044, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.HIP044 <- FeatureScatter(HIP044, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.HIP044, plot2.HIP044))
HIP044<- subset(HIP044, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 20)

##you can do sct transform 
HIP044 <- SCTransform(HIP044, vars.to.regress = "percent.mt", verbose = FALSE)
HIP044

Idents(object = HIP044) <- "sample"
levels(HIP044)


###linear dimensional reduction
HIP044<- RunPCA(HIP044, features = VariableFeatures(object = HIP044))
DimPlot(HIP044, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(HIP044, ndims = 100)
# Determine percent of variation associated with each PC
pct <- HIP044[["pca"]]@stdev / sum(HIP044[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
# Determine which PC exhibits cumulative percent greater than 90% and % variation associated with the PC as less than 5
co1 <- which(cumu > 90 & pct < 5)[1]
co1

co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
# last point where change of % of variation is more than 0.1%.
co2
#Now, to determine the selection of PCs, we will use the minimum of the two metrics:
# Minimum of the two calculation
pcs <- min(co1, co2)
pcs

###clustering
HIP044 <- FindNeighbors(HIP044, dims = 1:12)
HIP044 <- FindClusters(HIP044, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
HIP044 <- RunUMAP(HIP044, dims = 1:12)
DimPlot(HIP044, reduction = "umap", min.dist = 0.75, pt.size =0.2, label = TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
HIP044.markers <- FindAllMarkers(HIP044, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
HIP044.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


#check for mature monocyte markers and plot to see which cluster has highest level of expression
VlnPlot(HIP044, features = c("FCGR3B", "CXCR2", "FCGR3A","MS4A7"))
# you can plot raw counts as well
VlnPlot(HIP044, features = c("FCGR3B", "CXCR2","FCGR3A","MS4A7"), slot = "counts", log = TRUE)
FeaturePlot(HIP044, features = c("FCGR3B", "CXCR2","FCGR3A"))

levels(HIP044)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5", "CD16 monocytes","7", "8","9","10")
names(new.cluster.ids) <- levels(HIP044)
HIP044 <- RenameIdents(HIP044, new.cluster.ids)
levels(HIP044)
DimPlot(HIP044, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes

saveRDS(HIP044, file = "~/Desktop/Covid analysis/VC start/HIP044.rds")

table(Idents(HIP044))

HIP044_monocytes <- subset(HIP044, idents = c("CD16 monocytes"))
table(Idents(HIP044_monocytes))

Idents(object = HIP044_monocytes) <- "Healthy PBMC Sample 5 monocytes"
HIP044_monocytes$sample <- "Healthy PBMC Sample 5"
HIP044_monocytes$stim <- "Uninfected PBMC"
levels(HIP044_monocytes)
HIP044_monocytes
saveRDS(HIP044_monocytes, file = "~/Desktop/Covid analysis/VC start/HIP044_monocytes.rds")

###HIP045
HIP045 <- readRDS(file = "~/Desktop/Covid analysis/GSE150728_RAW/GSM4557339_HIP045_cell.counts.matrices.rds")
HIP045 <- CreateSeuratObject(counts = HIP045$exon, project = "uninf pbmc", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
HIP045
Idents(object = HIP045) <- "Healthy PBMC Sample 6"
HIP045$sample <- "Healthy PBMC Sample 6"
HIP045$stim <- "Uninfected PBMC"
levels(HIP045)
HIP045

HIP045[["percent.mt"]] <- PercentageFeatureSet(HIP045, pattern = "^MT-")

VlnPlot(HIP045, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1.HIP045 <- FeatureScatter(HIP045, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2.HIP045 <- FeatureScatter(HIP045, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1.HIP045, plot2.HIP045))
HIP045<- subset(HIP045, subset = nFeature_RNA > 200 & nFeature_RNA < 2000 & percent.mt < 20)

##you can do sct transform 
HIP045 <- SCTransform(HIP045, vars.to.regress = "percent.mt", verbose = FALSE)
HIP045

Idents(object = HIP045) <- "sample"
levels(HIP045)


###linear dimensional reduction
HIP045<- RunPCA(HIP045, features = VariableFeatures(object = HIP045))
DimPlot(HIP045, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(HIP045, ndims = 100)
# Determine percent of variation associated with each PC
pct <- HIP045[["pca"]]@stdev / sum(HIP045[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
# Determine which PC exhibits cumulative percent greater than 90% and % variation associated with the PC as less than 5
co1 <- which(cumu > 90 & pct < 5)[1]
co1

co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
# last point where change of % of variation is more than 0.1%.
co2
#Now, to determine the selection of PCs, we will use the minimum of the two metrics:
# Minimum of the two calculation
pcs <- min(co1, co2)
pcs

###clustering
HIP045<- FindNeighbors(HIP045, dims = 1:8)
HIP045 <- FindClusters(HIP045, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
HIP045 <- RunUMAP(HIP045, dims = 1:8)
DimPlot(HIP045, reduction = "umap", min.dist = 0.75, pt.size =0.2, label = TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
HIP045.markers <- FindAllMarkers(HIP045, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
HIP045.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


#check for mature monocyte markers and plot to see which cluster has highest level of expression
VlnPlot(HIP045, features = c("FCGR3B", "CXCR2", "FCGR3A","MS4A7"))
# you can plot raw counts as well
VlnPlot(HIP045, features = c("FCGR3B", "CXCR2","FCGR3A","MS4A7"), slot = "counts", log = TRUE)
FeaturePlot(HIP045, features = c("FCGR3B", "CXCR2","FCGR3A"))

levels(HIP045)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5", "6","7", "8","CD16 monocytes","10")
names(new.cluster.ids) <- levels(HIP045)
HIP045 <- RenameIdents(HIP045, new.cluster.ids)
levels(HIP045)
DimPlot(HIP045, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes

saveRDS(HIP045, file = "~/Desktop/Covid analysis/VC start/HIP045.rds")

table(Idents(HIP045)) #very few monocytes

HIP045_monocytes <- subset(HIP045, idents = c("CD16 monocytes"))
table(Idents(HIP045_monocytes))

Idents(object = HIP045_monocytes) <- "Healthy PBMC Sample 6 monocytes"
HIP045_monocytes$sample <- "Healthy PBMC Sample 6"
HIP045_monocytes$stim <- "Uninfected PBMC"
levels(HIP045_monocytes)
HIP045_monocytes
saveRDS(HIP045_monocytes, file = "~/Desktop/Covid analysis/VC start/HIP045_monocytes.rds")

###isolate monocytes from healthy donors from 10x 5kpbmc sample.

PBMC_data <- Read10X(data.dir = "/Users/bermanlab/Desktop/Covid analysis/PM folder/10x healthy donor/filtered_feature_bc_matrix")
PBMC_10x <- CreateSeuratObject(counts = PBMC_data, project = "covid", min.cells = 3, min.features = 200, names.field = 2, names.delim = "-")
PBMC_10x
Idents(object = PBMC_10x) <- "Uninfected PBMC 10x"
PBMC_10x$sample <- "Uninfected PBMC 10x"
PBMC_10x$stim <- "Uninfected"
levels(PBMC_10x)
PBMC_10x

PBMC_10x[["percent.mt"]] <- PercentageFeatureSet(PBMC_10x, pattern = "^MT-")

#VlnPlot(PBMC_10x, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
#plot1.PBMC_10x <- FeatureScatter(PBMC_10x, feature1 = "nCount_RNA", feature2 = "percent.mt")
#plot2.PBMC_10x <- FeatureScatter(PBMC_10x, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
#CombinePlots(plots = list(plot1.PBMC_10x, plot2.PBMC_10x))
#PBMC_10x<- subset(PBMC_10x, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 20)

##you can do sct transform 
PBMC_10x <- SCTransform(PBMC_10x, vars.to.regress = "percent.mt", verbose = FALSE)
PBMC_10x

Idents(object = PBMC_10x) <- "sample"
levels(PBMC_10x)


###linear dimensional reduction
PBMC_10x<- RunPCA(PBMC_10x, features = VariableFeatures(object = PBMC_10x))
DimPlot(PBMC_10x, reduction = "pca")

####determine dimentionality of your data
ElbowPlot(PBMC_10x, ndims = 100)
# Determine percent of variation associated with each PC
pct <- PBMC_10x[["pca"]]@stdev / sum(PBMC_10x[["pca"]]@stdev) * 100
# Calculate cumulative percents for each PC
cumu <- cumsum(pct)
# Determine which PC exhibits cumulative percent greater than 90% and % variation associated with the PC as less than 5
co1 <- which(cumu > 90 & pct < 5)[1]
co1

co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
# last point where change of % of variation is more than 0.1%.
co2
#Now, to determine the selection of PCs, we will use the minimum of the two metrics:
# Minimum of the two calculation
pcs <- min(co1, co2)
pcs

###clustering
PBMC_10x <- FindNeighbors(PBMC_10x, dims = 1:11)
PBMC_10x <- FindClusters(PBMC_10x, resolution = 0.5)

###Run non-linear dimensional reduction (UMAP/tSNE)
PBMC_10x <- RunUMAP(PBMC_10x, dims = 1:11)
DimPlot(PBMC_10x, reduction = "umap", min.dist = 0.75, pt.size =0.2, label = TRUE)


# find markers for every cluster compared to all remaining cells, report only the positive ones
PBMC_10x.markers <- FindAllMarkers(PBMC_10x, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
PBMC_10x.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)


#check for mature monocyte markers and plot to see which cluster has highest level of expression
VlnPlot(PBMC_10x, features = c( "FCGR3A","MS4A7", "CD14", "CXCR2", "FCGR3B"))
# you can plot raw counts as well
VlnPlot(PBMC_10x, features = c( "FCGR3A","MS4A7", "CD14", "CXCR2", "FCGR3B"), slot = "counts", log = TRUE)
FeaturePlot(PBMC_10x, features = c("CD14","FCGR3A"))

levels(PBMC_10x)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5","6", "7", "8","9","10", "11", "12", "CD16 monocytes", "14", "15", "16", "17", "CD16monocytes")
names(new.cluster.ids) <- levels(PBMC_10x)
PBMC_10x <- RenameIdents(PBMC_10x, new.cluster.ids)
levels(PBMC_10x)
DimPlot(PBMC_10x, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes

saveRDS(PBMC_10x, file = "~/Desktop/Covid analysis/VC start/PBMC_10x.rds")

table(Idents(PBMC_10x))


monocytes_PBMC_10x <- subset(PBMC_10x, idents = c("CD16 monocytes","CD16monocytes"))
table(Idents(monocytes_PBMC_10x))

Idents(object = monocytes_PBMC_10x) <- "Uninfected monocytes PBMC 10x"
monocytes_PBMC_10x$sample <- "Uninfected monocytes PBMC 10x"
monocytes_PBMC_10x$stim <- "Uninfected"
levels(monocytes_PBMC_10x)
monocytes_PBMC_10x
saveRDS(monocytes_PBMC_10x, file = "~/Desktop/Covid analysis/VC start/monocytes_PBMC_10x.rds")

##subset each donor to an independent seurat object
####healthy donors from Reyes et al., Nat Med, 2020
P18F <- subset(data, idents = c("P18F"))
P17H <- subset(data, idents = c("P17H"))
P20H <- subset(data, idents = c("P20H"))
P15F <- subset(data, idents = c("P15F"))
P08H <- subset(data, idents = c("P08H"))
P13H <- subset(data, idents = c("P13H"))
P07H <- subset(data, idents = c("P07H"))
P06F <- subset(data, idents = c("P06F"))
P04H <- subset(data, idents = c("P04H"))
C2P01H <- subset(data, idents = c("C2P01H"))
P09H  <- subset(data, idents = c("P09H"))
P02H <- subset(data, idents = c("P02H"))
C2P05F <- subset(data, idents = c("C2P05F"))
C2P07H <- subset(data, idents = c("C2P07H"))
C2P13F <- subset(data, idents = c("C2P13F"))
C2P16H <- subset(data, idents = c("C2P16H"))
C2P10H <- subset(data, idents = c("C2P10H"))
C2P19H <- subset(data, idents = c("C2P19H"))
C2P15H <- subset(data, idents = c("C2P15H"))

###covid patients from berlin
C19_CB_0001 <- subset(data, idents = c("C19-CB-0001"))
C19_CB_0002 <- subset(data, idents = c("C19-CB-0002"))
C19_CB_0003 <- subset(data, idents = c("C19-CB-0003"))
C19_CB_0005 <- subset(data, idents = c("C19-CB-0005"))
C19_CB_0052 <- subset(data, idents = c("C19-CB-0052"))
C19_CB_0053 <- subset(data, idents = c("C19-CB-0053"))
C19_CB_0204 <- subset(data, idents = c("C19-CB-0204"))
C19_CB_0214 <- subset(data, idents = c("C19-CB-0214"))

C19_CB_0008 <- subset(data, idents = c("C19-CB-0008"))
C19_CB_0009 <- subset(data, idents = c("C19-CB-0009"))
C19_CB_0011 <- subset(data, idents = c("C19-CB-0011"))
C19_CB_0012 <- subset(data, idents = c("C19-CB-0012"))
C19_CB_0013 <- subset(data, idents = c("C19-CB-0013"))
C19_CB_0016 <- subset(data, idents = c("C19-CB-0016"))
C19_CB_0020 <- subset(data, idents = c("C19-CB-0020"))
C19_CB_0021 <- subset(data, idents = c("C19-CB-0021"))
C19_CB_0198 <- subset(data, idents = c("C19-CB-0198"))
C19_CB_0199 <- subset(data, idents = c("C19-CB-0199"))

rm(data)

####P18F
P18F
Idents(object = P18F) <- "Control 1"
P18F$sample <- "Control 1"
P18F$stim <- "Healthy"
levels(P18F)
P18F
P18F[["percent.mt"]] <- PercentageFeatureSet(P18F, pattern = "^MT-")
VlnPlot(P18F, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
P18F<- subset(P18F, subset = nFeature_RNA > 200 & nFeature_RNA < 1500 & percent.mt < 5)
P18F
P18F <- PercentageFeatureSet(P18F, pattern = "^MT-", col.name = "percent.mt")
P18F <- SCTransform(P18F, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = P18F) <- "sample"
levels(P18F)
P18F<- RunPCA(P18F, features = VariableFeatures(object = P18F))
DimPlot(P18F, reduction = "pca")
ElbowPlot(P18F, ndims = 100)
pct <- P18F[["pca"]]@stdev / sum(P18F[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
P18F <- FindNeighbors(P18F, dims = 1:9)
P18F <- FindClusters(P18F, resolution = 0.5)
P18F <- RunUMAP(P18F, dims = 1:9)
DimPlot(P18F, reduction = "umap", pt.size =0.2, label = TRUE)
P18F.markers <- FindAllMarkers(P18F, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
P18F.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(P18F, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(P18F, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(P18F, features = c("FCGR3A","MS4A7", "CD14"))
levels(P18F)
new.cluster.ids <- c("0", "1", "2", "3", "CD16 monocytes", "5", "6", "7")
names(new.cluster.ids) <- levels(P18F)
P18F <- RenameIdents(P18F, new.cluster.ids)
levels(P18F)
DimPlot(P18F, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(P18F, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P18F.rds")
P18F <- readRDS(file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P18F.rds")
table(Idents(P18F)) 

#subset  and save CD16 monocytes 
P18F_monocytes <- subset(P18F, idents = c("CD16 monocytes"))
table(Idents(P18F_monocytes))

Idents(object = P18F_monocytes) <- "Control 1 monocytes"
P18F_monocytes$sample <- "Control 1"
P18F_monocytes$stim <- "Healthy"
levels(P18F_monocytes)
P18F_monocytes
saveRDS(P18F_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P18F_monocytes.rds")
rm(P18F, P18F_monocytes,P18F.markers)

####P17H
P17H
Idents(object = P17H) <- "Control 2"
P17H$sample <- "Control 2"
P17H$stim <- "Healthy"
levels(P17H)
P17H
P17H[["percent.mt"]] <- PercentageFeatureSet(P17H, pattern = "^MT-")
VlnPlot(P17H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
P17H<- subset(P17H, subset = nFeature_RNA > 200 & nFeature_RNA < 1500 & percent.mt < 4)
P17H
P17H <- PercentageFeatureSet(P17H, pattern = "^MT-", col.name = "percent.mt")
P17H <- SCTransform(P17H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = P17H) <- "sample"
levels(P17H)
P17H<- RunPCA(P17H, features = VariableFeatures(object = P17H))
DimPlot(P17H, reduction = "pca")
ElbowPlot(P17H, ndims = 100)
pct <- P17H[["pca"]]@stdev / sum(P17H[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
P17H <- FindNeighbors(P17H, dims = 1:8)
P17H <- FindClusters(P17H, resolution = 0.5)
P17H <- RunUMAP(P17H, dims = 1:8)
DimPlot(P17H, reduction = "umap", pt.size =0.2, label = TRUE)
P17H.markers <- FindAllMarkers(P17H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
P17H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(P17H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(P17H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(P17H, features = c("FCGR3A","MS4A7", "CD14"))
levels(P17H)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5", "6", "CD16 monocytes","8")
names(new.cluster.ids) <- levels(P17H)
P17H <- RenameIdents(P17H, new.cluster.ids)
levels(P17H)
DimPlot(P17H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(P17H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P17H.rds")
table(Idents(P17H)) 

#subset  and save CD16 monocytes 
P17H_monocytes <- subset(P17H, idents = c("CD16 monocytes"))
table(Idents(P17H_monocytes))

Idents(object = P17H_monocytes) <- "Control 2 monocytes"
P17H_monocytes$sample <- "Control 2"
P17H_monocytes$stim <- "Healthy"
levels(P17H_monocytes)
P17H_monocytes
saveRDS(P17H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P17H_monocytes.rds")
rm(P17H, P17H_monocytes,P17H.markers)

####P20H
P20H
Idents(object = P20H) <- "Control 3"
P20H$sample <- "Control 3"
P20H$stim <- "Healthy"
levels(P20H)
P20H
P20H[["percent.mt"]] <- PercentageFeatureSet(P20H, pattern = "^MT-")
VlnPlot(P20H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
P20H<- subset(P20H, subset = nFeature_RNA > 200 & nFeature_RNA < 1500 & percent.mt < 3)
P20H
P20H <- PercentageFeatureSet(P20H, pattern = "^MT-", col.name = "percent.mt")
P20H <- SCTransform(P20H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = P20H) <- "sample"
levels(P20H)
P20H<- RunPCA(P20H, features = VariableFeatures(object = P20H))
DimPlot(P20H, reduction = "pca")
ElbowPlot(P20H, ndims = 100)
pct <- P20H[["pca"]]@stdev / sum(P20H[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
P20H <- FindNeighbors(P20H, dims = 1:7)
P20H <- FindClusters(P20H, resolution = 0.5)
P20H <- RunUMAP(P20H, dims = 1:7)
DimPlot(P20H, reduction = "umap", pt.size =0.2, label = TRUE)
P20H.markers <- FindAllMarkers(P20H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
P20H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(P20H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(P20H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(P20H, features = c("FCGR3A","MS4A7", "CD14"))
levels(P20H)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5", "CD16 monocytes")
names(new.cluster.ids) <- levels(P20H)
P20H <- RenameIdents(P20H, new.cluster.ids)
levels(P20H)
DimPlot(P20H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(P20H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P20H.rds")
table(Idents(P20H)) 
###only 42 cells but proceed 

#subset  and save CD16 monocytes 
P20H_monocytes <- subset(P20H, idents = c("CD16 monocytes"))
table(Idents(P20H_monocytes))

Idents(object = P20H_monocytes) <- "Control 3 monocytes"
P20H_monocytes$sample <- "Control 3"
P20H_monocytes$stim <- "Healthy"
levels(P20H_monocytes)
P20H_monocytes
saveRDS(P20H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P20H_monocytes.rds")
rm(P20H, P20H_monocytes,P20H.markers)

####P15F
P15F
Idents(object = P15F) <- "Control 4"
P15F$sample <- "Control 4"
P15F$stim <- "Healthy"
levels(P15F)
P15F
P15F[["percent.mt"]] <- PercentageFeatureSet(P15F, pattern = "^MT-")
VlnPlot(P15F, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
P15F<- subset(P15F, subset = nFeature_RNA > 200 & nFeature_RNA < 1000 & percent.mt < 3)
P15F
P15F <- PercentageFeatureSet(P15F, pattern = "^MT-", col.name = "percent.mt")
P15F <- SCTransform(P15F, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = P15F) <- "sample"
levels(P15F)
P15F<- RunPCA(P15F, features = VariableFeatures(object = P15F))
DimPlot(P15F, reduction = "pca")
ElbowPlot(P15F, ndims = 100)
pct <- P15F[["pca"]]@stdev / sum(P15F[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
P15F <- FindNeighbors(P15F, dims = 1:6)
P15F <- FindClusters(P15F, resolution = 0.5)
P15F <- RunUMAP(P15F, dims = 1:6)
DimPlot(P15F, reduction = "umap", pt.size =0.2, label = TRUE)
P15F.markers <- FindAllMarkers(P15F, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
P15F.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(P15F, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(P15F, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(P15F, features = c("FCGR3A","MS4A7", "CD14"))
levels(P15F)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5", "CD16 monocytes")
names(new.cluster.ids) <- levels(P15F)
P15F <- RenameIdents(P15F, new.cluster.ids)
levels(P15F)
DimPlot(P15F, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(P15F, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P15F.rds")
table(Idents(P15F)) 


#subset  and save CD16 monocytes 
P15F_monocytes <- subset(P15F, idents = c("CD16 monocytes"))
table(Idents(P15F_monocytes))

Idents(object = P15F_monocytes) <- "Control 4 monocytes"
P15F_monocytes$sample <- "Control 4"
P15F_monocytes$stim <- "Healthy"
levels(P15F_monocytes)
P15F_monocytes
saveRDS(P15F_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P15F_monocytes.rds")
rm(P15F, P15F_monocytes,P15F.markers)

####P08H
P08H
Idents(object = P08H) <- "Control 5"
P08H$sample <- "Control 5"
P08H$stim <- "Healthy"
levels(P08H)
P08H
P08H[["percent.mt"]] <- PercentageFeatureSet(P08H, pattern = "^MT-")
VlnPlot(P08H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
P08H<- subset(P08H, subset = nFeature_RNA > 200 & nFeature_RNA < 1000 & percent.mt < 3)
P08H
P08H <- PercentageFeatureSet(P08H, pattern = "^MT-", col.name = "percent.mt")
P08H <- SCTransform(P08H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = P08H) <- "sample"
levels(P08H)
P08H<- RunPCA(P08H, features = VariableFeatures(object = P08H))
DimPlot(P08H, reduction = "pca")
ElbowPlot(P08H, ndims = 100)
pct <- P08H[["pca"]]@stdev / sum(P08H[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
P08H <- FindNeighbors(P08H, dims = 1:6)
P08H <- FindClusters(P08H, resolution = 0.5)
P08H <- RunUMAP(P08H, dims = 1:6)
DimPlot(P08H, reduction = "umap", pt.size =0.2, label = TRUE)
P08H.markers <- FindAllMarkers(P08H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
P08H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(P08H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(P08H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(P08H, features = c("FCGR3A","MS4A7", "CD14"))
levels(P08H)
new.cluster.ids <- c("0", "1", "2", "3", "CD16 monocytes", "5")
names(new.cluster.ids) <- levels(P08H)
P08H <- RenameIdents(P08H, new.cluster.ids)
levels(P08H)
DimPlot(P08H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(P08H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P08H.rds")
table(Idents(P08H)) 


#subset  and save CD16 monocytes 
P08H_monocytes <- subset(P08H, idents = c("CD16 monocytes"))
table(Idents(P08H_monocytes))

Idents(object = P08H_monocytes) <- "Control 5 monocytes"
P08H_monocytes$sample <- "Control 5"
P08H_monocytes$stim <- "Healthy"
levels(P08H_monocytes)
P08H_monocytes
saveRDS(P08H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P08H_monocytes.rds")
rm(P08H, P08H_monocytes,P08H.markers)

####P13H
P13H
Idents(object = P13H) <- "Control 6"
P13H$sample <- "Control 6"
P13H$stim <- "Healthy"
levels(P13H)
P13H
P13H[["percent.mt"]] <- PercentageFeatureSet(P13H, pattern = "^MT-")
VlnPlot(P13H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
P13H<- subset(P13H, subset = nFeature_RNA > 200 & nFeature_RNA < 1000 & percent.mt < 3)
P13H
P13H <- PercentageFeatureSet(P13H, pattern = "^MT-", col.name = "percent.mt")
P13H <- SCTransform(P13H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = P13H) <- "sample"
levels(P13H)
P13H<- RunPCA(P13H, features = VariableFeatures(object = P13H))
DimPlot(P13H, reduction = "pca")
ElbowPlot(P13H, ndims = 100)
pct <- P13H[["pca"]]@stdev / sum(P13H[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
P13H <- FindNeighbors(P13H, dims = 1:7)
P13H <- FindClusters(P13H, resolution = 0.5)
P13H <- RunUMAP(P13H, dims = 1:7)
DimPlot(P13H, reduction = "umap", pt.size =0.2, label = TRUE)
P13H.markers <- FindAllMarkers(P13H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
P13H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(P13H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(P13H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(P13H, features = c("FCGR3A","MS4A7", "CD14"))
levels(P13H)
new.cluster.ids <- c("0", "1", "2", "3", "CD16 monocytes", "5","6")
names(new.cluster.ids) <- levels(P13H)
P13H <- RenameIdents(P13H, new.cluster.ids)
levels(P13H)
DimPlot(P13H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(P13H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P13H.rds")
table(Idents(P13H)) 

#subset  and save CD16 monocytes 
P13H_monocytes <- subset(P13H, idents = c("CD16 monocytes"))
table(Idents(P13H_monocytes))

Idents(object = P13H_monocytes) <- "Control 6 monocytes"
P13H_monocytes$sample <- "Control 6"
P13H_monocytes$stim <- "Healthy"
levels(P13H_monocytes)
P13H_monocytes
saveRDS(P13H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P13H_monocytes.rds")
rm(P13H, P13H_monocytes,P13H.markers)

####P07H
P07H
Idents(object = P07H) <- "Control 7"
P07H$sample <- "Control 7"
P07H$stim <- "Healthy"
levels(P07H)
P07H
P07H[["percent.mt"]] <- PercentageFeatureSet(P07H, pattern = "^MT-")
VlnPlot(P07H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
P07H<- subset(P07H, subset = nFeature_RNA > 200 & nFeature_RNA < 1000 & percent.mt < 3)
P07H
P07H <- PercentageFeatureSet(P07H, pattern = "^MT-", col.name = "percent.mt")
P07H <- SCTransform(P07H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = P07H) <- "sample"
levels(P07H)
P07H<- RunPCA(P07H, features = VariableFeatures(object = P07H))
DimPlot(P07H, reduction = "pca")
ElbowPlot(P07H, ndims = 100)
pct <- P07H[["pca"]]@stdev / sum(P07H[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
P07H <- FindNeighbors(P07H, dims = 1:6)
P07H <- FindClusters(P07H, resolution = 0.5)
P07H <- RunUMAP(P07H, dims = 1:6)
DimPlot(P07H, reduction = "umap", pt.size =0.2, label = TRUE)
P07H.markers <- FindAllMarkers(P07H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
P07H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(P07H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(P07H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(P07H, features = c("FCGR3A","MS4A7", "CD14"))
levels(P07H)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5","CD16 monocytes")
names(new.cluster.ids) <- levels(P07H)
P07H <- RenameIdents(P07H, new.cluster.ids)
levels(P07H)
DimPlot(P07H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(P07H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P07H.rds")
table(Idents(P07H)) 

#subset  and save CD16 monocytes 
P07H_monocytes <- subset(P07H, idents = c("CD16 monocytes"))
table(Idents(P07H_monocytes))

Idents(object = P07H_monocytes) <- "Control 7 monocytes"
P07H_monocytes$sample <- "Control 7"
P07H_monocytes$stim <- "Healthy"
levels(P07H_monocytes)
P07H_monocytes
saveRDS(P07H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P07H_monocytes.rds")
rm(P07H, P07H_monocytes,P07H.markers)

####P06F
P06F
Idents(object = P06F) <- "Control 8"
P06F$sample <- "Control 8"
P06F$stim <- "Healthy"
levels(P06F)
P06F
P06F[["percent.mt"]] <- PercentageFeatureSet(P06F, pattern = "^MT-")
VlnPlot(P06F, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
P06F<- subset(P06F, subset = nFeature_RNA > 200 & nFeature_RNA < 800 & percent.mt < 3)
P06F
P06F <- PercentageFeatureSet(P06F, pattern = "^MT-", col.name = "percent.mt")
P06F <- SCTransform(P06F, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = P06F) <- "sample"
levels(P06F)
P06F<- RunPCA(P06F, features = VariableFeatures(object = P06F))
DimPlot(P06F, reduction = "pca")
ElbowPlot(P06F, ndims = 100)
pct <- P06F[["pca"]]@stdev / sum(P06F[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
P06F <- FindNeighbors(P06F, dims = 1:6)
P06F <- FindClusters(P06F, resolution = 0.5)
P06F <- RunUMAP(P06F, dims = 1:6)
DimPlot(P06F, reduction = "umap", pt.size =0.2, label = TRUE)
P06F.markers <- FindAllMarkers(P06F, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
P06F.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(P06F, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(P06F, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(P06F, features = c("FCGR3A","MS4A7", "CD14"))
levels(P06F)
new.cluster.ids <- c("0", "1", "2", "3", "CD16 monocytes", "5","6")
names(new.cluster.ids) <- levels(P06F)
P06F <- RenameIdents(P06F, new.cluster.ids)
levels(P06F)
DimPlot(P06F, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(P06F, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P06F.rds")
table(Idents(P06F)) 

#subset  and save CD16 monocytes 
P06F_monocytes <- subset(P06F, idents = c("CD16 monocytes"))
table(Idents(P06F_monocytes))

Idents(object = P06F_monocytes) <- "Control 8 monocytes"
P06F_monocytes$sample <- "Control 8"
P06F_monocytes$stim <- "Healthy"
levels(P06F_monocytes)
P06F_monocytes
saveRDS(P06F_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P06F_monocytes.rds")
rm(P06F, P06F_monocytes,P06F.markers)

####P04H
P04H
Idents(object = P04H) <- "Control 9"
P04H$sample <- "Control 9"
P04H$stim <- "Healthy"
levels(P04H)
P04H
P04H[["percent.mt"]] <- PercentageFeatureSet(P04H, pattern = "^MT-")
VlnPlot(P04H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
P04H<- subset(P04H, subset = nFeature_RNA > 200 & nFeature_RNA < 800 & percent.mt < 3)
P04H
P04H <- PercentageFeatureSet(P04H, pattern = "^MT-", col.name = "percent.mt")
P04H <- SCTransform(P04H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = P04H) <- "sample"
levels(P04H)
P04H<- RunPCA(P04H, features = VariableFeatures(object = P04H))
DimPlot(P04H, reduction = "pca")
ElbowPlot(P04H, ndims = 100)
pct <- P04H[["pca"]]@stdev / sum(P04H[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
P04H <- FindNeighbors(P04H, dims = 1:7)
P04H <- FindClusters(P04H, resolution = 0.5)
P04H <- RunUMAP(P04H, dims = 1:7)
DimPlot(P04H, reduction = "umap", pt.size =0.2, label = TRUE)
P04H.markers <- FindAllMarkers(P04H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
P04H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(P04H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(P04H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(P04H, features = c("FCGR3A","MS4A7", "CD14"))
levels(P04H)
new.cluster.ids <- c("0", "1", "2", "3","4", "CD16 monocytes","6")
names(new.cluster.ids) <- levels(P04H)
P04H <- RenameIdents(P04H, new.cluster.ids)
levels(P04H)
DimPlot(P04H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(P04H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P04H.rds")
table(Idents(P04H)) 

#subset  and save CD16 monocytes 
P04H_monocytes <- subset(P04H, idents = c("CD16 monocytes"))
table(Idents(P04H_monocytes))

Idents(object = P04H_monocytes) <- "Control 9 monocytes"
P04H_monocytes$sample <- "Control 9"
P04H_monocytes$stim <- "Healthy"
levels(P04H_monocytes)
P04H_monocytes
saveRDS(P04H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P04H_monocytes.rds")
rm(P04H, P04H_monocytes,P04H.markers)

####C2P01H
C2P01H
Idents(object = C2P01H) <- "Control 10"
C2P01H$sample <- "Control 10"
C2P01H$stim <- "Healthy"
levels(C2P01H)
C2P01H
C2P01H[["percent.mt"]] <- PercentageFeatureSet(C2P01H, pattern = "^MT-")
VlnPlot(C2P01H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C2P01H<- subset(C2P01H, subset = nFeature_RNA > 200 & nFeature_RNA < 1500 & percent.mt < 6)
C2P01H
C2P01H <- PercentageFeatureSet(C2P01H, pattern = "^MT-", col.name = "percent.mt")
C2P01H <- SCTransform(C2P01H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C2P01H) <- "sample"
levels(C2P01H)
C2P01H<- RunPCA(C2P01H, features = VariableFeatures(object = C2P01H))
DimPlot(C2P01H, reduction = "pca")
ElbowPlot(C2P01H, ndims = 100)
pct <- C2P01H[["pca"]]@stdev / sum(C2P01H[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C2P01H <- FindNeighbors(C2P01H, dims = 1:10)
C2P01H <- FindClusters(C2P01H, resolution = 0.5)
C2P01H <- RunUMAP(C2P01H, dims = 1:10)
DimPlot(C2P01H, reduction = "umap", pt.size =0.2, label = TRUE)
C2P01H.markers <- FindAllMarkers(C2P01H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C2P01H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C2P01H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C2P01H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C2P01H, features = c("FCGR3A","MS4A7", "CD14"))
levels(C2P01H)
new.cluster.ids <- c("0", "1", "2", "3","4", "CD16 monocytes","6")
names(new.cluster.ids) <- levels(C2P01H)
C2P01H <- RenameIdents(C2P01H, new.cluster.ids)
levels(C2P01H)
DimPlot(C2P01H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C2P01H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P01H.rds")
table(Idents(C2P01H)) 

#subset  and save CD16 monocytes 
C2P01H_monocytes <- subset(C2P01H, idents = c("CD16 monocytes"))
table(Idents(C2P01H_monocytes))

Idents(object = C2P01H_monocytes) <- "Control 10 monocytes"
C2P01H_monocytes$sample <- "Control 10"
C2P01H_monocytes$stim <- "Healthy"
levels(C2P01H_monocytes)
C2P01H_monocytes
saveRDS(C2P01H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P01H_monocytes.rds")
rm(C2P01H, C2P01H_monocytes,C2P01H.markers)

####P09H
P09H
Idents(object = P09H) <- "Control 11"
P09H$sample <- "Control 11"
P09H$stim <- "Healthy"
levels(P09H)
P09H
P09H[["percent.mt"]] <- PercentageFeatureSet(P09H, pattern = "^MT-")
VlnPlot(P09H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
P09H<- subset(P09H, subset = nFeature_RNA > 200 & nFeature_RNA < 2000 & percent.mt < 5)
P09H
P09H <- PercentageFeatureSet(P09H, pattern = "^MT-", col.name = "percent.mt")
P09H <- SCTransform(P09H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = P09H) <- "sample"
levels(P09H)
P09H<- RunPCA(P09H, features = VariableFeatures(object = P09H))
DimPlot(P09H, reduction = "pca")
ElbowPlot(P09H, ndims = 100)
pct <- P09H[["pca"]]@stdev / sum(P09H[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
P09H <- FindNeighbors(P09H, dims = 1:9)
P09H <- FindClusters(P09H, resolution = 0.5)
P09H <- RunUMAP(P09H, dims = 1:9)
DimPlot(P09H, reduction = "umap", pt.size =0.2, label = TRUE)
P09H.markers <- FindAllMarkers(P09H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
P09H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(P09H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(P09H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(P09H, features = c("FCGR3A","MS4A7", "CD14"))
levels(P09H)
new.cluster.ids <- c("0", "1", "2", "3","4","5","6","7","8", "CD16 monocytes")
names(new.cluster.ids) <- levels(P09H)
P09H <- RenameIdents(P09H, new.cluster.ids)
levels(P09H)
DimPlot(P09H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(P09H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P09H.rds")
table(Idents(P09H)) 

#subset  and save CD16 monocytes 
P09H_monocytes <- subset(P09H, idents = c("CD16 monocytes"))
table(Idents(P09H_monocytes))

Idents(object = P09H_monocytes) <- "Control 11 monocytes"
P09H_monocytes$sample <- "Control 11"
P09H_monocytes$stim <- "Healthy"
levels(P09H_monocytes)
P09H_monocytes
saveRDS(P09H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P09H_monocytes.rds")
rm(P09H,P09H_monocytes,P09H.markers)

####P02H
P02H
Idents(object = P02H) <- "Control 12"
P02H$sample <- "Control 12"
P02H$stim <- "Healthy"
levels(P02H)
P02H
P02H[["percent.mt"]] <- PercentageFeatureSet(P02H, pattern = "^MT-")
VlnPlot(P02H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
P02H<- subset(P02H, subset = nFeature_RNA > 200 & nFeature_RNA < 2000 & percent.mt < 5)
P02H
P02H <- PercentageFeatureSet(P02H, pattern = "^MT-", col.name = "percent.mt")
P02H <- SCTransform(P02H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = P02H) <- "sample"
levels(P02H)
P02H<- RunPCA(P02H, features = VariableFeatures(object = P02H))
DimPlot(P02H, reduction = "pca")
ElbowPlot(P02H, ndims = 100)
pct <- P09H[["pca"]]@stdev / sum(P09H[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
P02H <- FindNeighbors(P02H, dims = 1:9)
P02H <- FindClusters(P02H, resolution = 0.5)
P02H <- RunUMAP(P02H, dims = 1:9)
DimPlot(P02H, reduction = "umap", pt.size =0.2, label = TRUE)
P02H.markers <- FindAllMarkers(P02H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
P02H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(P02H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(P02H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(P02H, features = c("FCGR3A","MS4A7", "CD14"))
levels(P02H)
new.cluster.ids <- c("0", "1", "2", "3","CD16 monocytes","5","6", "CD16 monocytes","8")
names(new.cluster.ids) <- levels(P02H)
P02H <- RenameIdents(P02H, new.cluster.ids)
levels(P02H)
DimPlot(P02H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(P02H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P02H.rds")
table(Idents(P02H)) 

#subset  and save CD16 monocytes 
P02H_monocytes <- subset(P02H, idents = c("CD16 monocytes"))
table(Idents(P02H_monocytes))

Idents(object = P02H_monocytes) <- "Control 12 monocytes"
P02H_monocytes$sample <- "Control 12"
P02H_monocytes$stim <- "Healthy"
levels(P02H_monocytes)
P02H_monocytes
saveRDS(P02H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/P02H_monocytes.rds")
rm(P02H,P02H_monocytes,P02H.markers)

####P02H
C2P05F
Idents(object = C2P05F) <- "Control 13"
C2P05F$sample <- "Control 13"
C2P05F$stim <- "Healthy"
levels(C2P05F)
C2P05F
C2P05F[["percent.mt"]] <- PercentageFeatureSet(C2P05F, pattern = "^MT-")
VlnPlot(C2P05F, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C2P05F<- subset(C2P05F, subset = nFeature_RNA > 200 & nFeature_RNA < 2000 & percent.mt < 7)
C2P05F
C2P05F <- PercentageFeatureSet(C2P05F, pattern = "^MT-", col.name = "percent.mt")
C2P05F <- SCTransform(C2P05F, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C2P05F) <- "sample"
levels(C2P05F)
C2P05F<- RunPCA(C2P05F, features = VariableFeatures(object = C2P05F))
DimPlot(C2P05F, reduction = "pca")
ElbowPlot(C2P05F, ndims = 100)
pct <- C2P05F[["pca"]]@stdev / sum(C2P05F[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C2P05F <- FindNeighbors(C2P05F, dims = 1:10)
C2P05F <- FindClusters(C2P05F, resolution = 0.5)
C2P05F <- RunUMAP(C2P05F, dims = 1:10)
DimPlot(C2P05F, reduction = "umap", pt.size =0.2, label = TRUE)
C2P05F.markers <- FindAllMarkers(C2P05F, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C2P05F.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C2P05F, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C2P05F, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C2P05F, features = c("FCGR3A","MS4A7", "CD14"))
levels(C2P05F)
new.cluster.ids <- c("0", "1", "2", "3","4", "CD16 monocytes","6", "7","8")
names(new.cluster.ids) <- levels(C2P05F)
C2P05F <- RenameIdents(C2P05F, new.cluster.ids)
levels(C2P05F)
DimPlot(C2P05F, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C2P05F, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P05F.rds")
table(Idents(C2P05F)) 

#subset  and save CD16 monocytes 
C2P05F_monocytes <- subset(C2P05F, idents = c("CD16 monocytes"))
table(Idents(C2P05F_monocytes))

Idents(object = C2P05F_monocytes) <- "Control 13 monocytes"
C2P05F_monocytes$sample <- "Control 13"
C2P05F_monocytes$stim <- "Healthy"
levels(C2P05F_monocytes)
C2P05F_monocytes
saveRDS(C2P05F_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P05F_monocytes.rds")
rm(C2P05F,C2P05F_monocytes,C2P05F.markers)

####C2P07H
C2P07H
Idents(object = C2P07H) <- "Control 14"
C2P07H$sample <- "Control 14"
C2P07H$stim <- "Healthy"
levels(C2P07H)
C2P07H
C2P07H[["percent.mt"]] <- PercentageFeatureSet(C2P07H, pattern = "^MT-")
VlnPlot(C2P07H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C2P07H<- subset(C2P07H, subset = nFeature_RNA > 200 & nFeature_RNA < 2000 & percent.mt < 7)
C2P07H
C2P07H <- PercentageFeatureSet(C2P07H, pattern = "^MT-", col.name = "percent.mt")
C2P07H <- SCTransform(C2P07H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C2P07H) <- "sample"
levels(C2P07H)
C2P07H<- RunPCA(C2P07H, features = VariableFeatures(object = C2P07H))
DimPlot(C2P07H, reduction = "pca")
ElbowPlot(C2P07H, ndims = 100)
pct <- C2P07H[["pca"]]@stdev / sum(C2P05F[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C2P07H <- FindNeighbors(C2P07H, dims = 1:10)
C2P07H <- FindClusters(C2P07H, resolution = 0.5)
C2P07H <- RunUMAP(C2P07H, dims = 1:10)
DimPlot(C2P07H, reduction = "umap", pt.size =0.2, label = TRUE)
C2P07H.markers <- FindAllMarkers(C2P07H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C2P07H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C2P07H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C2P07H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C2P07H, features = c("FCGR3A","MS4A7", "CD14"))
levels(C2P07H)
new.cluster.ids <- c("0", "1", "2", "3", "CD16 monocytes","5")
names(new.cluster.ids) <- levels(C2P07H)
C2P07H <- RenameIdents(C2P07H, new.cluster.ids)
levels(C2P07H)
DimPlot(C2P07H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C2P07H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P07H.rds")
table(Idents(C2P07H)) 

#subset  and save CD16 monocytes 
C2P07H_monocytes <- subset(C2P07H, idents = c("CD16 monocytes"))
table(Idents(C2P07H_monocytes))

Idents(object = C2P07H_monocytes) <- "Control 14 monocytes"
C2P07H_monocytes$sample <- "Control 14"
C2P07H_monocytes$stim <- "Healthy"
levels(C2P07H_monocytes)
C2P07H_monocytes
saveRDS(C2P07H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P07H_monocytes.rds")
rm(C2P07H,C2P07H_monocytes,C2P07H.markers)

####C2P13F
C2P13F
Idents(object = C2P13F) <- "Control 15"
C2P13F$sample <- "Control 15"
C2P13F$stim <- "Healthy"
levels(C2P13F)
C2P13F
C2P13F[["percent.mt"]] <- PercentageFeatureSet(C2P13F, pattern = "^MT-")
VlnPlot(C2P13F, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C2P13F<- subset(C2P13F, subset = nFeature_RNA > 200 & nFeature_RNA < 1500 & percent.mt < 7)
C2P13F
C2P13F <- PercentageFeatureSet(C2P13F, pattern = "^MT-", col.name = "percent.mt")
C2P13F <- SCTransform(C2P13F, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C2P13F) <- "sample"
levels(C2P13F)
C2P13F<- RunPCA(C2P13F, features = VariableFeatures(object = C2P13F))
DimPlot(C2P13F, reduction = "pca")
ElbowPlot(C2P13F, ndims = 100)
pct <- C2P13F[["pca"]]@stdev / sum(C2P05F[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C2P13F <- FindNeighbors(C2P13F, dims = 1:10)
C2P13F <- FindClusters(C2P13F, resolution = 0.5)
C2P13F <- RunUMAP(C2P13F, dims = 1:10)
DimPlot(C2P13F, reduction = "umap", pt.size =0.2, label = TRUE)
C2P13F.markers <- FindAllMarkers(C2P13F, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C2P13F.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C2P13F, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C2P13F, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C2P13F, features = c("FCGR3A","MS4A7", "CD14"))
levels(C2P13F)
new.cluster.ids <- c("0", "1", "2", "3","4", "CD16 monocytes","6", "7","8","9")
names(new.cluster.ids) <- levels(C2P13F)
C2P13F <- RenameIdents(C2P13F, new.cluster.ids)
levels(C2P13F)
DimPlot(C2P13F, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C2P13F, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P13F.rds")
table(Idents(C2P13F)) 

#subset  and save CD16 monocytes 
C2P13F_monocytes <- subset(C2P13F, idents = c("CD16 monocytes"))
table(Idents(C2P13F_monocytes))

Idents(object = C2P13F_monocytes) <- "Control 15 monocytes"
C2P13F_monocytes$sample <- "Control 15"
C2P13F_monocytes$stim <- "Healthy"
levels(C2P13F_monocytes)
C2P13F_monocytes
saveRDS(C2P13F_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P13F_monocytes.rds")
rm(C2P13F,C2P13F_monocytes,C2P13F.markers)

####C2P16H
C2P16H
Idents(object = C2P16H) <- "Control 16"
C2P16H$sample <- "Control 16"
C2P16H$stim <- "Healthy"
levels(C2P16H)
C2P16H
C2P16H[["percent.mt"]] <- PercentageFeatureSet(C2P16H, pattern = "^MT-")
VlnPlot(C2P16H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C2P16H<- subset(C2P16H, subset = nFeature_RNA > 200 & nFeature_RNA < 1500 & percent.mt < 5)
C2P16H
C2P16H <- PercentageFeatureSet(C2P16H, pattern = "^MT-", col.name = "percent.mt")
C2P16H <- SCTransform(C2P16H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C2P16H) <- "sample"
levels(C2P16H)
C2P16H<- RunPCA(C2P16H, features = VariableFeatures(object = C2P16H))
DimPlot(C2P16H, reduction = "pca")
ElbowPlot(C2P16H, ndims = 100)
pct <- C2P16H[["pca"]]@stdev / sum(C2P05F[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C2P16H <- FindNeighbors(C2P16H, dims = 1:10)
C2P16H <- FindClusters(C2P16H, resolution = 0.5)
C2P16H <- RunUMAP(C2P16H, dims = 1:10)
DimPlot(C2P16H, reduction = "umap", pt.size =0.2, label = TRUE)
C2P16H.markers <- FindAllMarkers(C2P16H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C2P16H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C2P16H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C2P16H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C2P16H, features = c("FCGR3A","MS4A7", "CD14"))
levels(C2P16H)
new.cluster.ids <- c("0", "1", "2", "3", "CD16 monocytes","5", "6")
names(new.cluster.ids) <- levels(C2P16H)
C2P16H <- RenameIdents(C2P16H, new.cluster.ids)
levels(C2P16H)
DimPlot(C2P16H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C2P16H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P16H.rds")
table(Idents(C2P16H)) 

#subset  and save CD16 monocytes 
C2P16H_monocytes <- subset(C2P16H, idents = c("CD16 monocytes"))
table(Idents(C2P16H_monocytes))

Idents(object = C2P16H_monocytes) <- "Control 16 monocytes"
C2P16H_monocytes$sample <- "Control 16"
C2P16H_monocytes$stim <- "Healthy"
levels(C2P16H_monocytes)
C2P16H_monocytes
saveRDS(C2P16H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P16H_monocytes.rds")
rm(C2P16H,C2P16H_monocytes,C2P16H.markers)

####C2P10H
C2P10H
Idents(object = C2P10H) <- "Control 17"
C2P10H$sample <- "Control 17"
C2P10H$stim <- "Healthy"
levels(C2P10H)
C2P10H
C2P10H[["percent.mt"]] <- PercentageFeatureSet(C2P10H, pattern = "^MT-")
VlnPlot(C2P10H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C2P10H<- subset(C2P10H, subset = nFeature_RNA > 200 & nFeature_RNA < 1500 & percent.mt < 7)
C2P10H
C2P10H <- PercentageFeatureSet(C2P10H, pattern = "^MT-", col.name = "percent.mt")
C2P10H <- SCTransform(C2P10H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C2P10H) <- "sample"
levels(C2P10H)
C2P10H<- RunPCA(C2P10H, features = VariableFeatures(object = C2P10H))
DimPlot(C2P10H, reduction = "pca")
ElbowPlot(C2P10H, ndims = 100)
pct <- C2P10H[["pca"]]@stdev / sum(C2P10H[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C2P10H <- FindNeighbors(C2P10H, dims = 1:9)
C2P10H <- FindClusters(C2P10H, resolution = 0.5)
C2P10H <- RunUMAP(C2P10H, dims = 1:9)
DimPlot(C2P10H, reduction = "umap", pt.size =0.2, label = TRUE)
C2P10H.markers <- FindAllMarkers(C2P10H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C2P10H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C2P10H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C2P10H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C2P10H, features = c("FCGR3A","MS4A7", "CD14"))
levels(C2P10H)
new.cluster.ids <- c("0", "1", "2", "3","4","5","6","CD16 monocytes","8")
names(new.cluster.ids) <- levels(C2P10H)
C2P10H <- RenameIdents(C2P10H, new.cluster.ids)
levels(C2P10H)
DimPlot(C2P10H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C2P10H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P10H.rds")
table(Idents(C2P10H)) 

#subset  and save CD16 monocytes 
C2P10H_monocytes <- subset(C2P10H, idents = c("CD16 monocytes"))
table(Idents(C2P10H_monocytes))

Idents(object = C2P10H_monocytes) <- "Control 17 monocytes"
C2P10H_monocytes$sample <- "Control 17"
C2P10H_monocytes$stim <- "Healthy"
levels(C2P10H_monocytes)
C2P10H_monocytes
saveRDS(C2P10H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P10H_monocytes.rds")
rm(C2P10H,C2P10H_monocytes,C2P10H.markers)

####C2P19H
C2P19H
Idents(object = C2P19H) <- "Control 18"
C2P19H$sample <- "Control 18"
C2P19H$stim <- "Healthy"
levels(C2P19H)
C2P19H
C2P19H[["percent.mt"]] <- PercentageFeatureSet(C2P19H, pattern = "^MT-")
VlnPlot(C2P19H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C2P19H<- subset(C2P19H, subset = nFeature_RNA > 200 & nFeature_RNA < 1500 & percent.mt < 6)
C2P19H
C2P19H <- PercentageFeatureSet(C2P19H, pattern = "^MT-", col.name = "percent.mt")
C2P19H <- SCTransform(C2P19H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C2P19H) <- "sample"
levels(C2P19H)
C2P19H<- RunPCA(C2P19H, features = VariableFeatures(object = C2P19H))
DimPlot(C2P19H, reduction = "pca")
ElbowPlot(C2P19H, ndims = 100)
pct <- C2P19H[["pca"]]@stdev / sum(C2P19H[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C2P19H <- FindNeighbors(C2P19H, dims = 1:13)
C2P19H <- FindClusters(C2P19H, resolution = 0.5)
C2P19H <- RunUMAP(C2P19H, dims = 1:13)
DimPlot(C2P19H, reduction = "umap", pt.size =0.2, label = TRUE)
C2P19H.markers <- FindAllMarkers(C2P19H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C2P19H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C2P19H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C2P19H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C2P19H, features = c("FCGR3A","MS4A7", "CD14"))
levels(C2P19H)
new.cluster.ids <- c("0", "1", "2", "3","4","5","CD16 monocytes","7")
names(new.cluster.ids) <- levels(C2P19H)
C2P19H <- RenameIdents(C2P19H, new.cluster.ids)
levels(C2P19H)
DimPlot(C2P19H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C2P19H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P19H.rds")
table(Idents(C2P19H)) 

#subset  and save CD16 monocytes 
C2P19H_monocytes <- subset(C2P19H, idents = c("CD16 monocytes"))
table(Idents(C2P19H_monocytes))

Idents(object = C2P19H_monocytes) <- "Control 18 monocytes"
C2P19H_monocytes$sample <- "Control 18"
C2P19H_monocytes$stim <- "Healthy"
levels(C2P19H_monocytes)
C2P19H_monocytes
saveRDS(C2P19H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P19H_monocytes.rds")
rm(C2P19H,C2P19H_monocytes,C2P19H.markers)

####C2P15H
C2P15H
Idents(object = C2P15H) <- "Control 19"
C2P15H$sample <- "Control 19"
C2P15H$stim <- "Healthy"
levels(C2P15H)
C2P15H
C2P15H[["percent.mt"]] <- PercentageFeatureSet(C2P15H, pattern = "^MT-")
VlnPlot(C2P15H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C2P15H<- subset(C2P15H, subset = nFeature_RNA > 200 & nFeature_RNA < 1500 & percent.mt < 5)
C2P15H
C2P15H <- PercentageFeatureSet(C2P15H, pattern = "^MT-", col.name = "percent.mt")
C2P15H <- SCTransform(C2P15H, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C2P15H) <- "sample"
levels(C2P15H)
C2P15H<- RunPCA(C2P15H, features = VariableFeatures(object = C2P15H))
DimPlot(C2P15H, reduction = "pca")
ElbowPlot(C2P15H, ndims = 100)
pct <- C2P15H[["pca"]]@stdev / sum(C2P15H[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C2P15H <- FindNeighbors(C2P15H, dims = 1:11)
C2P15H <- FindClusters(C2P15H, resolution = 0.5)
C2P15H <- RunUMAP(C2P15H, dims = 1:11)
DimPlot(C2P15H, reduction = "umap", pt.size =0.2, label = TRUE)
C2P15H.markers <- FindAllMarkers(C2P15H, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C2P15H.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C2P15H, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C2P15H, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C2P15H, features = c("FCGR3A","MS4A7", "CD14"))
levels(C2P15H)
new.cluster.ids <- c("0", "1", "2", "3","4","5","6","7" ,"CD16 monocytes","9")
names(new.cluster.ids) <- levels(C2P15H)
C2P15H <- RenameIdents(C2P15H, new.cluster.ids)
levels(C2P15H)
DimPlot(C2P15H, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C2P15H, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P15H.rds")
table(Idents(C2P15H)) 

#very few cells
#subset  and save CD16 monocytes 
C2P15H_monocytes <- subset(C2P15H, idents = c("CD16 monocytes"))
table(Idents(C2P15H_monocytes))

Idents(object = C2P15H_monocytes) <- "Control 19 monocytes"
C2P15H_monocytes$sample <- "Control 19"
C2P15H_monocytes$stim <- "Healthy"
levels(C2P15H_monocytes)
C2P15H_monocytes
saveRDS(C2P15H_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C2P15H_monocytes.rds")
rm(C2P15H,C2P15H_monocytes,C2P15H.markers)

#####start extracting from covid samples
####C19_CB_0001
C19_CB_0001
Idents(object = C19_CB_0001) <- "Cov2 Sample 1c"
C19_CB_0001$sample <- "Cov2 Sample 1c"
C19_CB_0001$stim <- "Cov2"
levels(C19_CB_0001)
C19_CB_0001
C19_CB_0001[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0001, pattern = "^MT-")
VlnPlot(C19_CB_0001, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0001<- subset(C19_CB_0001, subset = nFeature_RNA > 200 & nFeature_RNA < 5000 & percent.mt < 25)
C19_CB_0001
C19_CB_0001 <- PercentageFeatureSet(C19_CB_0001, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0001 <- SCTransform(C19_CB_0001, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0001) <- "sample"
levels(C19_CB_0001)
C19_CB_0001<- RunPCA(C19_CB_0001, features = VariableFeatures(object = C19_CB_0001))
DimPlot(C19_CB_0001, reduction = "pca")
ElbowPlot(C19_CB_0001, ndims = 100)
pct <- C19_CB_0001[["pca"]]@stdev / sum(C19_CB_0001[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0001 <- FindNeighbors(C19_CB_0001, dims = 1:13)
C19_CB_0001 <- FindClusters(C19_CB_0001, resolution = 0.5)
C19_CB_0001 <- RunUMAP(C19_CB_0001, dims = 1:13)
DimPlot(C19_CB_0001, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0001.markers <- FindAllMarkers(C19_CB_0001, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0001.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0001, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0001, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0001, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0001)
new.cluster.ids <- c("0", "CD16 monocytes", "2", "3","4","5","6","7" ,"8","9","10")
names(new.cluster.ids) <- levels(C19_CB_0001)
C19_CB_0001 <- RenameIdents(C19_CB_0001, new.cluster.ids)
levels(C19_CB_0001)
DimPlot(C19_CB_0001, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C19_CB_0001, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0001.rds")
table(Idents(C19_CB_0001)) 

#very few cells
#subset  and save CD16 monocytes 
C19_CB_0001_monocytes <- subset(C19_CB_0001, idents = c("CD16 monocytes"))
table(Idents(C19_CB_0001_monocytes))

Idents(object = C19_CB_0001_monocytes) <- "Cov2 Sample 1c"
C19_CB_0001_monocytes$sample <- "Cov2 Sample 1c"
C19_CB_0001_monocytes$stim <- "Cov2"
levels(C19_CB_0001_monocytes)
C19_CB_0001_monocytes
saveRDS(C19_CB_0001_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0001_monocytes.rds")
rm(C19_CB_0001,C19_CB_0001_monocytes,C19_CB_0001.markers)

####C19_CB_0002
C19_CB_0002
Idents(object = C19_CB_0002) <- "Cov2 Sample 2c"
C19_CB_0002$sample <- "Cov2 Sample 2c"
C19_CB_0002$stim <- "Cov2"
levels(C19_CB_0002)
C19_CB_0002
C19_CB_0002[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0002, pattern = "^MT-")
VlnPlot(C19_CB_0002, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0002<- subset(C19_CB_0002, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt <15)
C19_CB_0002
C19_CB_0002 <- PercentageFeatureSet(C19_CB_0002, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0002 <- SCTransform(C19_CB_0002, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0002) <- "sample"
levels(C19_CB_0002)
C19_CB_0002<- RunPCA(C19_CB_0002, features = VariableFeatures(object = C19_CB_0002))
DimPlot(C19_CB_0002, reduction = "pca")
ElbowPlot(C19_CB_0002, ndims = 100)
pct <- C19_CB_0002[["pca"]]@stdev / sum(C19_CB_0002[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0002 <- FindNeighbors(C19_CB_0002, dims = 1:15)
C19_CB_0002 <- FindClusters(C19_CB_0002, resolution = 0.5)
C19_CB_0002 <- RunUMAP(C19_CB_0002, dims = 1:15)
DimPlot(C19_CB_0002, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0002.markers <- FindAllMarkers(C19_CB_0002, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0002.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0002, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0002, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0002, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0002)

###no monocytes 
rm(C19_CB_0002,C19_CB_0002.markers)

####C19_CB_0003
C19_CB_0003
Idents(object = C19_CB_0003) <- "Cov2 Sample 3c"
C19_CB_0003$sample <- "Cov2 Sample 3c"
C19_CB_0003$stim <- "Cov2"
levels(C19_CB_0003)
C19_CB_0003
C19_CB_0003[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0003, pattern = "^MT-")
VlnPlot(C19_CB_0003, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0003<- subset(C19_CB_0003, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt <15)
C19_CB_0003
C19_CB_0003 <- PercentageFeatureSet(C19_CB_0003, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0003 <- SCTransform(C19_CB_0003, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0003) <- "sample"
levels(C19_CB_0003)
C19_CB_0003<- RunPCA(C19_CB_0003, features = VariableFeatures(object = C19_CB_0003))
DimPlot(C19_CB_0003, reduction = "pca")
ElbowPlot(C19_CB_0003, ndims = 100)
pct <- C19_CB_0003[["pca"]]@stdev / sum(C19_CB_0003[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0003 <- FindNeighbors(C19_CB_0003, dims = 1:18)
C19_CB_0003 <- FindClusters(C19_CB_0003, resolution = 0.5)
C19_CB_0003 <- RunUMAP(C19_CB_0003, dims = 1:18)
DimPlot(C19_CB_0003, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0003.markers <- FindAllMarkers(C19_CB_0003, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0003.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0003, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0003, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0003, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0003)


new.cluster.ids <- c("0", "1", "2", "3","CD16 monocytes","5","6","7" ,"8")
names(new.cluster.ids) <- levels(C19_CB_0003)
C19_CB_0003 <- RenameIdents(C19_CB_0003, new.cluster.ids)
levels(C19_CB_0003)
DimPlot(C19_CB_0003, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C19_CB_0003, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0003.rds")
table(Idents(C19_CB_0003)) 

#subset  and save CD16 monocytes 
C19_CB_0003_monocytes <- subset(C19_CB_0003, idents = c("CD16 monocytes"))
table(Idents(C19_CB_0003_monocytes))

Idents(object = C19_CB_0003_monocytes) <- "Cov2 Sample 3c"
C19_CB_0003_monocytes$sample <- "Cov2 Sample 3c"
C19_CB_0003_monocytes$stim <- "Cov2"
levels(C19_CB_0003_monocytes)
C19_CB_0003_monocytes
saveRDS(C19_CB_0003_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0003_monocytes.rds")
rm(C19_CB_0003,C19_CB_0003_monocytes,C19_CB_0003.markers)

####C19_CB_0005
C19_CB_0005
Idents(object = C19_CB_0005) <- "Cov2 Sample 4c"
C19_CB_0005$sample <- "Cov2 Sample 4c"
C19_CB_0005$stim <- "Cov2"
levels(C19_CB_0005)
C19_CB_0005
C19_CB_0005[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0005, pattern = "^MT-")
VlnPlot(C19_CB_0005, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0005<- subset(C19_CB_0005, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt <10)
C19_CB_0005
C19_CB_0005 <- PercentageFeatureSet(C19_CB_0005, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0005 <- SCTransform(C19_CB_0005, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0005) <- "sample"
levels(C19_CB_0005)
C19_CB_0005<- RunPCA(C19_CB_0005, features = VariableFeatures(object = C19_CB_0005))
DimPlot(C19_CB_0005, reduction = "pca")
ElbowPlot(C19_CB_0005, ndims = 100)
pct <- C19_CB_0005[["pca"]]@stdev / sum(C19_CB_0005[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0005 <- FindNeighbors(C19_CB_0005, dims = 1:11)
C19_CB_0005 <- FindClusters(C19_CB_0005, resolution = 0.5)
C19_CB_0005 <- RunUMAP(C19_CB_0005, dims = 1:11)
DimPlot(C19_CB_0005, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0005.markers <- FindAllMarkers(C19_CB_0005, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0005.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0005, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0005, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0005, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0005)


new.cluster.ids <- c("0", "1", "2","CD16 monocytes","4", "5","6","7" ,"8")
names(new.cluster.ids) <- levels(C19_CB_0005)
C19_CB_0005 <- RenameIdents(C19_CB_0005, new.cluster.ids)
levels(C19_CB_0005)
DimPlot(C19_CB_0005, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C19_CB_0005, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0005.rds")
table(Idents(C19_CB_0005)) 

#subset  and save CD16 monocytes 
C19_CB_0005_monocytes <- subset(C19_CB_0005, idents = c("CD16 monocytes"))
table(Idents(C19_CB_0005_monocytes))

Idents(object = C19_CB_0005_monocytes) <- "Cov2 Sample 4c"
C19_CB_0005_monocytes$sample <- "Cov2 Sample 4c"
C19_CB_0005_monocytes$stim <- "Cov2"
levels(C19_CB_0005_monocytes)
C19_CB_0005_monocytes
saveRDS(C19_CB_0005_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0005_monocytes.rds")
rm(C19_CB_0005,C19_CB_0005_monocytes,C19_CB_0005.markers)

####C19_CB_0052
C19_CB_0052
Idents(object = C19_CB_0052) <- "Cov2 Sample 5c"
C19_CB_0052$sample <- "Cov2 Sample c"
C19_CB_0052$stim <- "Cov2"
levels(C19_CB_0052)
C19_CB_0052
C19_CB_0052[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0052, pattern = "^MT-")
VlnPlot(C19_CB_0052, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0052<- subset(C19_CB_0052, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt <10)
C19_CB_0052
C19_CB_0052 <- PercentageFeatureSet(C19_CB_0052, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0052 <- SCTransform(C19_CB_0052, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0052) <- "sample"
levels(C19_CB_0052)
C19_CB_0052<- RunPCA(C19_CB_0052, features = VariableFeatures(object = C19_CB_0052))
DimPlot(C19_CB_0052, reduction = "pca")
ElbowPlot(C19_CB_0052, ndims = 100)
pct <- C19_CB_0052[["pca"]]@stdev / sum(C19_CB_0052[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0052 <- FindNeighbors(C19_CB_0052, dims = 1:12)
C19_CB_0052 <- FindClusters(C19_CB_0052, resolution = 0.5)
C19_CB_0052 <- RunUMAP(C19_CB_0052, dims = 1:12)
DimPlot(C19_CB_0052, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0052.markers <- FindAllMarkers(C19_CB_0052, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0052.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0052, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0052, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0052, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0052)

#no monocytes

####C19_CB_0053
C19_CB_0053
Idents(object = C19_CB_0053) <- "Cov2 Sample 6c"
C19_CB_0053$sample <- "Cov2 Sample 6c"
C19_CB_0053$stim <- "Cov2"
levels(C19_CB_0053)
C19_CB_0053
C19_CB_0053[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0053, pattern = "^MT-")
VlnPlot(C19_CB_0053, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0053<- subset(C19_CB_0053, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt <10)
C19_CB_0053
C19_CB_0053 <- PercentageFeatureSet(C19_CB_0053, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0053 <- SCTransform(C19_CB_0053, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0053) <- "sample"
levels(C19_CB_0053)
C19_CB_0053<- RunPCA(C19_CB_0053, features = VariableFeatures(object = C19_CB_0053))
DimPlot(C19_CB_0053, reduction = "pca")
ElbowPlot(C19_CB_0053, ndims = 100)
pct <- C19_CB_0053[["pca"]]@stdev / sum(C19_CB_0053[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0053 <- FindNeighbors(C19_CB_0053, dims = 1:11)
C19_CB_0053 <- FindClusters(C19_CB_0053, resolution = 0.5)
C19_CB_0053 <- RunUMAP(C19_CB_0053, dims = 1:11)
DimPlot(C19_CB_0053, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0053.markers <- FindAllMarkers(C19_CB_0053, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0053.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0053, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0053, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0053, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0053)
##no monocytes 

####C19_CB_0204
C19_CB_0204
Idents(object = C19_CB_0204) <- "Cov2 Sample 7c"
C19_CB_0204$sample <- "Cov2 Sample 7c"
C19_CB_0204$stim <- "Cov2"
levels(C19_CB_0204)
C19_CB_0204
C19_CB_0204[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0204, pattern = "^MT-")
VlnPlot(C19_CB_0204, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0204<- subset(C19_CB_0204, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt <10)
C19_CB_0204
C19_CB_0204 <- PercentageFeatureSet(C19_CB_0204, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0204 <- SCTransform(C19_CB_0204, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0204) <- "sample"
levels(C19_CB_0204)
C19_CB_0204<- RunPCA(C19_CB_0204, features = VariableFeatures(object = C19_CB_0204))
DimPlot(C19_CB_0204, reduction = "pca")
ElbowPlot(C19_CB_0204, ndims = 100)
pct <- C19_CB_0204[["pca"]]@stdev / sum(C19_CB_0204[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0204 <- FindNeighbors(C19_CB_0204, dims = 1:15)
C19_CB_0204 <- FindClusters(C19_CB_0204, resolution = 0.5)
C19_CB_0204 <- RunUMAP(C19_CB_0204, dims = 1:15)
DimPlot(C19_CB_0204, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0204.markers <- FindAllMarkers(C19_CB_0204, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0204.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0204, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0204, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0204, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0204)


new.cluster.ids <- c("0", "1", "2","3","4", "CD16 monocytes","6","7" ,"8","9")
names(new.cluster.ids) <- levels(C19_CB_0204)
C19_CB_0204 <- RenameIdents(C19_CB_0204, new.cluster.ids)
levels(C19_CB_0204)
DimPlot(C19_CB_0204, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C19_CB_0204, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0204.rds")
table(Idents(C19_CB_0204)) 

#subset  and save CD16 monocytes 
C19_CB_0204_monocytes <- subset(C19_CB_0204, idents = c("CD16 monocytes"))
table(Idents(C19_CB_0204_monocytes))

Idents(object = C19_CB_0204_monocytes) <- "Cov2 Sample 7c"
C19_CB_0204_monocytes$sample <- "Cov2 Sample 7c"
C19_CB_0204_monocytes$stim <- "Cov2"
levels(C19_CB_0204_monocytes)
C19_CB_0204_monocytes
saveRDS(C19_CB_0204_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0204_monocytes.rds")
rm(C19_CB_0204,C19_CB_0204_monocytes,C19_CB_0204.markers)

####C19_CB_0214
C19_CB_0214
Idents(object = C19_CB_0214) <- "Cov2 Sample 8c"
C19_CB_0214$sample <- "Cov2 Sample 8c"
C19_CB_0214$stim <- "Cov2"
levels(C19_CB_0214)
C19_CB_0214
C19_CB_0214[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0214, pattern = "^MT-")
VlnPlot(C19_CB_0214, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0214<- subset(C19_CB_0214, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt <10)
C19_CB_0214
C19_CB_0214 <- PercentageFeatureSet(C19_CB_0214, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0214 <- SCTransform(C19_CB_0214, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0214) <- "sample"
levels(C19_CB_0214)
C19_CB_0214<- RunPCA(C19_CB_0214, features = VariableFeatures(object = C19_CB_0214))
DimPlot(C19_CB_0214, reduction = "pca")
ElbowPlot(C19_CB_0214, ndims = 100)
pct <- C19_CB_0214[["pca"]]@stdev / sum(C19_CB_0214[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0214 <- FindNeighbors(C19_CB_0214, dims = 1:15)
C19_CB_0214 <- FindClusters(C19_CB_0214, resolution = 0.5)
C19_CB_0214 <- RunUMAP(C19_CB_0214, dims = 1:15)
DimPlot(C19_CB_0214, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0214.markers <- FindAllMarkers(C19_CB_0214, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0214.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0214, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0214, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0214, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0214)

#no monocytes 

####C19_CB_0008
C19_CB_0008
Idents(object = C19_CB_0008) <- "Cov2 Sample 9c"
C19_CB_0008$sample <- "Cov2 Sample 9c"
C19_CB_0008$stim <- "Cov2"
levels(C19_CB_0008)
C19_CB_0008
C19_CB_0008[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0008, pattern = "^MT-")
VlnPlot(C19_CB_0008, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0008<- subset(C19_CB_0008, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt <10)
C19_CB_0008
C19_CB_0008 <- PercentageFeatureSet(C19_CB_0008, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0008 <- SCTransform(C19_CB_0008, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0008) <- "sample"
levels(C19_CB_0008)
C19_CB_0008<- RunPCA(C19_CB_0008, features = VariableFeatures(object = C19_CB_0008))
DimPlot(C19_CB_0008, reduction = "pca")
ElbowPlot(C19_CB_0008, ndims = 100)
pct <- C19_CB_0008[["pca"]]@stdev / sum(C19_CB_0008[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0008 <- FindNeighbors(C19_CB_0008, dims = 1:19)
C19_CB_0008 <- FindClusters(C19_CB_0008, resolution = 0.5)
C19_CB_0008 <- RunUMAP(C19_CB_0008, dims = 1:19)
DimPlot(C19_CB_0008, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0008.markers <- FindAllMarkers(C19_CB_0008, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0008.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0008, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0008, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0008, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0008)


new.cluster.ids <- c("0", "1", "2","3","4","5","6","7", "CD16 monocytes","9","10","11")
names(new.cluster.ids) <- levels(C19_CB_0008)
C19_CB_0008 <- RenameIdents(C19_CB_0008, new.cluster.ids)
levels(C19_CB_0008)
DimPlot(C19_CB_0008, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C19_CB_0008, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0008.rds")
table(Idents(C19_CB_0008)) 

#subset  and save CD16 monocytes 
C19_CB_0008_monocytes <- subset(C19_CB_0008, idents = c("CD16 monocytes"))
table(Idents(C19_CB_0008_monocytes))

Idents(object = C19_CB_0008_monocytes) <- "Cov2 Sample 9c"
C19_CB_0008_monocytes$sample <- "Cov2 Sample 9c"
C19_CB_0008_monocytes$stim <- "Cov2"
levels(C19_CB_0008_monocytes)
C19_CB_0008_monocytes
saveRDS(C19_CB_0008_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0008_monocytes.rds")
rm(C19_CB_0008,C19_CB_0008_monocytes,C19_CB_0008.markers)

####C19_CB_0009
C19_CB_0009
Idents(object = C19_CB_0009) <- "Cov2 Sample 10c"
C19_CB_0009$sample <- "Cov2 Sample 10c"
C19_CB_0009$stim <- "Cov2"
levels(C19_CB_0009)
C19_CB_0009
C19_CB_0009[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0009, pattern = "^MT-")
VlnPlot(C19_CB_0009, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0009<- subset(C19_CB_0009, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt <10)
C19_CB_0009
C19_CB_0009 <- PercentageFeatureSet(C19_CB_0009, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0009 <- SCTransform(C19_CB_0009, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0009) <- "sample"
levels(C19_CB_0009)
C19_CB_0009<- RunPCA(C19_CB_0009, features = VariableFeatures(object = C19_CB_0009))
DimPlot(C19_CB_0009, reduction = "pca")
ElbowPlot(C19_CB_0009, ndims = 100)
pct <- C19_CB_0009[["pca"]]@stdev / sum(C19_CB_0009[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0009 <- FindNeighbors(C19_CB_0009, dims = 1:22)
C19_CB_0009 <- FindClusters(C19_CB_0009, resolution = 0.5)
C19_CB_0009 <- RunUMAP(C19_CB_0009, dims = 1:22)
DimPlot(C19_CB_0009, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0009.markers <- FindAllMarkers(C19_CB_0009, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0009.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0009, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0009, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0009, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0009)



new.cluster.ids <- c("0", "1", "2","3","4","5" ,"6","7" ,"8","9","10", "CD16 monocytes","12","13" ,"14","15","16","17")
names(new.cluster.ids) <- levels(C19_CB_0009)
C19_CB_0009 <- RenameIdents(C19_CB_0009, new.cluster.ids)
levels(C19_CB_0009)
DimPlot(C19_CB_0009, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C19_CB_0009, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0009.rds")
table(Idents(C19_CB_0009)) 

#subset  and save CD16 monocytes 
C19_CB_0009_monocytes <- subset(C19_CB_0009, idents = c("CD16 monocytes"))
table(Idents(C19_CB_0009_monocytes))

Idents(object = C19_CB_0009_monocytes) <- "Cov2 Sample 10c"
C19_CB_0009_monocytes$sample <- "Cov2 Sample 10c"
C19_CB_0009_monocytes$stim <- "Cov2"
levels(C19_CB_0009_monocytes)
C19_CB_0009_monocytes
saveRDS(C19_CB_0009_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0009_monocytes.rds")
rm(C19_CB_0009,C19_CB_0009_monocytes,C19_CB_0009.markers)

####C19_CB_0011
C19_CB_0011
Idents(object = C19_CB_0011) <- "Cov2 Sample 11c"
C19_CB_0011$sample <- "Cov2 Sample 11c"
C19_CB_0011$stim <- "Cov2"
levels(C19_CB_0011)
C19_CB_0011
C19_CB_0011[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0011, pattern = "^MT-")
VlnPlot(C19_CB_0011, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0011<- subset(C19_CB_0011, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt <7)
C19_CB_0011
C19_CB_0011 <- PercentageFeatureSet(C19_CB_0011, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0011 <- SCTransform(C19_CB_0011, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0011) <- "sample"
levels(C19_CB_0011)
C19_CB_0011<- RunPCA(C19_CB_0011, features = VariableFeatures(object = C19_CB_0011))
DimPlot(C19_CB_0011, reduction = "pca")
ElbowPlot(C19_CB_0011, ndims = 100)
pct <- C19_CB_0011[["pca"]]@stdev / sum(C19_CB_0011[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0011 <- FindNeighbors(C19_CB_0011, dims = 1:19)
C19_CB_0011 <- FindClusters(C19_CB_0011, resolution = 0.5)
C19_CB_0011 <- RunUMAP(C19_CB_0011, dims = 1:19)
DimPlot(C19_CB_0011, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0011.markers <- FindAllMarkers(C19_CB_0011, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0011.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0011, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0011, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0011, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0011)


new.cluster.ids <- c("0", "1", "2","3","4","5","6","7","8", "CD16 monocytes","9")
names(new.cluster.ids) <- levels(C19_CB_0011)
C19_CB_0011 <- RenameIdents(C19_CB_0011, new.cluster.ids)
levels(C19_CB_0011)
DimPlot(C19_CB_0011, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C19_CB_0011, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0011.rds")
table(Idents(C19_CB_0011)) 
#few monocytes
#subset  and save CD16 monocytes 
C19_CB_0011_monocytes <- subset(C19_CB_0011, idents = c("CD16 monocytes"))
table(Idents(C19_CB_0011_monocytes))

Idents(object = C19_CB_0011_monocytes) <- "Cov2 Sample 11c"
C19_CB_0011_monocytes$sample <- "Cov2 Sample 11c"
C19_CB_0011_monocytes$stim <- "Cov2"
levels(C19_CB_0011_monocytes)
C19_CB_0011_monocytes
saveRDS(C19_CB_0011_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0011_monocytes.rds")
rm(C19_CB_0011,C19_CB_0011_monocytes,C19_CB_0011.markers)

####C19_CB_0012
C19_CB_0012
Idents(object = C19_CB_0012) <- "Cov2 Sample 12c"
C19_CB_0012$sample <- "Cov2 Sample 12c"
C19_CB_0012$stim <- "Cov2"
levels(C19_CB_0012)
C19_CB_0012
C19_CB_0012[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0012, pattern = "^MT-")
VlnPlot(C19_CB_0012, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0012<- subset(C19_CB_0012, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt <10)
C19_CB_0012
C19_CB_0012 <- PercentageFeatureSet(C19_CB_0012, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0012 <- SCTransform(C19_CB_0012, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0012) <- "sample"
levels(C19_CB_0012)
C19_CB_0012<- RunPCA(C19_CB_0012, features = VariableFeatures(object = C19_CB_0012))
DimPlot(C19_CB_0012, reduction = "pca")
ElbowPlot(C19_CB_0012, ndims = 100)
pct <- C19_CB_0012[["pca"]]@stdev / sum(C19_CB_0012[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0012 <- FindNeighbors(C19_CB_0012, dims = 1:16)
C19_CB_0012 <- FindClusters(C19_CB_0012, resolution = 0.5)
C19_CB_0012 <- RunUMAP(C19_CB_0012, dims = 1:16)
DimPlot(C19_CB_0012, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0012.markers <- FindAllMarkers(C19_CB_0012, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0012.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0012, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0012, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0012, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0012)



new.cluster.ids <- c("0", "1", "2","3","4","5" ,"6","7" ,"8","9","10","11", "CD16 monocytes","13" ,"14")
names(new.cluster.ids) <- levels(C19_CB_0012)
C19_CB_0012 <- RenameIdents(C19_CB_0012, new.cluster.ids)
levels(C19_CB_0012)
DimPlot(C19_CB_0012, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C19_CB_0012, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0012.rds")
table(Idents(C19_CB_0012)) 
#few monocytes
#subset  and save CD16 monocytes 
C19_CB_0012_monocytes <- subset(C19_CB_0012, idents = c("CD16 monocytes"))
table(Idents(C19_CB_0012_monocytes))

Idents(object = C19_CB_0012_monocytes) <- "Cov2 Sample 12c"
C19_CB_0012_monocytes$sample <- "Cov2 Sample 12c"
C19_CB_0012_monocytes$stim <- "Cov2"
levels(C19_CB_0012_monocytes)
C19_CB_0012_monocytes
saveRDS(C19_CB_0012_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0012_monocytes.rds")
rm(C19_CB_0012,C19_CB_0012_monocytes,C19_CB_0012.markers)

####C19_CB_0013
C19_CB_0013
Idents(object = C19_CB_0013) <- "Cov2 Sample 13c"
C19_CB_0013$sample <- "Cov2 Sample 13c"
C19_CB_0013$stim <- "Cov2"
levels(C19_CB_0013)
C19_CB_0013
C19_CB_0013[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0013, pattern = "^MT-")
VlnPlot(C19_CB_0013, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0013<- subset(C19_CB_0013, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 7)
C19_CB_0013
C19_CB_0013 <- PercentageFeatureSet(C19_CB_0013, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0013 <- SCTransform(C19_CB_0013, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0013) <- "sample"
levels(C19_CB_0013)
C19_CB_0013<- RunPCA(C19_CB_0013, features = VariableFeatures(object = C19_CB_0013))
DimPlot(C19_CB_0013, reduction = "pca")
ElbowPlot(C19_CB_0013, ndims = 100)
pct <- C19_CB_0013[["pca"]]@stdev / sum(C19_CB_0013[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0013 <- FindNeighbors(C19_CB_0013, dims = 1:12)
C19_CB_0013 <- FindClusters(C19_CB_0013, resolution = 0.5)
C19_CB_0013 <- RunUMAP(C19_CB_0013, dims = 1:12)
DimPlot(C19_CB_0013, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0013.markers <- FindAllMarkers(C19_CB_0013, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0013.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0013, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0013, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0013, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0013)


new.cluster.ids <- c("0", "1", "2","3","4","5","6","7", "CD16 monocytes","9")
names(new.cluster.ids) <- levels(C19_CB_0013)
C19_CB_0013 <- RenameIdents(C19_CB_0013, new.cluster.ids)
levels(C19_CB_0013)
DimPlot(C19_CB_0013, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C19_CB_0013, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0013.rds")
table(Idents(C19_CB_0013)) 
#few monocytes

####C19_CB_0016
C19_CB_0016
Idents(object = C19_CB_0016) <- "Cov2 Sample 14c"
C19_CB_0016$sample <- "Cov2 Sample 14c"
C19_CB_0016$stim <- "Cov2"
levels(C19_CB_0016)
C19_CB_0016
C19_CB_0016[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0016, pattern = "^MT-")
VlnPlot(C19_CB_0016, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0016<- subset(C19_CB_0016, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt <10)
C19_CB_0016
C19_CB_0016 <- PercentageFeatureSet(C19_CB_0016, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0016 <- SCTransform(C19_CB_0016, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0016) <- "sample"
levels(C19_CB_0016)
C19_CB_0016<- RunPCA(C19_CB_0016, features = VariableFeatures(object = C19_CB_0016))
DimPlot(C19_CB_0016, reduction = "pca")
ElbowPlot(C19_CB_0016, ndims = 100)
pct <- C19_CB_0016[["pca"]]@stdev / sum(C19_CB_0016[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0016 <- FindNeighbors(C19_CB_0016, dims = 1:16)
C19_CB_0016 <- FindClusters(C19_CB_0016, resolution = 0.5)
C19_CB_0016 <- RunUMAP(C19_CB_0016, dims = 1:16)
DimPlot(C19_CB_0016, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0016.markers <- FindAllMarkers(C19_CB_0016, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0016.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0016, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0016, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0016, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0016)

###no monocytes

####C19_CB_0021
C19_CB_0021
Idents(object = C19_CB_0021) <- "Cov2 Sample 16c"
C19_CB_0021$sample <- "Cov2 Sample 16c"
C19_CB_0021$stim <- "Cov2"
levels(C19_CB_0021)
C19_CB_0021
C19_CB_0021[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0021, pattern = "^MT-")
VlnPlot(C19_CB_0021, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0021<- subset(C19_CB_0021, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 10)
C19_CB_0021
C19_CB_0021 <- PercentageFeatureSet(C19_CB_0021, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0021 <- SCTransform(C19_CB_0021, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0021) <- "sample"
levels(C19_CB_0021)
C19_CB_0021<- RunPCA(C19_CB_0021, features = VariableFeatures(object = C19_CB_0021))
DimPlot(C19_CB_0021, reduction = "pca")
ElbowPlot(C19_CB_0021, ndims = 100)
pct <- C19_CB_0021[["pca"]]@stdev / sum(C19_CB_0021[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0021 <- FindNeighbors(C19_CB_0021, dims = 1:14)
C19_CB_0021 <- FindClusters(C19_CB_0021, resolution = 0.5)
C19_CB_0021 <- RunUMAP(C19_CB_0021, dims = 1:14)
DimPlot(C19_CB_0021, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0021.markers <- FindAllMarkers(C19_CB_0021, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0021.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0021, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0021, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0021, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0021)


new.cluster.ids <- c("0", "1", "2","3","4","5","6", "CD16 monocytes","8","9","10","11","12","13","14")
names(new.cluster.ids) <- levels(C19_CB_0021)
C19_CB_0021 <- RenameIdents(C19_CB_0021, new.cluster.ids)
levels(C19_CB_0021)
DimPlot(C19_CB_0021, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C19_CB_0021, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0021.rds")
table(Idents(C19_CB_0021)) 

#subset  and save CD16 monocytes 
C19_CB_0021_monocytes <- subset(C19_CB_0021, idents = c("CD16 monocytes"))
table(Idents(C19_CB_0021_monocytes))

Idents(object = C19_CB_0021_monocytes) <- "Cov2 Sample 16c"
C19_CB_0021_monocytes$sample <- "Cov2 Sample 16c"
C19_CB_0021_monocytes$stim <- "Cov2"
levels(C19_CB_0021_monocytes)
C19_CB_0021_monocytes
saveRDS(C19_CB_0021_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0021_monocytes.rds")
rm(C19_CB_0021,C19_CB_0021_monocytes,C19_CB_0021.markers)

####C19_CB_0020
C19_CB_0020
Idents(object = C19_CB_0020) <- "Cov2 Sample 15c"
C19_CB_0020$sample <- "Cov2 Sample 15c"
C19_CB_0020$stim <- "Cov2"
levels(C19_CB_0020)
C19_CB_0020
C19_CB_0020[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0020, pattern = "^MT-")
VlnPlot(C19_CB_0020, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0020<- subset(C19_CB_0020, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt <10)
C19_CB_0020
C19_CB_0020 <- PercentageFeatureSet(C19_CB_0020, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0020 <- SCTransform(C19_CB_0020, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0020) <- "sample"
levels(C19_CB_0020)
C19_CB_0020<- RunPCA(C19_CB_0020, features = VariableFeatures(object = C19_CB_0020))
DimPlot(C19_CB_0020, reduction = "pca")
ElbowPlot(C19_CB_0020, ndims = 100)
pct <- C19_CB_0020[["pca"]]@stdev / sum(C19_CB_0020[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0020 <- FindNeighbors(C19_CB_0020, dims = 1:17)
C19_CB_0020 <- FindClusters(C19_CB_0020, resolution = 0.5)
C19_CB_0020 <- RunUMAP(C19_CB_0020, dims = 1:17)
DimPlot(C19_CB_0020, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0020.markers <- FindAllMarkers(C19_CB_0020, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0020.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0020, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0020, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0020, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0020)

###no monocytes

new.cluster.ids <- c("0", "1", "2","3","4","5" ,"6","7" ,"8","9","10","11", "CD16 monocytes","13")
names(new.cluster.ids) <- levels(C19_CB_0020)
C19_CB_0020 <- RenameIdents(C19_CB_0020, new.cluster.ids)
levels(C19_CB_0020)
DimPlot(C19_CB_0020, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(C19_CB_0020, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0020.rds")
table(Idents(C19_CB_0020)) 
#few monocytes
#subset  and save CD16 monocytes 
C19_CB_0020_monocytes <- subset(C19_CB_0020, idents = c("CD16 monocytes"))
table(Idents(C19_CB_0020_monocytes))

Idents(object = C19_CB_0020_monocytes) <- "Cov2 Sample 15c"
C19_CB_0020_monocytes$sample <- "Cov2 Sample 15c"
C19_CB_0020_monocytes$stim <- "Cov2"
levels(CC19_CB_0020_monocytes)
C19_CB_0020_monocytes
saveRDS(C19_CB_0020_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/C19_CB_0016_monocytes.rds")
rm(C19_CB_0020,C19_CB_0020_monocytes,C19_CB_0020.markers)

####C19_CB_0198
C19_CB_0198
Idents(object = C19_CB_0198) <- "Cov2 Sample 17c"
C19_CB_0198$sample <- "Cov2 Sample 17c"
C19_CB_0198$stim <- "Cov2"
levels(C19_CB_0198)
C19_CB_0198
C19_CB_0198[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0198, pattern = "^MT-")
VlnPlot(C19_CB_0198, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0198<- subset(C19_CB_0198, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 10)
C19_CB_0198
C19_CB_0198 <- PercentageFeatureSet(C19_CB_0198, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0198 <- SCTransform(C19_CB_0198, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0198) <- "sample"
levels(C19_CB_0198)
C19_CB_0198<- RunPCA(C19_CB_0198, features = VariableFeatures(object = C19_CB_0198))
DimPlot(C19_CB_0198, reduction = "pca")
ElbowPlot(C19_CB_0198, ndims = 100)
pct <- C19_CB_0198[["pca"]]@stdev / sum(C19_CB_0198[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0198 <- FindNeighbors(C19_CB_0198, dims = 1:14)
C19_CB_0198 <- FindClusters(C19_CB_0198, resolution = 0.5)
C19_CB_0198 <- RunUMAP(C19_CB_0198, dims = 1:14)
DimPlot(C19_CB_0198, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0198.markers <- FindAllMarkers(C19_CB_0198, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0198.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0198, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0198, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0198, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0198)

#few cells. didnt proceed

####C19_CB_0199
C19_CB_0199
Idents(object = C19_CB_0199) <- "Cov2 Sample 18c"
C19_CB_0199$sample <- "Cov2 Sample 18c"
C19_CB_0199$stim <- "Cov2"
levels(C19_CB_0199)
C19_CB_0199
C19_CB_0199[["percent.mt"]] <- PercentageFeatureSet(C19_CB_0199, pattern = "^MT-")
VlnPlot(C19_CB_0199, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
C19_CB_0199<- subset(C19_CB_0199, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt <10)
C19_CB_0199
C19_CB_0199 <- PercentageFeatureSet(C19_CB_0199, pattern = "^MT-", col.name = "percent.mt")
C19_CB_0199 <- SCTransform(C19_CB_0199, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = C19_CB_0199) <- "sample"
levels(C19_CB_0199)
C19_CB_0199<- RunPCA(C19_CB_0199, features = VariableFeatures(object = C19_CB_0199))
DimPlot(C19_CB_0199, reduction = "pca")
ElbowPlot(C19_CB_0199, ndims = 100)
pct <- C19_CB_0199[["pca"]]@stdev / sum(C19_CB_0199[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
C19_CB_0199 <- FindNeighbors(C19_CB_0199, dims = 1:15)
C19_CB_0199 <- FindClusters(C19_CB_0199, resolution = 0.5)
C19_CB_0199 <- RunUMAP(C19_CB_0199, dims = 1:15)
DimPlot(C19_CB_0199, reduction = "umap", pt.size =0.2, label = TRUE)
C19_CB_0199.markers <- FindAllMarkers(C19_CB_0199, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
C19_CB_0199.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(C19_CB_0199, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(C19_CB_0199, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(C19_CB_0199, features = c("FCGR3A","MS4A7", "CD14"))
levels(C19_CB_0199)

###no monocytes

#this is already a seurat object, delete data that we are not interested in (NULL)
data
data[["HTO"]]<- NULL
data
colnames(data@meta.data)                      
data@meta.data$donor
data@meta.data$group_per_sample

Idents(data) <- "donor"
table(Idents(data))

###subset each donor to an independent seurat object
####healthy donors from Reyes et al., Nat Med, 2020
BN_26 <- subset(data, idents = c("BN-26"))
BN_27 <- subset(data, idents = c("BN-27"))
BN_28 <- subset(data, idents = c("BN-28"))
BN_29 <- subset(data, idents = c("BN-29"))
BN_30 <- subset(data, idents = c("BN-30"))
BN_31 <- subset(data, idents = c("BN-31"))
BN_32  <- subset(data, idents = c("BN-32"))
BN_33 <- subset(data, idents = c("BN-33"))
BN_34 <- subset(data, idents = c("BN-34"))
BN_35 <- subset(data, idents = c("BN-35"))
BN_36 <- subset(data, idents = c("BN-36"))
BN_37 <- subset(data, idents = c("BN-37"))

###covid patients from Bonn (cohort 2)
BN_01 <- subset(data, idents = c("BN-01"))
BN_02 <- subset(data, idents = c("BN-02"))
BN_03 <- subset(data, idents = c("BN-03"))
BN_04 <- subset(data, idents = c("BN-04"))
BN_05 <- subset(data, idents = c("BN-05"))
BN_06 <- subset(data, idents = c("BN-06"))
BN_07 <- subset(data, idents = c("BN-07"))
BN_08 <- subset(data, idents = c("BN-08"))
BN_10 <- subset(data, idents = c("BN-10"))
BN_11 <- subset(data, idents = c("BN-11"))
BN_12 <- subset(data, idents = c("BN-12"))
BN_13 <- subset(data, idents = c("BN-13"))
BN_14 <- subset(data, idents = c("BN-14"))
BN_15 <- subset(data, idents = c("BN-15"))
BN_16 <- subset(data, idents = c("BN-16"))
BN_17 <- subset(data, idents = c("BN-17"))
BN_19 <- subset(data, idents = c("BN-19"))

rm(data)

##extract monocytes from healthy donors first

####BN-26
BN_26
Idents(object = BN_26) <- "Control 20"
BN_26$sample <- "Control 20"
BN_26$stim <- "Healthy"
levels(BN_26)
BN_26
BN_26[["percent.mt"]] <- PercentageFeatureSet(BN_26, pattern = "^MT-")
VlnPlot(BN_26, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_26<- subset(BN_26, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_26
BN_26 <- PercentageFeatureSet(BN_26, pattern = "^MT-", col.name = "percent.mt")
BN_26 <- SCTransform(BN_26, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_26) <- "sample"
levels(BN_26)
BN_26<- RunPCA(BN_26, features = VariableFeatures(object = BN_26))
DimPlot(BN_26, reduction = "pca")
ElbowPlot(BN_26, ndims = 100)
pct <- BN_26[["pca"]]@stdev / sum(BN_26[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_26 <- FindNeighbors(BN_26, dims = 1:12)
BN_26 <- FindClusters(BN_26, resolution = 0.5)
BN_26 <- RunUMAP(BN_26, dims = 1:12)
DimPlot(BN_26, reduction = "umap", pt.size =0.2, label = TRUE)
BN_26.markers <- FindAllMarkers(BN_22, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_26.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_26, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_26, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_26, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_26)
new.cluster.ids <- c("0", "1", "2", "3","4","5","6","CD16 monocytes" ,"8","9", "10", "11")
names(new.cluster.ids) <- levels(BN_26)
BN_26 <- RenameIdents(BN_26, new.cluster.ids)
levels(BN_26)
DimPlot(BN_26, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_26, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_26.rds")
table(Idents(BN_26)) 

#subset  and save CD16 monocytes (we have more than 50 cells (our cut-off)) 
BN_26_monocytes <- subset(BN_26, idents = c("CD16 monocytes"))
table(Idents(BN_26_monocytes))

Idents(object = BN_26_monocytes) <- "Control 20 monocytes"
BN_26_monocytes$sample <- "Control 20"
BN_26_monocytes$stim <- "Healthy"
levels(BN_26_monocytes)
BN_26_monocytes
saveRDS(BN_26_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_26_monocytes.rds")
rm(BN_26,BN_26_monocytes,BN_26.markers)

####BN-27
BN_27
Idents(object = BN_27) <- "Control 21"
BN_27$sample <- "Control 21"
BN_27$stim <- "Healthy"
levels(BN_27)
BN_27
BN_27[["percent.mt"]] <- PercentageFeatureSet(BN_27, pattern = "^MT-")
VlnPlot(BN_27, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_27<- subset(BN_27, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_27
BN_27 <- PercentageFeatureSet(BN_27, pattern = "^MT-", col.name = "percent.mt")
BN_27 <- SCTransform(BN_27, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_27) <- "sample"
levels(BN_27)
BN_27<- RunPCA(BN_27, features = VariableFeatures(object = BN_27))
DimPlot(BN_27, reduction = "pca")
ElbowPlot(BN_27, ndims = 100)
pct <- BN_27[["pca"]]@stdev / sum(BN_27[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_27 <- FindNeighbors(BN_27, dims = 1:9)
BN_27 <- FindClusters(BN_27, resolution = 0.5)
BN_27 <- RunUMAP(BN_27, dims = 1:9)
DimPlot(BN_27, reduction = "umap", pt.size =0.2, label = TRUE)
BN_27.markers <- FindAllMarkers(BN_27, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_27.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_27, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_27, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_27, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_27)
new.cluster.ids <- c("0", "1", "2", "3","4","5","CD16 monocytes","7")
names(new.cluster.ids) <- levels(BN_27)
BN_27 <- RenameIdents(BN_27, new.cluster.ids)
levels(BN_27)
DimPlot(BN_27, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_27, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_27.rds")
table(Idents(BN_27)) 

#few monocytes (43)

#subset  and save CD16 monocytes (we have less than 50 cells (our cut-off)) 
BN_27_monocytes <- subset(BN_27, idents = c("CD16 monocytes"))
table(Idents(BN_27_monocytes))

Idents(object = BN_27_monocytes) <- "Control 21 monocytes"
BN_27_monocytes$sample <- "Control 21"
BN_27_monocytes$stim <- "Healthy"
levels(BN_27_monocytes)
BN_27_monocytes
saveRDS(BN_27_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_27_monocytes.rds")
rm(BN_27,BN_27_monocytes,BN_27.markers)

####BN-28
BN_28
Idents(object = BN_28) <- "Control 22"
BN_28$sample <- "Control 22"
BN_28$stim <- "Healthy"
levels(BN_28)
BN_28
BN_28[["percent.mt"]] <- PercentageFeatureSet(BN_28, pattern = "^MT-")
VlnPlot(BN_28, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_28<- subset(BN_28, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_28
BN_28 <- PercentageFeatureSet(BN_28, pattern = "^MT-", col.name = "percent.mt")
BN_28 <- SCTransform(BN_28, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_28) <- "sample"
levels(BN_28)
BN_28<- RunPCA(BN_28, features = VariableFeatures(object = BN_28))
DimPlot(BN_28, reduction = "pca")
ElbowPlot(BN_28, ndims = 100)
pct <- BN_28[["pca"]]@stdev / sum(BN_28[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_28 <- FindNeighbors(BN_28, dims = 1:9)
BN_28 <- FindClusters(BN_28, resolution = 0.5)
BN_28 <- RunUMAP(BN_28, dims = 1:9)
DimPlot(BN_28, reduction = "umap", pt.size =0.2, label = TRUE)
BN_28.markers <- FindAllMarkers(BN_28, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_28.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_28, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_28, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_28, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_28)
new.cluster.ids <- c("0", "1", "2", "3","4","5","6","7","CD16 monocytes","9")
names(new.cluster.ids) <- levels(BN_28)
BN_28 <- RenameIdents(BN_28, new.cluster.ids)
levels(BN_28)
DimPlot(BN_28, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_28, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_28.rds")
table(Idents(BN_28)) 

#few monocytes (49)

#subset  and save CD16 monocytes (we have less than 50 cells (our cut-off)) 
BN_28_monocytes <- subset(BN_28, idents = c("CD16 monocytes"))
table(Idents(BN_28_monocytes))

Idents(object = BN_28_monocytes) <- "Control 22 monocytes"
BN_28_monocytes$sample <- "Control 22"
BN_28_monocytes$stim <- "Healthy"
levels(BN_28_monocytes)
BN_28_monocytes
saveRDS(BN_28_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_28_monocytes.rds")
rm(BN_28,BN_28_monocytes,BN_28.markers)

####BN-29
BN_29
Idents(object = BN_29) <- "Control 23"
BN_29$sample <- "Control 23"
BN_29$stim <- "Healthy"
levels(BN_29)
BN_29
BN_29[["percent.mt"]] <- PercentageFeatureSet(BN_29, pattern = "^MT-")
VlnPlot(BN_29, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_29<- subset(BN_29, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_29
BN_29 <- PercentageFeatureSet(BN_29, pattern = "^MT-", col.name = "percent.mt")
BN_29 <- SCTransform(BN_29, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_29) <- "sample"
levels(BN_29)
BN_29<- RunPCA(BN_29, features = VariableFeatures(object = BN_29))
DimPlot(BN_29, reduction = "pca")
ElbowPlot(BN_29, ndims = 100)
pct <- BN_29[["pca"]]@stdev / sum(BN_29[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_29 <- FindNeighbors(BN_29, dims = 1:9)
BN_29 <- FindClusters(BN_29, resolution = 0.5)
BN_29 <- RunUMAP(BN_29, dims = 1:9)
DimPlot(BN_29, reduction = "umap", pt.size =0.2, label = TRUE)
BN_29.markers <- FindAllMarkers(BN_29, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_29.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_29, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_29, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_29, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_29)
new.cluster.ids <- c("0", "1", "2", "3","4","5","6","CD16 monocytes","8","9")
names(new.cluster.ids) <- levels(BN_29)
BN_29 <- RenameIdents(BN_29, new.cluster.ids)
levels(BN_29)
DimPlot(BN_29, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_29, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_29.rds")
table(Idents(BN_29)) 

#subset  and save CD16 monocytes (we have less than 50 cells (our cut-off)) 
BN_29_monocytes <- subset(BN_29, idents = c("CD16 monocytes"))
table(Idents(BN_29_monocytes))

Idents(object = BN_29_monocytes) <- "Control 23 monocytes"
BN_29_monocytes$sample <- "Control 23"
BN_29_monocytes$stim <- "Healthy"
levels(BN_29_monocytes)
BN_29_monocytes
saveRDS(BN_29_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_29_monocytes.rds")
rm(BN_29,BN_29_monocytes,BN_29.markers)

####BN-30
BN_30
Idents(object = BN_30) <- "Control 24"
BN_30$sample <- "Control 24"
BN_30$stim <- "Healthy"
levels(BN_30)
BN_30
BN_30[["percent.mt"]] <- PercentageFeatureSet(BN_30, pattern = "^MT-")
VlnPlot(BN_30, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_30<- subset(BN_30, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_30
BN_30 <- PercentageFeatureSet(BN_30, pattern = "^MT-", col.name = "percent.mt")
BN_30 <- SCTransform(BN_30, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_30) <- "sample"
levels(BN_30)
BN_30<- RunPCA(BN_30, features = VariableFeatures(object = BN_30))
DimPlot(BN_30, reduction = "pca")
ElbowPlot(BN_30, ndims = 100)
pct <- BN_30[["pca"]]@stdev / sum(BN_30[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_30 <- FindNeighbors(BN_30, dims = 1:12)
BN_30 <- FindClusters(BN_30, resolution = 0.5)
BN_30 <- RunUMAP(BN_30, dims = 1:12)
DimPlot(BN_30, reduction = "umap", pt.size =0.2, label = TRUE)
BN_30.markers <- FindAllMarkers(BN_30, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_30.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_30, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_30, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_30, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_30)
new.cluster.ids <- c("0", "1", "2", "3","4","CD16 monocytes","6","7")
names(new.cluster.ids) <- levels(BN_30)
BN_30 <- RenameIdents(BN_30, new.cluster.ids)
levels(BN_30)
DimPlot(BN_30, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_30, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_30.rds")
table(Idents(BN_30)) 

#subset  and save CD16 monocytes 
BN_30_monocytes <- subset(BN_30, idents = c("CD16 monocytes"))
table(Idents(BN_30_monocytes))



Idents(object = BN_30_monocytes) <- "Control 24 monocytes"
BN_30_monocytes$sample <- "Control 24"
BN_30_monocytes$stim <- "Healthy"
levels(BN_30_monocytes)
BN_30_monocytes
saveRDS(BN_30_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_30_monocytes.rds")
rm(BN_30,BN_30_monocytes,BN_30.markers)

####BN-31
BN_31
Idents(object = BN_31) <- "Control 25"
BN_31$sample <- "Control 25"
BN_31$stim <- "Healthy"
levels(BN_31)
BN_31
BN_31[["percent.mt"]] <- PercentageFeatureSet(BN_31, pattern = "^MT-")
VlnPlot(BN_31, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_31<- subset(BN_31, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_31
BN_31 <- PercentageFeatureSet(BN_31, pattern = "^MT-", col.name = "percent.mt")
BN_31 <- SCTransform(BN_31, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_31) <- "sample"
levels(BN_31)
BN_31<- RunPCA(BN_31, features = VariableFeatures(object = BN_31))
DimPlot(BN_31, reduction = "pca")
ElbowPlot(BN_31, ndims = 100)
pct <- BN_31[["pca"]]@stdev / sum(BN_31[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_31 <- FindNeighbors(BN_31, dims = 1:10)
BN_31 <- FindClusters(BN_31, resolution = 0.5)
BN_31 <- RunUMAP(BN_31, dims = 1:10)
DimPlot(BN_31, reduction = "umap", pt.size =0.2, label = TRUE)
BN_31.markers <- FindAllMarkers(BN_31, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_31.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_31, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_31, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_31, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_31)
new.cluster.ids <- c("0", "1", "2", "3","4","5","6","7", "CD16 monocytes", "9", "10")
names(new.cluster.ids) <- levels(BN_31)
BN_31 <- RenameIdents(BN_31, new.cluster.ids)
levels(BN_31)
DimPlot(BN_31, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_31, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_31.rds")
table(Idents(BN_31)) 

#cluster 8 monocytes seem to be intermediate 

#subset  and save CD16 monocytes 
BN_31_monocytes <- subset(BN_31, idents = c("CD16 monocytes"))
table(Idents(BN_31_monocytes))

Idents(object = BN_31_monocytes) <- "Control 25 monocytes"
BN_31_monocytes$sample <- "Control 25"
BN_31_monocytes$stim <- "Healthy"
levels(BN_31_monocytes)
BN_31_monocytes
saveRDS(BN_31_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_31_monocytes.rds")
rm(BN_31,BN_31_monocytes,BN_31.markers)

####BN-32
BN_32
Idents(object = BN_32) <- "Control 26"
BN_32$sample <- "Control 26"
BN_32$stim <- "Healthy"
levels(BN_32)
BN_32
BN_32[["percent.mt"]] <- PercentageFeatureSet(BN_32, pattern = "^MT-")
VlnPlot(BN_32, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_32<- subset(BN_32, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_32
BN_32 <- PercentageFeatureSet(BN_32, pattern = "^MT-", col.name = "percent.mt")
BN_32 <- SCTransform(BN_32, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_32) <- "sample"
levels(BN_32)
BN_32<- RunPCA(BN_32, features = VariableFeatures(object = BN_32))
DimPlot(BN_32, reduction = "pca")
ElbowPlot(BN_32, ndims = 100)
pct <- BN_32[["pca"]]@stdev / sum(BN_32[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_32 <- FindNeighbors(BN_32, dims = 1:5)
BN_32 <- FindClusters(BN_32, resolution = 0.5)
BN_32 <- RunUMAP(BN_32, dims = 1:5)
DimPlot(BN_32, reduction = "umap", pt.size =0.2, label = TRUE)
BN_32.markers <- FindAllMarkers(BN_32, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_32.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_32, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_32, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_32, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_32)
new.cluster.ids <- c("0", "1", "CD16 monocytes")
names(new.cluster.ids) <- levels(BN_32)
BN_32 <- RenameIdents(BN_32, new.cluster.ids)
levels(BN_32)
DimPlot(BN_32, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_32, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_32.rds")
table(Idents(BN_32)) 

#very few cd16 monocytes do not include this donor

#subset  and save CD16 monocytes 
BN_32_monocytes <- subset(BN_32, idents = c("CD16 monocytes"))
table(Idents(BN_32_monocytes))

Idents(object = BN_32_monocytes) <- "Control 26 monocytes"
BN_32_monocytes$sample <- "Control 26"
BN_32_monocytes$stim <- "Healthy"
levels(BN_32_monocytes)
BN_32_monocytes
saveRDS(BN_32_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_32_monocytes.rds")
rm(BN_32,BN_32_monocytes,BN_32.markers)

####BN-33
BN_33
Idents(object = BN_33) <- "Control 27"
BN_33$sample <- "Control 27"
BN_33$stim <- "Healthy"
levels(BN_33)
BN_33
BN_33[["percent.mt"]] <- PercentageFeatureSet(BN_33, pattern = "^MT-")
VlnPlot(BN_33, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_33<- subset(BN_33, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_33
BN_33 <- PercentageFeatureSet(BN_33, pattern = "^MT-", col.name = "percent.mt")
BN_33 <- SCTransform(BN_33, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_33) <- "sample"
levels(BN_33)
BN_33<- RunPCA(BN_33, features = VariableFeatures(object = BN_33))
DimPlot(BN_33, reduction = "pca")
ElbowPlot(BN_33, ndims = 100)
pct <- BN_33[["pca"]]@stdev / sum(BN_33[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_33 <- FindNeighbors(BN_33, dims = 1:7)
BN_33 <- FindClusters(BN_33, resolution = 0.5)
BN_33 <- RunUMAP(BN_33, dims = 1:7)
DimPlot(BN_33, reduction = "umap", pt.size =0.2, label = TRUE)
BN_33.markers <- FindAllMarkers(BN_33, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_33.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_33, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_33, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_33, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_33)
new.cluster.ids <- c("0", "1", "2", "3", "4", "CD16 monocytes", "6", "CD16monocytes", "8")
names(new.cluster.ids) <- levels(BN_33)
BN_33 <- RenameIdents(BN_33, new.cluster.ids)
levels(BN_33)
DimPlot(BN_33, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_33, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_33.rds")
table(Idents(BN_33)) 

#subset  and save CD16 monocytes 
BN_33_monocytes <- subset(BN_33, idents = c("CD16 monocytes", "CD16monocytes"))
table(Idents(BN_33_monocytes))

Idents(object = BN_33_monocytes) <- "Control 27 monocytes"
BN_33_monocytes$sample <- "Control 27"
BN_33_monocytes$stim <- "Healthy"
levels(BN_33_monocytes)
BN_33_monocytes
saveRDS(BN_33_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_33_monocytes.rds")
rm(BN_33,BN_33_monocytes,BN_33.markers)


####BN-34
BN_34
Idents(object = BN_34) <- "Control 28"
BN_34$sample <- "Control 28"
BN_34$stim <- "Healthy"
levels(BN_34)
BN_34
BN_34[["percent.mt"]] <- PercentageFeatureSet(BN_34, pattern = "^MT-")
VlnPlot(BN_34, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_34<- subset(BN_34, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_34
BN_34 <- PercentageFeatureSet(BN_34, pattern = "^MT-", col.name = "percent.mt")
BN_34 <- SCTransform(BN_34, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_34) <- "sample"
levels(BN_34)
BN_34<- RunPCA(BN_34, features = VariableFeatures(object = BN_34))
DimPlot(BN_34, reduction = "pca")
ElbowPlot(BN_34, ndims = 100)
pct <- BN_34[["pca"]]@stdev / sum(BN_34[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_34 <- FindNeighbors(BN_34, dims = 1:14)
BN_34 <- FindClusters(BN_34, resolution = 0.5)
BN_34 <- RunUMAP(BN_34, dims = 1:14)
DimPlot(BN_34, reduction = "umap", pt.size =0.2, label = TRUE)
BN_34.markers <- FindAllMarkers(BN_34, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_34.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_34, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_34, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_34, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_34)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5", "6", "7", "8", "CD16 monocytes", "10", "11")
names(new.cluster.ids) <- levels(BN_34)
BN_34 <- RenameIdents(BN_34, new.cluster.ids)
levels(BN_34)
DimPlot(BN_34, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_34, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_34.rds")
table(Idents(BN_34)) 

#subset  and save CD16 monocytes 
BN_34_monocytes <- subset(BN_34, idents = c("CD16 monocytes"))
table(Idents(BN_34_monocytes))

Idents(object = BN_34_monocytes) <- "Control 28 monocytes"
BN_34_monocytes$sample <- "Control 28"
BN_34_monocytes$stim <- "Healthy"
levels(BN_34_monocytes)
BN_34_monocytes
saveRDS(BN_34_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_34_monocytes.rds")
rm(BN_34,BN_34_monocytes,BN_34.markers)

####BN-35
BN_35
Idents(object = BN_35) <- "Control 29"
BN_35$sample <- "Control 29"
BN_35$stim <- "Healthy"
levels(BN_35)
BN_35
BN_35[["percent.mt"]] <- PercentageFeatureSet(BN_35, pattern = "^MT-")
VlnPlot(BN_35, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_35<- subset(BN_35, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_35
BN_35 <- PercentageFeatureSet(BN_35, pattern = "^MT-", col.name = "percent.mt")
BN_35 <- SCTransform(BN_35, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_35) <- "sample"
levels(BN_35)
BN_35<- RunPCA(BN_35, features = VariableFeatures(object = BN_35))
DimPlot(BN_35, reduction = "pca")
ElbowPlot(BN_35, ndims = 100)
pct <- BN_35[["pca"]]@stdev / sum(BN_35[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_35 <- FindNeighbors(BN_35, dims = 1:14)
BN_35 <- FindClusters(BN_35, resolution = 0.5)
BN_35 <- RunUMAP(BN_35, dims = 1:14)
DimPlot(BN_35, reduction = "umap", pt.size =0.2, label = TRUE)
BN_35.markers <- FindAllMarkers(BN_35, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_35.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_35, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_35, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_35, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_35)
new.cluster.ids <- c("0", "1", "2", "3", "4", "5", "6", "CD16 monocytes", "8", "9", "10")
names(new.cluster.ids) <- levels(BN_35)
BN_35 <- RenameIdents(BN_35, new.cluster.ids)
levels(BN_35)
DimPlot(BN_35, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_35, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_35.rds")
table(Idents(BN_35)) 

#subset  and save CD16 monocytes 
BN_35_monocytes <- subset(BN_35, idents = c("CD16 monocytes"))
table(Idents(BN_35_monocytes))

Idents(object = BN_35_monocytes) <- "Control 29 monocytes"
BN_35_monocytes$sample <- "Control 29"
BN_35_monocytes$stim <- "Healthy"
levels(BN_35_monocytes)
BN_35_monocytes
saveRDS(BN_35_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_35_monocytes.rds")
rm(BN_35,BN_35_monocytes,BN_35.markers)

####BN-36
BN_36
Idents(object = BN_36) <- "Control 30"
BN_36$sample <- "Control 30"
BN_36$stim <- "Healthy"
levels(BN_36)
BN_36
BN_36[["percent.mt"]] <- PercentageFeatureSet(BN_36, pattern = "^MT-")
VlnPlot(BN_36, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_36<- subset(BN_36, subset = nFeature_RNA > 200 & nFeature_RNA < 2000 & percent.mt < 25)
BN_36
BN_36 <- PercentageFeatureSet(BN_36, pattern = "^MT-", col.name = "percent.mt")
BN_36 <- SCTransform(BN_36, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_36) <- "sample"
levels(BN_36)
BN_36<- RunPCA(BN_36, features = VariableFeatures(object = BN_36))
DimPlot(BN_36, reduction = "pca")
ElbowPlot(BN_36, ndims = 100)
pct <- BN_36[["pca"]]@stdev / sum(BN_36[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_36 <- FindNeighbors(BN_36, dims = 1:8)
BN_36 <- FindClusters(BN_36, resolution = 0.5)
BN_36 <- RunUMAP(BN_36, dims = 1:8)
DimPlot(BN_36, reduction = "umap", pt.size =0.2, label = TRUE)
BN_36.markers <- FindAllMarkers(BN_36, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_36.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_36, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_36, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_36, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_36)
new.cluster.ids <- c("0", "1", "2", "3", "CD16 monocytes", "5")
names(new.cluster.ids) <- levels(BN_36)
BN_36 <- RenameIdents(BN_36, new.cluster.ids)
levels(BN_36)
DimPlot(BN_36, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_36, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_36.rds")
table(Idents(BN_36)) 

#subset  and save CD16 monocytes 
BN_36_monocytes <- subset(BN_36, idents = c("CD16 monocytes"))
table(Idents(BN_36_monocytes))

#Do not use this donor very few cells

Idents(object = BN_36_monocytes) <- "Control 30 monocytes"
BN_36_monocytes$sample <- "Control 30"
BN_36_monocytes$stim <- "Healthy"
levels(BN_36_monocytes)
BN_36_monocytes
saveRDS(BN_36_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_36_monocytes.rds")
rm(BN_36,BN_36_monocytes,BN_36.markers)


####BN-37
BN_37
Idents(object = BN_37) <- "Control 31"
BN_37$sample <- "Control 31"
BN_37$stim <- "Healthy"
levels(BN_37)
BN_37
BN_37[["percent.mt"]] <- PercentageFeatureSet(BN_37, pattern = "^MT-")
VlnPlot(BN_37, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_37<- subset(BN_37, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_37
BN_37 <- PercentageFeatureSet(BN_37, pattern = "^MT-", col.name = "percent.mt")
BN_37 <- SCTransform(BN_37, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_37) <- "sample"
levels(BN_37)
BN_37<- RunPCA(BN_37, features = VariableFeatures(object = BN_37))
DimPlot(BN_37, reduction = "pca")
ElbowPlot(BN_37, ndims = 100)
pct <- BN_37[["pca"]]@stdev / sum(BN_37[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_37 <- FindNeighbors(BN_37, dims = 1:11)
BN_37 <- FindClusters(BN_37, resolution = 0.5)
BN_37 <- RunUMAP(BN_37, dims = 1:11)
DimPlot(BN_37, reduction = "umap", pt.size =0.2, label = TRUE)
BN_37.markers <- FindAllMarkers(BN_37, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_37.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_37, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_37, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_37, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_37)

#no CD16 monocytes

new.cluster.ids <- c("0", "1", "2", "3", "CD16 monocytes", "5", "6", "7")
names(new.cluster.ids) <- levels(BN_37)
BN_37 <- RenameIdents(BN_37, new.cluster.ids)
levels(BN_37)
DimPlot(BN_37, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_37, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_37.rds")
table(Idents(BN_37)) 

#subset  and save CD16 monocytes 
BN_37_monocytes <- subset(BN_37, idents = c("CD16 monocytes"))
table(Idents(BN_37_monocytes))

#Do not use this donor very few cells

Idents(object = BN_37_monocytes) <- "Control 31 monocytes"
BN_37_monocytes$sample <- "Control 31"
BN_37_monocytes$stim <- "Healthy"
levels(BN_37_monocytes)
BN_37_monocytes
saveRDS(BN_37_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/Cohort 2 Bonn/BN_37_monocytes.rds")
rm(BN_37,BN_37_monocytes,BN_37.markers)


#####start extracting from covid samples
####BN-01
BN_01
Idents(object = BN_01) <- "Cov2 Sample 19c"
BN_01$sample <- "Cov2 Sample 19c"
BN_01$stim <- "Cov2"
levels(BN_01)
BN_01
BN_01[["percent.mt"]] <- PercentageFeatureSet(BN_01, pattern = "^MT-")
VlnPlot(BN_01, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_01<- subset(BN_01, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_01
BN_01 <- PercentageFeatureSet(BN_01, pattern = "^MT-", col.name = "percent.mt")
BN_01 <- SCTransform(BN_01, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_01) <- "sample"
levels(BN_01)
BN_01<- RunPCA(BN_01, features = VariableFeatures(object = BN_01))
DimPlot(BN_01, reduction = "pca")
ElbowPlot(BN_01, ndims = 100)
pct <- BN_01[["pca"]]@stdev / sum(BN_01[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_01 <- FindNeighbors(BN_01, dims = 1:15)
BN_01 <- FindClusters(BN_01, resolution = 0.5)
BN_01 <- RunUMAP(BN_01, dims = 1:15)
DimPlot(BN_01, reduction = "umap", pt.size =0.2, label = TRUE)
BN_01.markers <- FindAllMarkers(BN_01, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_01.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_01, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_01, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_01, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_01)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_01)
BN_01 <- RenameIdents(BN_01, new.cluster.ids)
levels(BN_01)
DimPlot(BN_01, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

##save before extracting CD16 monocytes 
saveRDS(BN_01, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_01.rds")
table(Idents(BN_01)) 


#subset  and save CD16 monocytes 
BN_01_monocytes <- subset(BN_01, idents = c("CD16 monocytes"))
table(Idents(BN_01_monocytes))

Idents(object = BN_01_monocytes) <- "Cov2 Sample 19c"
BN_01_monocytes$sample <- "Cov2 Sample 19c"
BN_01_monocytes$stim <- "Cov2"
levels(BN_01_monocytes)
BN_01_monocytes
saveRDS(BN_01_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_01_monocytes.rds")
rm(BN_01,BN_01_monocytes,BN_01.markers)


####BN-02
BN_02
Idents(object = BN_02) <- "Cov2 Sample 20c"
BN_02$sample <- "Cov2 Sample 20c"
BN_02$stim <- "Cov2"
levels(BN_02)
BN_02
BN_02[["percent.mt"]] <- PercentageFeatureSet(BN_02, pattern = "^MT-")
VlnPlot(BN_02, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_02<- subset(BN_02, subset = nFeature_RNA > 200 & nFeature_RNA < 2200 & percent.mt < 25)
BN_02
BN_02 <- PercentageFeatureSet(BN_02, pattern = "^MT-", col.name = "percent.mt")
BN_02 <- SCTransform(BN_02, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_02) <- "sample"
levels(BN_02)
BN_02<- RunPCA(BN_02, features = VariableFeatures(object = BN_02))
DimPlot(BN_02, reduction = "pca")
ElbowPlot(BN_02, ndims = 100)
pct <- BN_02[["pca"]]@stdev / sum(BN_02[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_02 <- FindNeighbors(BN_02, dims = 1:11)
BN_02 <- FindClusters(BN_02, resolution = 0.5)
BN_02 <- RunUMAP(BN_02, dims = 1:11)
DimPlot(BN_02, reduction = "umap", pt.size =0.2, label = TRUE)
BN_02.markers <- FindAllMarkers(BN_02, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_02.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_02, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_02, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_02, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_02)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_02)
BN_02 <- RenameIdents(BN_02, new.cluster.ids)
levels(BN_02)
DimPlot(BN_02, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#no CD16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_02, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_02.rds")
table(Idents(BN_02)) 

#subset  and save CD16 monocytes 
BN_02_monocytes <- subset(BN_02, idents = c("CD16 monocytes"))
table(Idents(BN_02_monocytes))

Idents(object = BN_02_monocytes) <- "Cov2 Sample 20c"
BN_02_monocytes$sample <- "Cov2 Sample 20c"
BN_02_monocytes$stim <- "Cov2"
levels(BN_02_monocytes)
BN_02_monocytes
saveRDS(BN_02_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_02_monocytes.rds")
rm(BN_02,BN_02_monocytes,BN_02.markers)

####BN-03
BN_03
Idents(object = BN_03) <- "Cov2 Sample 21c"
BN_03$sample <- "Cov2 Sample 21c"
BN_03$stim <- "Cov2"
levels(BN_03)
BN_03
BN_03[["percent.mt"]] <- PercentageFeatureSet(BN_03, pattern = "^MT-")
VlnPlot(BN_03, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_03<- subset(BN_03, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_03
BN_03 <- PercentageFeatureSet(BN_03, pattern = "^MT-", col.name = "percent.mt")
BN_03 <- SCTransform(BN_03, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_03) <- "sample"
levels(BN_03)
BN_03<- RunPCA(BN_03, features = VariableFeatures(object = BN_03))
DimPlot(BN_03, reduction = "pca")
ElbowPlot(BN_03, ndims = 100)
pct <- BN_03[["pca"]]@stdev / sum(BN_03[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_03 <- FindNeighbors(BN_03, dims = 1:14)
BN_03 <- FindClusters(BN_03, resolution = 0.5)
BN_03 <- RunUMAP(BN_03, dims = 1:14)
DimPlot(BN_03, reduction = "umap", pt.size =0.2, label = TRUE)
BN_03.markers <- FindAllMarkers(BN_03, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_03.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_03, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_03, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_03, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_03)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_03)
BN_03 <- RenameIdents(BN_03, new.cluster.ids)
levels(BN_03)
DimPlot(BN_03, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#no cd16

##save before extracting CD16 monocytes 
saveRDS(BN_03, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_03.rds")
table(Idents(BN_03)) 

#subset  and save CD16 monocytes 
BN_03_monocytes <- subset(BN_03, idents = c("CD16 monocytes"))
table(Idents(BN_03_monocytes))

Idents(object = BN_03_monocytes) <- "Cov2 Sample 21c"
BN_03_monocytes$sample <- "Cov2 Sample 21c"
BN_03_monocytes$stim <- "Cov2"
levels(BN_03_monocytes)
BN_03_monocytes
saveRDS(BN_03_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_03_monocytes.rds")
rm(BN_03,BN_03_monocytes,BN_03.markers)

####BN-04
BN_04
Idents(object = BN_04) <- "Cov2 Sample 22c"
BN_04$sample <- "Cov2 Sample 22c"
BN_04$stim <- "Cov2"
levels(BN_04)
BN_04
BN_04[["percent.mt"]] <- PercentageFeatureSet(BN_04, pattern = "^MT-")
VlnPlot(BN_04, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_04<- subset(BN_04, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_04
BN_04 <- PercentageFeatureSet(BN_04, pattern = "^MT-", col.name = "percent.mt")
BN_04 <- SCTransform(BN_04, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_04) <- "sample"
levels(BN_04)
BN_04<- RunPCA(BN_04, features = VariableFeatures(object = BN_04))
DimPlot(BN_04, reduction = "pca")
ElbowPlot(BN_04, ndims = 100)
pct <- BN_04[["pca"]]@stdev / sum(BN_04[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_04 <- FindNeighbors(BN_04, dims = 1:14)
BN_04 <- FindClusters(BN_04, resolution = 0.5)
BN_04 <- RunUMAP(BN_04, dims = 1:14)
DimPlot(BN_04, reduction = "umap", pt.size =0.2, label = TRUE)
BN_04.markers <- FindAllMarkers(BN_04, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_04.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_04, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_04, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_04, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_04)
new.cluster.ids <- c("0", "1", "2", "3","4","5","6","7" ,"8","9","CD16 monocytes", "11", "12", "13", "14", "15", "16", "17")
names(new.cluster.ids) <- levels(BN_04)
BN_04 <- RenameIdents(BN_04, new.cluster.ids)
levels(BN_04)
DimPlot(BN_04, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()


##save before extracting CD16 monocytes 
saveRDS(BN_04, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_04.rds")
table(Idents(BN_04)) 

#subset  and save CD16 monocytes 
BN_04_monocytes <- subset(BN_04, idents = c("CD16 monocytes"))
table(Idents(BN_04_monocytes))

Idents(object = BN_04_monocytes) <- "Cov2 Sample 22c"
BN_04_monocytes$sample <- "Cov2 Sample 22c"
BN_04_monocytes$stim <- "Cov2"
levels(BN_04_monocytes)
BN_04_monocytes
saveRDS(BN_04_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_04_monocytes.rds")
rm(BN_04,BN_04_monocytes,BN_04.markers)


####BN-05
BN_05
Idents(object = BN_05) <- "Cov2 Sample 23c"
BN_05$sample <- "Cov2 Sample 23c"
BN_05$stim <- "Cov2"
levels(BN_05)
BN_05
BN_05[["percent.mt"]] <- PercentageFeatureSet(BN_05, pattern = "^MT-")
VlnPlot(BN_05, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_05<- subset(BN_05, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_05
BN_05 <- PercentageFeatureSet(BN_05, pattern = "^MT-", col.name = "percent.mt")
BN_05 <- SCTransform(BN_05, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_05) <- "sample"
levels(BN_05)
BN_05<- RunPCA(BN_05, features = VariableFeatures(object = BN_05))
DimPlot(BN_05, reduction = "pca")
ElbowPlot(BN_05, ndims = 100)
pct <- BN_05[["pca"]]@stdev / sum(BN_05[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_05 <- FindNeighbors(BN_05, dims = 1:12)
BN_05 <- FindClusters(BN_05, resolution = 0.5)
BN_05 <- RunUMAP(BN_05, dims = 1:12)
DimPlot(BN_05, reduction = "umap", pt.size =0.2, label = TRUE)
BN_05.markers <- FindAllMarkers(BN_05, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_05.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_05, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_05, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_05, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_05)
new.cluster.ids <- c("0", "1", "2", "3","4","CD16 monocytes","6","7" ,"8","9","10", "11", "12", "13", "14")
names(new.cluster.ids) <- levels(BN_05)
BN_05 <- RenameIdents(BN_05, new.cluster.ids)
levels(BN_05)
DimPlot(BN_05, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#cluster 5 are intermediate monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_05, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_05.rds")
table(Idents(BN_05)) 

#subset  and save CD16 monocytes 
BN_05_monocytes <- subset(BN_05, idents = c("CD16 monocytes"))
table(Idents(BN_05_monocytes))

Idents(object = BN_05_monocytes) <- "Cov2 Sample 23c"
BN_05_monocytes$sample <- "Cov2 Sample 23c"
BN_05_monocytes$stim <- "Cov2"
levels(BN_05_monocytes)
BN_05_monocytes
saveRDS(BN_05_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_05_monocytes.rds")
rm(BN_05,BN_05_monocytes,BN_05.markers)

####BN-06
BN_06
Idents(object = BN_06) <- "Cov2 Sample 24c"
BN_06$sample <- "Cov2 Sample 24c"
BN_06$stim <- "Cov2"
levels(BN_06)
BN_06
BN_06[["percent.mt"]] <- PercentageFeatureSet(BN_06, pattern = "^MT-")
VlnPlot(BN_06, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_06<- subset(BN_06, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 25)
BN_06
BN_06 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_06 <- SCTransform(BN_06, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_06) <- "sample"
levels(BN_06)
BN_06<- RunPCA(BN_06, features = VariableFeatures(object = BN_06))
DimPlot(BN_06, reduction = "pca")
ElbowPlot(BN_06, ndims = 100)
pct <- BN_06[["pca"]]@stdev / sum(BN_06[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_06 <- FindNeighbors(BN_06, dims = 1:13)
BN_06 <- FindClusters(BN_06, resolution = 0.5)
BN_06 <- RunUMAP(BN_06, dims = 1:13)
DimPlot(BN_06, reduction = "umap", pt.size =0.2, label = TRUE)
BN_06.markers <- FindAllMarkers(BN_06, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_06.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_06, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_06, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_06, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_06)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12")
names(new.cluster.ids) <- levels(BN_06)
BN_06 <- RenameIdents(BN_06, new.cluster.ids)
levels(BN_06)
DimPlot(BN_06, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16

##save before extracting CD16 monocytes 
saveRDS(BN_06, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_06.rds")
table(Idents(BN_06)) 

#subset  and save CD16 monocytes 
BN_06_monocytes <- subset(BN_06, idents = c("CD16 monocytes"))
table(Idents(BN_06_monocytes))

Idents(object = BN_06_monocytes) <- "Cov2 Sample 24c"
BN_06_monocytes$sample <- "Cov2 Sample 24c"
BN_06_monocytes$stim <- "Cov2"
levels(BN_06_monocytes)
BN_06_monocytes
saveRDS(BN_06_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_06_monocytes.rds")
rm(BN_06,BN_06_monocytes,BN_06.markers)

####BN-07
BN_07
Idents(object = BN_07) <- "Cov2 Sample 25c"
BN_07$sample <- "Cov2 Sample 25c"
BN_07$stim <- "Cov2"
levels(BN_07)
BN_07
BN_07[["percent.mt"]] <- PercentageFeatureSet(BN_07, pattern = "^MT-")
VlnPlot(BN_07, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_07<- subset(BN_07, subset = nFeature_RNA > 200 & nFeature_RNA < 4500 & percent.mt < 25)
BN_07
BN_07 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_07 <- SCTransform(BN_07, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_07) <- "sample"
levels(BN_07)
BN_07<- RunPCA(BN_07, features = VariableFeatures(object = BN_07))
DimPlot(BN_07, reduction = "pca")
ElbowPlot(BN_07, ndims = 100)
pct <- BN_07[["pca"]]@stdev / sum(BN_07[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_07 <- FindNeighbors(BN_07, dims = 1:13)
BN_07 <- FindClusters(BN_07, resolution = 0.5)
BN_07 <- RunUMAP(BN_07, dims = 1:13)
DimPlot(BN_07, reduction = "umap", pt.size =0.2, label = TRUE)
BN_07.markers <- FindAllMarkers(BN_07, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_07.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_07, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_07, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_07, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_07)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_07)
BN_07 <- RenameIdents(BN_07, new.cluster.ids)
levels(BN_07)
DimPlot(BN_07, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_07, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_07.rds")
table(Idents(BN_07)) 

#subset  and save CD16 monocytes 
BN_07_monocytes <- subset(BN_07, idents = c("CD16 monocytes"))
table(Idents(BN_07_monocytes))

Idents(object = BN_07_monocytes) <- "Cov2 Sample 25c"
BN_07_monocytes$sample <- "Cov2 Sample 25c"
BN_07_monocytes$stim <- "Cov2"
levels(BN_07_monocytes)
BN_07_monocytes
saveRDS(BN_07_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_07_monocytes.rds")
rm(BN_07,BN_07_monocytes,BN_07.markers)

####BN-08
BN_08
Idents(object = BN_08) <- "Cov2 Sample 26c"
BN_08$sample <- "Cov2 Sample 26c"
BN_08$stim <- "Cov2"
levels(BN_08)
BN_08
BN_08[["percent.mt"]] <- PercentageFeatureSet(BN_08, pattern = "^MT-")
VlnPlot(BN_08, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_08<- subset(BN_08, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_08
BN_08 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_08 <- SCTransform(BN_08, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_08) <- "sample"
levels(BN_08)
BN_08<- RunPCA(BN_08, features = VariableFeatures(object = BN_08))
DimPlot(BN_08, reduction = "pca")
ElbowPlot(BN_08, ndims = 100)
pct <- BN_08[["pca"]]@stdev / sum(BN_08[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_08 <- FindNeighbors(BN_08, dims = 1:13)
BN_08 <- FindClusters(BN_08, resolution = 0.5)
BN_08 <- RunUMAP(BN_08, dims = 1:13)
DimPlot(BN_08, reduction = "umap", pt.size =0.2, label = TRUE)
BN_08.markers <- FindAllMarkers(BN_08, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_08.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_08, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_08, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_08, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_08)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_08)
BN_08 <- RenameIdents(BN_08, new.cluster.ids)
levels(BN_08)
DimPlot(BN_08, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16

##save before extracting CD16 monocytes 
saveRDS(BN_08, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_08.rds")
table(Idents(BN_08)) 

#subset  and save CD16 monocytes 
BN_08_monocytes <- subset(BN_08, idents = c("CD16 monocytes"))
table(Idents(BN_08_monocytes))

Idents(object = BN_08_monocytes) <- "Cov2 Sample 26c"
BN_08_monocytes$sample <- "Cov2 Sample 26c"
BN_08_monocytes$stim <- "Cov2"
levels(BN_08_monocytes)
BN_08_monocytes
saveRDS(BN_08_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_08_monocytes.rds")
rm(BN_08,BN_08_monocytes,BN_08.markers)

####BN-10
BN_10
Idents(object = BN_10) <- "Cov2 Sample 27c"
BN_10$sample <- "Cov2 Sample 27c"
BN_10$stim <- "Cov2"
levels(BN_10)
BN_10
BN_10[["percent.mt"]] <- PercentageFeatureSet(BN_10, pattern = "^MT-")
VlnPlot(BN_10, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_10<- subset(BN_10, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 25)
BN_10
BN_10 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_10 <- SCTransform(BN_10, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_10) <- "sample"
levels(BN_10)
BN_10<- RunPCA(BN_10, features = VariableFeatures(object = BN_10))
DimPlot(BN_10, reduction = "pca")
ElbowPlot(BN_10, ndims = 100)
pct <- BN_10[["pca"]]@stdev / sum(BN_10[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_10 <- FindNeighbors(BN_10, dims = 1:13)
BN_10 <- FindClusters(BN_10, resolution = 0.5)
BN_10 <- RunUMAP(BN_10, dims = 1:13)
DimPlot(BN_10, reduction = "umap", pt.size =0.2, label = TRUE)
BN_10.markers <- FindAllMarkers(BN_10, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_10.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_10, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_10, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_10, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_10)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_10)
BN_10 <- RenameIdents(BN_10, new.cluster.ids)
levels(BN_10)
DimPlot(BN_10, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#very few cd16

##save before extracting CD16 monocytes 
saveRDS(BN_10, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_10.rds")
table(Idents(BN_10)) 

#subset  and save CD16 monocytes 
BN_10_monocytes <- subset(BN_10, idents = c("CD16 monocytes"))
table(Idents(BN_10_monocytes))

Idents(object = BN_10_monocytes) <- "Cov2 Sample 27c"
BN_10_monocytes$sample <- "Cov2 Sample 27c"
BN_10_monocytes$stim <- "Cov2"
levels(BN_10_monocytes)
BN_10_monocytes
saveRDS(BN_10_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_10_monocytes.rds")
rm(BN_10,BN_10_monocytes,BN_10.markers)

####BN-11
BN_11
Idents(object = BN_11) <- "Cov2 Sample 28c"
BN_11$sample <- "Cov2 Sample 28c"
BN_11$stim <- "Cov2"
levels(BN_11)
BN_11
BN_11[["percent.mt"]] <- PercentageFeatureSet(BN_11, pattern = "^MT-")
VlnPlot(BN_11, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_11<- subset(BN_11, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 25)
BN_11
BN_11 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_11 <- SCTransform(BN_11, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_11) <- "sample"
levels(BN_11)
BN_11<- RunPCA(BN_11, features = VariableFeatures(object = BN_11))
DimPlot(BN_11, reduction = "pca")
ElbowPlot(BN_11, ndims = 100)
pct <- BN_11[["pca"]]@stdev / sum(BN_11[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_11 <- FindNeighbors(BN_11, dims = 1:13)
BN_11 <- FindClusters(BN_11, resolution = 0.5)
BN_11 <- RunUMAP(BN_11, dims = 1:13)
DimPlot(BN_11, reduction = "umap", pt.size =0.2, label = TRUE)
BN_11.markers <- FindAllMarkers(BN_11, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_11.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_11, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_11, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_11, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_11)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_11)
BN_11 <- RenameIdents(BN_11, new.cluster.ids)
levels(BN_11)
DimPlot(BN_11, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#very few cd16

##save before extracting CD16 monocytes 
saveRDS(BN_11, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_11.rds")
table(Idents(BN_11)) 

#subset  and save CD16 monocytes 
BN_11_monocytes <- subset(BN_11, idents = c("CD16 monocytes"))
table(Idents(BN_11_monocytes))

Idents(object = BN_11_monocytes) <- "Cov2 Sample 28c"
BN_11_monocytes$sample <- "Cov2 Sample 28c"
BN_11_monocytes$stim <- "Cov2"
levels(BN_11_monocytes)
BN_11_monocytes
saveRDS(BN_11_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_11_monocytes.rds")
rm(BN_11,BN_11_monocytes,BN_11.markers)


####BN-12
BN_12
Idents(object = BN_12) <- "Cov2 Sample 29c"
BN_12$sample <- "Cov2 Sample 29c"
BN_12$stim <- "Cov2"
levels(BN_12)
BN_12
BN_12[["percent.mt"]] <- PercentageFeatureSet(BN_12, pattern = "^MT-")
VlnPlot(BN_12, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_12<- subset(BN_12, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 25)
BN_12
BN_12 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_12 <- SCTransform(BN_12, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_12) <- "sample"
levels(BN_12)
BN_12<- RunPCA(BN_12, features = VariableFeatures(object = BN_12))
DimPlot(BN_12, reduction = "pca")
ElbowPlot(BN_12, ndims = 100)
pct <- BN_12[["pca"]]@stdev / sum(BN_12[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_12 <- FindNeighbors(BN_12, dims = 1:13)
BN_12 <- FindClusters(BN_12, resolution = 0.5)
BN_12 <- RunUMAP(BN_12, dims = 1:13)
DimPlot(BN_12, reduction = "umap", pt.size =0.2, label = TRUE)
BN_12.markers <- FindAllMarkers(BN_12, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_12.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_12, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_12, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_12, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_12)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_12)
BN_12 <- RenameIdents(BN_12, new.cluster.ids)
levels(BN_12)
DimPlot(BN_12, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16
##save before extracting CD16 monocytes 
saveRDS(BN_12, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_12.rds")
table(Idents(BN_12)) 

#subset  and save CD16 monocytes 
BN_12_monocytes <- subset(BN_12, idents = c("CD16 monocytes"))
table(Idents(BN_12_monocytes))

Idents(object = BN_12_monocytes) <- "Cov2 Sample 29c"
BN_12_monocytes$sample <- "Cov2 Sample 29c"
BN_12_monocytes$stim <- "Cov2"
levels(BN_12_monocytes)
BN_12_monocytes
saveRDS(BN_12_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_12_monocytes.rds")
rm(BN_12,BN_12_monocytes,BN_12.markers)


####BN-13
BN_13
Idents(object = BN_13) <- "Cov2 Sample 30c"
BN_13$sample <- "Cov2 Sample 30c"
BN_13$stim <- "Cov2"
levels(BN_13)
BN_13
BN_13[["percent.mt"]] <- PercentageFeatureSet(BN_13, pattern = "^MT-")
VlnPlot(BN_13, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_13<- subset(BN_13, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 25)
BN_13
BN_13 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_13 <- SCTransform(BN_13, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_13) <- "sample"
levels(BN_13)
BN_13<- RunPCA(BN_13, features = VariableFeatures(object = BN_13))
DimPlot(BN_13, reduction = "pca")
ElbowPlot(BN_13, ndims = 100)
pct <- BN_13[["pca"]]@stdev / sum(BN_13[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_13 <- FindNeighbors(BN_13, dims = 1:13)
BN_13 <- FindClusters(BN_13, resolution = 0.5)
BN_13 <- RunUMAP(BN_13, dims = 1:13)
DimPlot(BN_13, reduction = "umap", pt.size =0.2, label = TRUE)
BN_13.markers <- FindAllMarkers(BN_13, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_13.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_13, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_13, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_13, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_13)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_13)
BN_13 <- RenameIdents(BN_13, new.cluster.ids)
levels(BN_13)
DimPlot(BN_13, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_13, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_13.rds")
table(Idents(BN_13)) 

#subset  and save CD16 monocytes 
BN_13_monocytes <- subset(BN_13, idents = c("CD16 monocytes"))
table(Idents(BN_13_monocytes))

Idents(object = BN_13_monocytes) <- "Cov2 Sample 30c"
BN_13_monocytes$sample <- "Cov2 Sample 30c"
BN_13_monocytes$stim <- "Cov2"
levels(BN_13_monocytes)
BN_13_monocytes
saveRDS(BN_13_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_13_monocytes.rds")
rm(BN_13,BN_13_monocytes,BN_13.markers)


####BN-14
BN_14
Idents(object = BN_14) <- "Cov2 Sample 31c"
BN_14$sample <- "Cov2 Sample 31c"
BN_14$stim <- "Cov2"
levels(BN_14)
BN_14
BN_14[["percent.mt"]] <- PercentageFeatureSet(BN_14, pattern = "^MT-")
VlnPlot(BN_14, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_14<- subset(BN_14, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 25)
BN_14
BN_14 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_14 <- SCTransform(BN_14, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_14) <- "sample"
levels(BN_14)
BN_14<- RunPCA(BN_14, features = VariableFeatures(object = BN_14))
DimPlot(BN_14, reduction = "pca")
ElbowPlot(BN_14, ndims = 100)
pct <- BN_14[["pca"]]@stdev / sum(BN_14[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_14 <- FindNeighbors(BN_14, dims = 1:13)
BN_14 <- FindClusters(BN_14, resolution = 0.5)
BN_14 <- RunUMAP(BN_14, dims = 1:13)
DimPlot(BN_14, reduction = "umap", pt.size =0.2, label = TRUE)
BN_14.markers <- FindAllMarkers(BN_14, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_14.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_14, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_14, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_14, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_14)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_14)
BN_14 <- RenameIdents(BN_14, new.cluster.ids)
levels(BN_14)
DimPlot(BN_14, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_14, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_14.rds")
table(Idents(BN_14)) 

#subset  and save CD16 monocytes 
BN_14_monocytes <- subset(BN_14, idents = c("CD16 monocytes"))
table(Idents(BN_14_monocytes))

Idents(object = BN_14_monocytes) <- "Cov2 Sample 31c"
BN_14_monocytes$sample <- "Cov2 Sample 31c"
BN_14_monocytes$stim <- "Cov2"
levels(BN_14_monocytes)
BN_14_monocytes
saveRDS(BN_14_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_14_monocytes.rds")
rm(BN_14,BN_14_monocytes,BN_14.markers)


####BN-15
BN_15
Idents(object = BN_15) <- "Cov2 Sample 32c"
BN_15$sample <- "Cov2 Sample 32c"
BN_15$stim <- "Cov2"
levels(BN_15)
BN_15
BN_15[["percent.mt"]] <- PercentageFeatureSet(BN_15, pattern = "^MT-")
VlnPlot(BN_15, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_15<- subset(BN_15, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 25)
BN_15
BN_15 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_15 <- SCTransform(BN_15, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_15) <- "sample"
levels(BN_15)
BN_15<- RunPCA(BN_15, features = VariableFeatures(object = BN_15))
DimPlot(BN_15, reduction = "pca")
ElbowPlot(BN_15, ndims = 100)
pct <- BN_15[["pca"]]@stdev / sum(BN_15[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_15 <- FindNeighbors(BN_15, dims = 1:13)
BN_15 <- FindClusters(BN_15, resolution = 0.5)
BN_15 <- RunUMAP(BN_15, dims = 1:13)
DimPlot(BN_15, reduction = "umap", pt.size =0.2, label = TRUE)
BN_15.markers <- FindAllMarkers(BN_15, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_15.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_15, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_15, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_15, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_15)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_15)
BN_15 <- RenameIdents(BN_15, new.cluster.ids)
levels(BN_15)
DimPlot(BN_15, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_15, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_15.rds")
table(Idents(BN_15)) 

#subset  and save CD16 monocytes 
BN_15_monocytes <- subset(BN_15, idents = c("CD16 monocytes"))
table(Idents(BN_15_monocytes))

Idents(object = BN_15_monocytes) <- "Cov2 Sample 32c"
BN_15_monocytes$sample <- "Cov2 Sample 32c"
BN_15_monocytes$stim <- "Cov2"
levels(BN_15_monocytes)
BN_15_monocytes
saveRDS(BN_15_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_15_monocytes.rds")
rm(BN_15,BN_15_monocytes,BN_15.markers)

####BN-16
BN_16
Idents(object = BN_16) <- "Cov2 Sample 33c"
BN_16$sample <- "Cov2 Sample 33c"
BN_16$stim <- "Cov2"
levels(BN_16)
BN_16
BN_16[["percent.mt"]] <- PercentageFeatureSet(BN_16, pattern = "^MT-")
VlnPlot(BN_16, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_16<- subset(BN_16, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_16
BN_16 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_16 <- SCTransform(BN_16, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_16) <- "sample"
levels(BN_16)
BN_16<- RunPCA(BN_16, features = VariableFeatures(object = BN_16))
DimPlot(BN_16, reduction = "pca")
ElbowPlot(BN_16, ndims = 100)
pct <- BN_16[["pca"]]@stdev / sum(BN_16[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_16 <- FindNeighbors(BN_16, dims = 1:15)
BN_16 <- FindClusters(BN_16, resolution = 0.5)
BN_16 <- RunUMAP(BN_16, dims = 1:15)
DimPlot(BN_16, reduction = "umap", pt.size =0.2, label = TRUE)
BN_16.markers <- FindAllMarkers(BN_16, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_16.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_16, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_16, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_16, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_16)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_16)
BN_16 <- RenameIdents(BN_16, new.cluster.ids)
levels(BN_16)
DimPlot(BN_16, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#no cd16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_16, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_16.rds")
table(Idents(BN_16)) 

#subset  and save CD16 monocytes 
BN_16_monocytes <- subset(BN_16, idents = c("CD16 monocytes"))
table(Idents(BN_16_monocytes))

Idents(object = BN_16_monocytes) <- "Cov2 Sample 33c"
BN_16_monocytes$sample <- "Cov2 Sample 33c"
BN_16_monocytes$stim <- "Cov2"
levels(BN_16_monocytes)
BN_16_monocytes
saveRDS(BN_16_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_16_monocytes.rds")
rm(BN_16,BN_16_monocytes,BN_16.markers)


####BN-17
BN_17
Idents(object = BN_17) <- "Cov2 Sample 34c"
BN_17$sample <- "Cov2 Sample 34c"
BN_17$stim <- "Cov2"
levels(BN_17)
BN_17
BN_17[["percent.mt"]] <- PercentageFeatureSet(BN_17, pattern = "^MT-")
VlnPlot(BN_17, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_17<- subset(BN_17, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 25)
BN_17
BN_17 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_17 <- SCTransform(BN_17, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_17) <- "sample"
levels(BN_17)
BN_17<- RunPCA(BN_17, features = VariableFeatures(object = BN_17))
DimPlot(BN_17, reduction = "pca")
ElbowPlot(BN_17, ndims = 100)
pct <- BN_17[["pca"]]@stdev / sum(BN_17[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_17 <- FindNeighbors(BN_17, dims = 1:13)
BN_17 <- FindClusters(BN_17, resolution = 0.5)
BN_17 <- RunUMAP(BN_17, dims = 1:13)
DimPlot(BN_17, reduction = "umap", pt.size =0.2, label = TRUE)
BN_17.markers <- FindAllMarkers(BN_17, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_17.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_17, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_17, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_17, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_17)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_17)
BN_17 <- RenameIdents(BN_17, new.cluster.ids)
levels(BN_17)
DimPlot(BN_17, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()


##save before extracting CD16 monocytes 
saveRDS(BN_17, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_17.rds")
table(Idents(BN_17)) 

#subset  and save CD16 monocytes 
BN_17_monocytes <- subset(BN_17, idents = c("CD16 monocytes"))
table(Idents(BN_17_monocytes))

#few cd16 monocytes

Idents(object = BN_17_monocytes) <- "Cov2 Sample 34c"
BN_17_monocytes$sample <- "Cov2 Sample 34c"
BN_17_monocytes$stim <- "Cov2"
levels(BN_17_monocytes)
BN_17_monocytes
saveRDS(BN_17_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_17_monocytes.rds")
rm(BN_17,BN_17_monocytes,BN_17.markers)


####BN-19
BN_19
Idents(object = BN_19) <- "Cov2 Sample 36c"
BN_19$sample <- "Cov2 Sample 36c"
BN_19$stim <- "Cov2"
levels(BN_19)
BN_19
BN_19[["percent.mt"]] <- PercentageFeatureSet(BN_19, pattern = "^MT-")
VlnPlot(BN_19, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
BN_19<- subset(BN_19, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 25)
BN_19
BN_19 <- PercentageFeatureSet(BN_06, pattern = "^MT-", col.name = "percent.mt")
BN_19 <- SCTransform(BN_19, vars.to.regress = "percent.mt", verbose = FALSE)
Idents(object = BN_19) <- "sample"
levels(BN_19)
BN_19<- RunPCA(BN_19, features = VariableFeatures(object = BN_19))
DimPlot(BN_19, reduction = "pca")
ElbowPlot(BN_19, ndims = 100)
pct <- BN_19[["pca"]]@stdev / sum(BN_19[["pca"]]@stdev) * 100
cumu <- cumsum(pct)
co1 <- which(cumu > 90 & pct < 5)[1]
co1
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1
pcs <- min(co1, co2)
pcs
###clustering
BN_19 <- FindNeighbors(BN_19, dims = 1:13)
BN_19 <- FindClusters(BN_19, resolution = 0.5)
BN_19 <- RunUMAP(BN_19, dims = 1:13)
DimPlot(BN_19, reduction = "umap", pt.size =0.2, label = TRUE)
BN_19.markers <- FindAllMarkers(BN_19, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
BN_19.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_logFC)
VlnPlot(BN_19, features = c( "FCGR3A","MS4A7", "CD14"))
VlnPlot(BN_19, features = c("FCGR3A","MS4A7", "CD14"), slot = "counts", log = TRUE)
FeaturePlot(BN_19, features = c("FCGR3A","MS4A7", "CD14"))
levels(BN_19)
new.cluster.ids <- c("0", "1", "CD16 monocytes", "3","4","5","6","7" ,"8","9","10", "11", "12", "13")
names(new.cluster.ids) <- levels(BN_19)
BN_19 <- RenameIdents(BN_19, new.cluster.ids)
levels(BN_19)
DimPlot(BN_19, reduction = "umap", label = TRUE, pt.size = 1, label.size = 5) + NoLegend()

#few cd16 monocytes

##save before extracting CD16 monocytes 
saveRDS(BN_19, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_19.rds")
table(Idents(BN_19)) 

#subset  and save CD16 monocytes 
BN_19_monocytes <- subset(BN_19, idents = c("CD16 monocytes"))
table(Idents(BN_19_monocytes))

Idents(object = BN_19_monocytes) <- "Cov2 Sample 36c"
BN_19_monocytes$sample <- "Cov2 Sample 36c"
BN_19_monocytes$stim <- "Cov2"
levels(BN_19_monocytes)
BN_19_monocytes
saveRDS(BN_19_monocytes, file = "/Users/bermanlab/Desktop/Covid analysis/Schulte_Schrepping_2020_COVID19_10x_PBMC_data/BN_19_monocytes.rds")
rm(BN_19,BN_19_monocytes,BN_19.markers)

###############################

